

<!DOCTYPE html>
<html lang="en" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/GeekFocus/img/fluid.png">
  <link rel="icon" href="/GeekFocus/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="banner">
  <meta name="keywords" content="Bioinformatics">
  
    <meta name="description" content="BRAKER2">
<meta property="og:type" content="article">
<meta property="og:title" content="BRAKER2">
<meta property="og:url" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/index.html">
<meta property="og:site_name" content="DrBanner">
<meta property="og:description" content="BRAKER2">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f2.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f3.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f4.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f5.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f6.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f7.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f8.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f9.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f3.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f4.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f5.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f6.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f7.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f8.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f10.png">
<meta property="article:published_time" content="2022-04-21T07:21:12.377Z">
<meta property="article:modified_time" content="2022-06-05T12:19:36.758Z">
<meta property="article:author" content="banner">
<meta property="article:tag" content="annotation">
<meta property="article:tag" content="Braker2">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/BRAKER/f2.png">
  
  
  
  <title>BRAKER2 - DrBanner</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/GeekFocus/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/GeekFocus/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/GeekFocus/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/GeekFocus/","version":"1.9.1","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"right","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":false,"follow_dnt":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":"https://wangc1106.github.io/GeekFocus","path":"window.location.pathname","ignore_local":false}},"search_path":"/GeekFocus/local-search.xml"};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/GeekFocus/js/utils.js" ></script>
  <script  src="/GeekFocus/js/color-schema.js" ></script>

  
<meta name="generator" content="Hexo 6.2.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 70vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/GeekFocus/">
      <strong>DrBanner</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/">
                <i class="iconfont icon-home-fill"></i>
                Home
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/archives/">
                <i class="iconfont icon-archive-fill"></i>
                Archives
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/categories/">
                <i class="iconfont icon-category-fill"></i>
                Categories
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/tags/">
                <i class="iconfont icon-tags-fill"></i>
                Tags
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/about/">
                <i class="iconfont icon-user-fill"></i>
                About
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              &nbsp;<i class="iconfont icon-search"></i>&nbsp;
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/GeekFocus/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="BRAKER2"></span>
          
        </div>

        
          
  <div class="mt-3">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-author" aria-hidden="true"></i>
        banner
      </span>
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2022-04-21 15:21" pubdate>
          April 21, 2022 pm
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          73k words
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          613 mins
        
      </span>
    

    
    
      
        <span id="leancloud-page-views-container" class="post-meta" style="display: none">
          <i class="iconfont icon-eye" aria-hidden="true"></i>
          <span id="leancloud-page-views"></span> views
        </span>
        
      
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">BRAKER2</h1>
            
              <p class="note note-info">
                
                  
                    Last updated on June 5, 2022 pm
                  
                
              </p>
            
            <div class="markdown-body">
              
              <p>BRAKER2</p>
<span id="more"></span>



<h1 id="Contents"><a href="#Contents" class="headerlink" title="Contents"></a>Contents</h1><ul>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#authors-of-braker">Authors</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#funding">Funding</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#what-is-braker">What is BRAKER?</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#keys-to-successful-gene-prediction">Keys to successful gene prediction</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#overview-of-modes-for-running-braker">Overview of modes for running BRAKER</a></li>
<li>Installation<ul>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#supported-software-versions">Supported software versions</a></li>
<li>BRAKER<ul>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#perl-pipeline-dependencies">Perl pipeline dependencies</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#braker-components">BRAKER components</a></li>
<li>Bioinformatics software dependencies<ul>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#mandatory-tools">Mandatory tools</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#optional-tools">Optional tools</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li>Running BRAKER<ul>
<li>BRAKER pipeline modes<ul>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#braker-with-rna-seq-data">BRAKER with RNA-Seq data</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#braker-with-proteins-of-any-evolutionary-distance">BRAKER with proteins of any evolutionary distance</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#braker-with-proteins-of-short-evolutionary-distance">BRAKER with proteins of short evolutionary distance</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#braker-with-rna-seq-and-protein-data">BRAKER with RNA-Seq and protein data</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#braker-with-short-and-long-read-RNA-Seq-and-protein-data">BRAKER with short and long read RNA-Seq and protein data</a></li>
</ul>
</li>
<li>Description of selected BRAKER command line options<ul>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--ab_initio">–ab_initio</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--augustus_args--some_argbla">–augustus_args&#x3D;–some_arg&#x3D;bla</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--coresint">–cores&#x3D;INT</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--fungus">–fungus</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--softmasking">–softmasking</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--useexisting">–useexisting</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--crf">–crf</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--lambdaint">–lambda&#x3D;int</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--utron">–UTR&#x3D;on</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--addutr">–addUTR&#x3D;on</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--stranded-">–stranded&#x3D;+,-,.,…</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#--makehub---emailyourmailde">–makehub --email&#x3D;your@mail.de</a></li>
</ul>
</li>
</ul>
</li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#output-of-braker">Output of BRAKER</a></li>
<li>Example data<ul>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#data-description">Data description</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#testing-braker-with-rna-seq-data">Testing BRAKER with RNA-Seq data</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#testing-braker-with-proteins-of-any-evolutionary-distance">Testing BRAKER with proteins of any evolutionary distance</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#testing-braker-with-proteins-of-any-evolutionary-distance-and-rna-seq">Testing BRAKER with proteins of any evolutionary distance and RNA-Seq</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#testing-braker-with-proteins-of-close-homology">Testing BRAKER with proteins of close homology</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#testing-braker-with-proteins-of-close-homology-and-rna-seq-data-rna-seq-supported-training">Testing BRAKER with proteins of close homology and RNA-Seq data (RNA-Seq supported training)</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#testing-braker-with-proteins-of-close-homoogy-and-rna-seq-data-rna-seq-and-protein-supported-training">Testing BRAKER with proteins of close homoogy and RNA-Seq data (RNA-Seq and protein supported training)</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#testing-braker-with-pre-trained-parameters">Testing BRAKER with pre-trained parameters</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#testing-braker-with-genome-sequence">Testing BRAKER with genome sequence</a></li>
</ul>
</li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#starting-braker-on-the-basis-of-previously-existing-braker-runs">Starting BRAKER on the basis of previously existing BRAKER runs</a></li>
<li>Bug reporting<ul>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#reporting-bugs-on-github">Reporting bugs on github</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#common-problems">Common problems</a></li>
</ul>
</li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#citing-braker-and-software-called-by-braker">Citing BRAKER and software called by BRAKER</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#license">License</a></li>
</ul>
<h1 id="What-is-BRAKER"><a href="#What-is-BRAKER" class="headerlink" title="What is BRAKER?"></a>What is BRAKER?</h1><p>The rapidly growing number of sequenced genomes requires fully automated methods for accurate gene structure annotation. With this goal in mind, we have developed <font color="blue">BRAKER1</font><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f1">R1</a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f0">R0</a>, a <strong>combination</strong> of <font color="blue">GeneMark-ET</font> <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f2">R2</a> and <font color="blue">AUGUSTUS</font> <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f3">R3, </a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f4">R4</a>, that uses genomic and RNA-Seq data to automatically generate full gene structure annotations in novel genome.</p>
<p>However, the quality of RNA-Seq data that is available for annotating a novel genome is variable, and in some cases, RNA-Seq data is not available, at all.</p>
<p><font color="blue">BRAKER2 is an extension of BRAKER1</font> which allows for <strong>fully automated training</strong> of the gene prediction tools <font color="blue"><strong>GeneMark</strong>-EX</font> <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f14">R14, </a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f15">R15, </a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f17">R17, </a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#g1">F1</a> and <font color="blue"><strong>AUGUSTUS</strong></font> from <strong>RNA-Seq and&#x2F;or protein homology</strong> information, and that integrates the extrinsic evidence from RNA-Seq and protein homology information into the <strong>prediction</strong>.</p>
<p>In contrast to other available methods that rely on protein homology information, BRAKER2 reaches high gene prediction accuracy even in the absence of the annotation of very closely related species and in the absence of RNA-Seq data.</p>
<p>In this user guide, we will refer to <font color="blue">BRAKER1 and BRAKER2</font> simply as <strong>BRAKER</strong> because they are <font color="blue">executed by the same script</font> (<code>braker.pl</code>).</p>
<h1 id="Keys-to-successful-gene-prediction"><a href="#Keys-to-successful-gene-prediction" class="headerlink" title="Keys to successful gene prediction"></a>Keys to successful gene prediction</h1><ul>
<li>Use a high quality genome assembly. If you have a huge number of very short scaffolds in your genome assembly, those short scaffolds will likely increase runtime dramatically but will not increase prediction accuracy. <strong>使用高质量组装</strong>。时间缩短质量提高。</li>
<li><strong>contig名字简短</strong>。Use simple scaffold names in the genome file (e.g. <code>&gt;contig1</code> will work better than <code>&gt;contig1my custom species namesome putative function /more/information/  and lots of special characters %&amp;!*()&#123;&#125;</code>). Make the scaffold names in all your fasta files simple before running any alignment program.</li>
<li>In order to predict genes accurately in a novel genome, the <font color="blue">genome should be <strong>masked</strong> for repeats</font>. This will <strong>avoid the prediction</strong> of false positive gene structures <strong>in repetitive and low complexitiy</strong> regions. Repeat masking is <strong>also essential for mapping RNA</strong>-Seq data to a genome with some tools (other RNA-Seq mappers, such as HISAT2, ignore masking information). In case of GeneMark-EX and AUGUSTUS, <strong>softmasking</strong> (i.e. putting repeat regions into lower case letters and all other regions into upper case letters) leads to <font color="blue"><strong>better results than hardmasking</strong></font> (i.e. replacing letters in repetitive regions by the letter <code>N</code> for unknown nucleotide). If the genome is masked, <font color="blue">use the <strong>–softmasking flag</strong> of <strong>braker.pl</strong></font>.</li>
<li>Many genomes have gene structures that will be predicted accurately with <font color="blue">standard parameters of GeneMark-EX and AUGUSTUS within BRAKER</font>. However, some genomes have clade-specific features, i.e. special branch point model in fungi, or non-standard splice-site patterns. Please read the options section [options] in order to determine whether any of the custom options may improve gene prediction accuracy in the genome of your target species. <strong>大部分genome使用默认参数结果不错。但仍需查看是否有参数可以提高特定物种的预测。</strong></li>
<li><font color="blue">Always check gene prediction results before further usage!</font> You can e.g. use a genome browser for visual inspection of gene models in context with extrinsic evidence data. BRAKER supports the generation of track data hubs for the UCSC Genome Browser with MakeHub for this purpose.</li>
</ul>
<h1 id="Overview-of-modes-for-running-BRAKER"><a href="#Overview-of-modes-for-running-BRAKER" class="headerlink" title="Overview of modes for running BRAKER"></a>Overview of modes for running BRAKER</h1><p>BRAKER mainly features semi-unsupervised, extrinsic evidence data (RNA-Seq and&#x2F;or protein spliced alignment information) supported <strong>training of GeneMark-EX</strong>[F1] and <strong>subsequent training of AUGUSTUS</strong> with integration of extrinsic evidence in the final gene prediction step. However, there are now a number of additional pipelines included in BRAKER. In the following, we give an overview of <font color="blue">possible input files and pipelines</font>:</p>
<ul>
<li><font color="red"><strong>Genome file, only</strong></font>. In this mode, <strong>GeneMark-ES</strong> is <strong>trained</strong> on the genome sequence, <strong>alone</strong>. <strong>Long genes predicted by GeneMark-ES</strong> are <font color="blue"><strong>selected</strong></font> for <strong>training AUGUSTUS</strong>. Final predictions by AUGUSTUS are <em>ab initio</em>. This approach will likely yield lower prediction accuracy than all other here described pipelines. (see Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig1">2</a>)</li>
</ul>
<img src="BRAKER/f2.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<p>Figure 2: BRAKER pipeline A: training GeneMark-ES on genome data, only; <em>ab initio</em> gene prediction withAUGUSTUS. 只用GeneMark-ES预测，选取较长预测结果训练AUGUSTUS</p>
<ul>
<li><font color="red"><strong>Genome</strong> and <strong>RNA-Seq</strong> file from the <strong>same</strong> species</font>; this approach is <strong>suitable for</strong> short read RNA-Seq libraries with a <strong>good coverage</strong> of the transcriptome, <strong>important:</strong> this approach <strong>requires</strong> that each <strong>intron</strong> is <strong>covered by many align</strong>ments, i.e. it does not work with assembled transcriptome mappings. In principle, also alignments of long read RNA-Seq data may lead to sufficient data for running BRAKER, but <strong>only if each transcript that will go into training was sequenced and aligned to the genome multiple times.</strong> Please be aware that at the current point in time, BRAKER <font color="blue">does <strong>not officially support</strong> the integration of <strong>long read RNA-Seq</strong></font> data, yet.</li>
</ul>
<img src="BRAKER/f3.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<p>Figure 3: BRAKER pipeline B: training GeneMark-ET supported by RNA-Seq spliced alignment information, prediction with AUGUSTUS with that same spliced alignment information.</p>
<ul>
<li><font color="red"><strong>Genome</strong> file and database of <strong>proteins</strong> that may be of <strong>unknown</strong> evolutionary distance to the target species</font> (see Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig3">4</a>); this approach is particularly suitable if no RNA-Seq data is available. This method will work better with proteins from species that are rather close to the target species, but <strong>accuracy will drop only very little if the reference proteins are more distant from the target species</strong>. <strong>Important:</strong> This approach requires a database of protein families, i.e. many representatives of each protein family must be present in the database. BRAKER has been tested with OrthoDB <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f19">R19</a>, successfully. <font color="red">The ProtHint <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f18">R18</a> <strong>protein mapping pipeline</strong></font> for generating required hints for BRAKER is available for <strong>download</strong> at <a target="_blank" rel="noopener" href="https://github.com/gatech-genemark/ProtHint">https://github.com/gatech-genemark/ProtHint</a>, the instructions on <font color="red"><strong>how to prepare</strong> the OrthoDB input proteins</font> are documented at <a target="_blank" rel="noopener" href="https://github.com/gatech-genemark/ProtHint#protein-database-preparation">https://github.com/gatech-genemark/ProtHint#protein-database-preparation</a>. You may add proteins of a closely related species to the OrthoDB fasta file in order to incorporate additional evidence into gene prediction.</li>
</ul>
<img src="BRAKER/f4.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<p>Figure 4: <font color="red"><strong>BRAKER pipeline C</strong>: <strong>training GeneMark-EP+</strong> on <strong>protein spliced alignment</strong>, <strong>start and stop information</strong>, prediction with <strong>AUGUSTUS</strong> with that <strong>same information</strong>, in addition <strong>chained CDSpart hints</strong>. Proteins used here can be of any evolutionary distance to the target organism.</font></p>
<ul>
<li><font color="red"><strong>Genome</strong> and <strong>RNA</strong>-Seq file from the same species, and <strong>proteins</strong> that may be of <strong>unknown</strong> evolutionary distance to the target species</font> (see figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig4">5</a>); <strong>important:</strong> this approach requires a database of protein families, i.e. many representatives of each protein family must be present in the database, e.g. OrthoDB is suitable. (You may add proteins of a closely related species to the OrthoDB fasta file in order to incorporate additional evidence into gene prediction.)</li>
</ul>
<img src="BRAKER/f5.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<p>Figure 5: <font color="red"><strong>BRAKER pipeline D</strong>: <strong>training</strong> <strong>GeneMark-ETP+</strong> supported by <strong>RNA</strong>-Seq alignment information and information from <strong>proteins</strong></font> (proteins can be of any evolutionary distance). Please be aware that GeneMark-ETP+ is still under development, BRAKER can currently execute a precursor of the mature version. Introns supported by both RNA-Seq and protein alignment information are treated as “true positive introns”, their prediction in gene structures by GeneMark-ETP+ and AUGUSTUS is enforced. <strong>Important:</strong> It is not always best to use all evidence! So far, <font color="blue">we found this approach to work well for <strong>large genomes</strong>, but accuracy on <strong>small and medium sized genomes</strong> is <strong>unstable</strong></font>. Please have a look at the poster from PAG 2020 before running this pipeline.</p>
<ul>
<li><font color="red"><strong>Genome</strong> file and file with <strong>proteins</strong> of <strong>short evolutionary distance</strong></font> (see Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig5">6</a>); this approach is suitable if RNA-Seq data is not available and if the reference species is very closely related. <em>Note:</em> This pipeline is deprecated since pipeline C can also use proteins of closely related species in addition to OrthoDB.</li>
</ul>
<img src="BRAKER/f6.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<p>Figure 6: Additional pipeline B: training AUGUSTUS on the basis of spliced alignment information from <font color="red">proteins of a very closely related species</font> against the target genome.</p>
<ul>
<li><font color="red"><strong>Genome</strong> and <strong>RNA</strong>-Seq file and <strong>proteins</strong> of <strong>short evolutionary distance</strong></font> (see Figures <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig6">6</a> and <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig7">7</a>). In both cases, GeneMark-ET is trained supported by RNA-Seq data, and the resulting gene predictions are used for training AUGUSTUS. In approach A), protein alignment information is used in the gene prediction step with AUGUSTUS, only. In approach C), protein spliced alignment data is used to complement the training set for AUGUSTUS. The latter approach is in particular suitable if RNA-Seq data does not produce a sufficiently high number of training gene structures for AUGUSTUS, and if a very closely related and already annotated species is available. <em>Note:</em> This pipeline is deprecated since pipeline D can also use proteins of closely related species in addition to OrthoDB.</li>
</ul>
<img src="BRAKER/f7.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<p>Figure 7: Additional pipeline A: training GeneMark-ET supported by RNA-Seq spliced alignment information, prediction with AUGUSTUS with spliced alignment information from RNA-Seq data and with gene features determined by alignments from proteins of a very closely related species against the target genome. <em>Note:</em> This pipeline is deprecated since pipeline C can also use proteins of closely related species in addition to OrthoDB.</p>
<img src="BRAKER/f8.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<p>Figure 8: Additional pipeline C: training GeneMark-ET on the basis of RNA-Seq spliced alignment information, training AUGUSTUS on a set of training gene structures compiled from RNA-Seq supported gene structures predicted by GeneMark-ET and spliced alignment of proteins of a very closely related species. <em>Note:</em> This pipeline is deprecated since pipeline D can also use proteins of closely related species in addition to OrthoDB.</p>
<h1 id="Installation"><a href="#Installation" class="headerlink" title="Installation"></a>Installation</h1><h2 id="Supported-software-versions"><a href="#Supported-software-versions" class="headerlink" title="Supported software versions"></a>Supported software versions</h2><p>At the time of release, this BRAKER version was tested with:</p>
<ul>
<li>AUGUSTUS 3.4.0 <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#g2">F2</a></li>
<li>GeneMark-ES&#x2F;ET&#x2F;EP 4.64_lic</li>
<li>BAMTOOLS 2.5.1<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f5">R5</a></li>
<li>SAMTOOLS 1.7-4-g93586ed<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f6">R6</a></li>
<li>ProtHint 2.6.0</li>
<li>GenomeThreader 1.7.0<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f7">R7</a></li>
<li>Spaln 2.3.3d <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f8">R8, </a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f9">R9, </a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f10">R10, </a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#g3">F3</a></li>
<li>(Exonerate 2.2.0 <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f11">R11</a>)[F3]</li>
<li>NCBI BLAST+ 2.2.31+ <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f12">R12, </a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f13">R13</a></li>
<li>DIAMOND 0.9.24</li>
<li>cdbfasta 0.99</li>
<li>cdbyank 0.981</li>
<li>GUSHR 1.0.0</li>
</ul>
<h2 id="BRAKER"><a href="#BRAKER" class="headerlink" title="BRAKER"></a>BRAKER</h2><h3 id="Perl-pipeline-dependencies"><a href="#Perl-pipeline-dependencies" class="headerlink" title="Perl pipeline dependencies"></a>Perl pipeline dependencies</h3><p>Running BRAKER requires a Linux-system with <code>bash</code> and Perl. Furthermore, BRAKER requires the following CPAN-Perl modules to be installed:</p>
<ul>
<li><code>File::Spec::Functions</code></li>
<li><code>Hash::Merge</code></li>
<li><code>List::Util</code></li>
<li><code>MCE::Mutex</code></li>
<li><code>Module::Load::Conditional</code></li>
<li><code>Parallel::ForkManager</code></li>
<li><code>POSIX</code></li>
<li><code>Scalar::Util::Numeric</code></li>
<li><code>YAML</code></li>
<li><code>Math::Utils</code></li>
<li><code>File::HomeDir</code></li>
</ul>
<p>For ProtHint, used when protein input is supplied, also install:</p>
<ul>
<li><code>threads</code></li>
</ul>
<p>On Ubuntu, for example, install the modules with CPANminus<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#g4">F4</a>: <code>sudo cpanm Module::Name</code>, e.g. <code>sudo cpanm Hash::Merge</code>.</p>
<p>BRAKER also uses a Perl module <code>helpMod.pm</code> that is not available on CPAN. This module is part of the BRAKER release and does not require separate installation.</p>
<p>If you do not have root permissions on the Linux machine, try setting up an <strong>Anaconda</strong> (<a target="_blank" rel="noopener" href="https://www.anaconda.com/distribution/">https://www.anaconda.com/distribution/</a>) environment as follows:</p>
<figure class="highlight cmake"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><code class="hljs cmake">wget https://repo.anaconda.com/archive/Anaconda3-<span class="hljs-number">2018.12</span>-Linux-x86_64.sh<br>bash bin/Anaconda3-<span class="hljs-number">2018.12</span>-Linux-x86_64.sh <span class="hljs-comment"># do not install VS (needs root privileges)</span><br>conda <span class="hljs-keyword">install</span> -c anaconda perl<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-app-cpanminus<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-hash-merge<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-parallel-forkmanager<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-scalar-util-numeric<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-yaml<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-class-data-inheritable<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-exception-class<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-<span class="hljs-keyword">test</span>-pod<br>conda <span class="hljs-keyword">install</span> -c anaconda biopython<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-<span class="hljs-keyword">file</span>-which <span class="hljs-comment"># skip if you are not comparing to reference annotation</span><br>conda <span class="hljs-keyword">install</span> -c bioconda perl-mce<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-threaded<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-<span class="hljs-keyword">list</span>-util<br>conda <span class="hljs-keyword">install</span> -c bioconda perl-<span class="hljs-keyword">math</span>-utils<br>conda <span class="hljs-keyword">install</span> -c bioconda cdbtools<br></code></pre></td></tr></table></figure>

<p>Subsequently install BRAKER and other software “as usual” while being in your conda environment. <strong>Note:</strong> <strong>There is a bioconda braker package, and a bioconda augustus package</strong>. They work. But <strong>they are usually lagging behind the development code of both tools on github</strong>. We therefore recommend manual installation and usage of lastest sources.</p>
<h3 id="BRAKER-components"><a href="#BRAKER-components" class="headerlink" title="BRAKER components"></a>BRAKER components</h3><p>BRAKER is a collection of Perl and Python scripts and a Perl module. The main script that will be called in order to run BRAKER is <code>braker.pl</code>. Additional Perl and Python components are:</p>
<ul>
<li><code>align2hints.pl</code></li>
<li><code>filterGenemark.pl</code></li>
<li><code>filterIntronsFindStrand.pl</code></li>
<li><code>startAlign.pl</code></li>
<li><code>helpMod.pm</code></li>
<li><code>findGenesInIntrons.pl</code></li>
<li><code>downsample_traingenes.pl</code></li>
<li><code>ensure_n_training_genes.py</code></li>
</ul>
<p>All scripts (files ending with <code>*.pl</code> and <code>*.py</code>) that are part of BRAKER must be executable in order to run BRAKER. This should already be the case if you download BRAKER from GitHub. Executability may be overwritten if you e.g. transfer BRAKER on a USB-stick to another computer. In order to check whether required files are executable, run the following command in the directory that contains BRAKER Perl scripts:</p>
<figure class="highlight jboss-cli"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs jboss-cli"><span class="hljs-keyword">ls</span> -l *<span class="hljs-string">.pl</span> *<span class="hljs-string">.py</span><br></code></pre></td></tr></table></figure>

<p>The output should be similar to this:</p>
<figure class="highlight tap"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><code class="hljs tap">-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina <span class="hljs-number"> 18191 </span>Mai <span class="hljs-number"> 7 </span>10:25 align2hints.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina  <span class="hljs-number"> 6090 </span>Feb<span class="hljs-number"> 19 </span>09:35 braker_cleanup.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina<span class="hljs-number"> 408782 </span>Aug<span class="hljs-number"> 17 </span>18:24 braker.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina  <span class="hljs-number"> 5024 </span>Mai <span class="hljs-number"> 7 </span>10:25 downsample_traingenes.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina  <span class="hljs-number"> 5024 </span>Mai <span class="hljs-number"> 7 </span>10:23 ensure_n_training_genes.py<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina  <span class="hljs-number"> 4542 </span>Apr <span class="hljs-number"> 3 </span><span class="hljs-number"> 2019 </span>filter_augustus_gff.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina <span class="hljs-number"> 30453 </span>Mai <span class="hljs-number"> 7 </span>10:25 filterGenemark.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina  <span class="hljs-number"> 5754 </span>Mai <span class="hljs-number"> 7 </span>10:25 filterIntronsFindStrand.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina  <span class="hljs-number"> 7765 </span>Mai <span class="hljs-number"> 7 </span>10:25 findGenesInIntrons.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina  <span class="hljs-number"> 1664 </span>Feb<span class="hljs-number"> 12 </span><span class="hljs-number"> 2019 </span>gatech_pmp2hints.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina  <span class="hljs-number"> 2250 </span>Jan <span class="hljs-number"> 9 </span>13:55 log_reg_prothints.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina  <span class="hljs-number"> 4679 </span>Jan <span class="hljs-number"> 9 </span>13:55 merge_transcript_sets.pl<br>-rwxr-xr-x<span class="hljs-number"> 1 </span>katharina katharina <span class="hljs-number"> 41674 </span>Mai <span class="hljs-number"> 7 </span>10:25 startAlign.pl<br></code></pre></td></tr></table></figure>

<p>It is important that the <code>x</code> in <code>-rwxr-xr-x</code> is present for each script. If that is not the case, run</p>
<figure class="highlight jboss-cli"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs jboss-cli">`chmod a+x *<span class="hljs-string">.pl</span> *<span class="hljs-string">.py</span>`<br></code></pre></td></tr></table></figure>

<p>in order to change file attributes.</p>
<p>You may find it helpful to add the directory in which BRAKER perl scripts reside to your <code>$PATH</code> environment variable. For a single bash session, enter:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs routeros"><span class="hljs-attribute">PATH</span>=/your_path_to_braker/:$PATH<br><span class="hljs-built_in">export</span> PATH<br></code></pre></td></tr></table></figure>

<p>To make this <code>$PATH</code> modification available to all bash sessions, add the above lines to a startup script (e.g.<code>~/.bashrc</code>).</p>
<h2 id="Bioinformatics-software-dependencies"><a href="#Bioinformatics-software-dependencies" class="headerlink" title="Bioinformatics software dependencies"></a>Bioinformatics software dependencies</h2><p>BRAKER calls upon various bioinformatics software tools that are not part of BRAKER. Some tools are obligatory, i.e. BRAKER will not run at all if these tools are not present on your system. Other tools are optional. Please install all tools that are required for running BRAKER in the mode of your choice.</p>
<h3 id="Mandatory-tools"><a href="#Mandatory-tools" class="headerlink" title="Mandatory tools"></a>Mandatory tools</h3><h4 id="GeneMark-EX"><a href="#GeneMark-EX" class="headerlink" title="GeneMark-EX"></a>GeneMark-EX</h4><p>Download GeneMark-EX<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#g1">F1</a> from <a target="_blank" rel="noopener" href="http://exon.gatech.edu/GeneMark/license_download.cgi">http://exon.gatech.edu/GeneMark/license_download.cgi</a> (the GeneMark-ES&#x2F;ET&#x2F;EP) option. Unpack and install GeneMark-EX as described in GeneMark-EX’s <code>README</code> file.</p>
<p>If already contained in your <code>$PATH</code> variable, BRAKER will guess the location of <code>gmes_petap.pl</code>, automatically. Otherwise, BRAKER can find GeneMark-EX executables either by locating them in an environment variable <code>GENEMARK_PATH</code>, or by taking a command line argument (<code>--GENEMARK_PATH=/your_path_to_GeneMark-EX/</code>).</p>
<p>In order to set the environment variable for your current Bash session, type:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs routeros"><span class="hljs-built_in">export</span> <span class="hljs-attribute">GENEMARK_PATH</span>=/your_path_to_GeneMark-EX/<br></code></pre></td></tr></table></figure>

<p>Add the above lines to a startup script (e.g. <code>~/.bashrc</code>) in order to make it available to all bash sessions.<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#g5">F5</a></p>
<p><strong>Important:</strong> GeneMark-EX will only run if a valid key file resides in your home directory. The key file will expire after 200 days, which means that you have to download a new GeneMark-EX release and a new key file after 200 days. The key file is downloaded as <code>gm_key.gz</code>. Unpack the key file and move it to a hidden file <strong>in your home directory</strong> as follows:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">cd</span> <span class="hljs-comment"># change to your home directory</span><br>gunzip gm_key_64.gz<br><span class="hljs-built_in">mv</span> gm_key_64 .gm_key<br></code></pre></td></tr></table></figure>

<p>Perl scripts within GeneMark-EX are configured with default Perl location at <code>/usr/bin/perl</code>.</p>
<p>If you are running GeneMark-EX in an Anaconda environment (or want to use Perl from the <code>$PATH</code> variable for any other reason), modify the shebang of all GeneMark-EX scripts with the following command located inside GeneMark-EX folder:</p>
<figure class="highlight nginx"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs nginx"><span class="hljs-attribute">perl</span> change_path_in_perl_scripts.pl <span class="hljs-string">&quot;/usr/bin/env perl&quot;</span><br></code></pre></td></tr></table></figure>

<p>You can check whether GeneMark-EX is installed properly by running the <code>check_install.bash</code> and&#x2F;or executing examples in <code>GeneMark-E-tests</code> directory.</p>
<figure class="highlight pgsql"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><code class="hljs pgsql">my<br>./gmes_linux_64/check_install.bash<br>Checking GeneMark-ES installation<br>Checking Perl setup<br><span class="hljs-keyword">All</span> required Perl modules were <span class="hljs-built_in">found</span><br>Checking GeneMark.hmm setup<br>GeneMark.hmm was <span class="hljs-built_in">found</span><br>GeneMark.hmm <span class="hljs-keyword">is</span> <span class="hljs-keyword">set</span><br>GeneMark.hmm <span class="hljs-keyword">is</span> executable<br>Performing GeneMark.hmm test run<br><span class="hljs-keyword">All</span> required components <span class="hljs-keyword">for</span> GeneMark-ES were <span class="hljs-built_in">found</span><br></code></pre></td></tr></table></figure>



<h4 id="AUGUSTUS"><a href="#AUGUSTUS" class="headerlink" title="AUGUSTUS"></a>AUGUSTUS</h4><p>Download AUGUSTUS from its master branch at <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/Augustus">https://github.com/Gaius-Augustus/Augustus</a>. Unpack AUGUSTUS and install AUGUSTUS according to AUGUSTUS <code>README.TXT</code>. <em><strong>Do not use outdated AUGUSTUS versions from other sources, e.g. Debian package or Bioconda package! BRAKER highly depends in particular on an up-to-date Augustus&#x2F;scripts directory, and other sources are often lagging behind.</strong></em></p>
<p>You should compile AUGUSTUS on your own system in order to avoid problems with versions of libraries used by AUGUSTUS. Compilation instructions are provided in the AUGUSTUS <code>README.TXT</code> file (<code>Augustus/README.txt</code>).</p>
<p>AUGUSTUS consists of <code>augustus</code>, the gene prediction tool, additional C++ tools located in <code>Augustus/auxprogs</code> and Perl scripts located in <code>Augustus/scripts</code>. Perl scripts must be executable (see instructions in section <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#executability">BRAKER components</a>.</p>
<p>The C++ tool <code>bam2hints</code> is an essential component of BRAKER when run with RNA-Seq. Sources are located in <code>Augustus/auxprogs/bam2hints</code>. Make sure that you compile <code>bam2hints</code> on your system (it should be automatically compiled when AUGUSTUS is compiled, but in case of problems with <code>bam2hints</code>, please read troubleshooting instructions in <code>Augustus/auxprogs/bam2hints/README</code>).</p>
<p>Since BRAKER is a pipeline that trains AUGUSTUS, i.e. writes species specific parameter files, BRAKER needs writing access to the configuration directory of AUGUSTUS that contains such files (<code>Augustus/config/</code>). If you install AUGUSTUS globally on your system, the <code>config</code> folder will typically not be writable by all users. Either make the directory where <code>config</code> resides recursively writable to users of AUGUSTUS, or copy the <code>config/</code> folder (recursively) to a location where users have writing permission.</p>
<p>AUGUSTUS will locate the <code>config</code> folder by looking for an environment variable <code>$AUGUSTUS_CONFIG_PATH</code>. If the <code>$AUGUSTUS_CONFIG_PATH</code> environment variable is not set, then BRAKER will look in the path <code>../config</code> relative to the directory in which it finds an AUGUSTUS executable. Alternatively, you can supply the variable as a command line argument to BRAKER (<code>--AUGUSTUS_CONFIG_PATH=/your_path_to_AUGUSTUS/Augustus/config/</code>). We recommend that you export the variable e.g. for your current bash session:</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs awk">export AUGUSTUS_CONFIG_PATH=<span class="hljs-regexp">/your_path_to_AUGUSTUS/</span>Augustus<span class="hljs-regexp">/config/</span><br></code></pre></td></tr></table></figure>

<p>In order to make the variable available to all Bash sessions, add the above line to a startup script, e.g. <code>~/.bashrc</code>.</p>
<h5 id="Important"><a href="#Important" class="headerlink" title="Important:"></a>Important:</h5><p>BRAKER expects the entire <code>config</code> directory of AUGUSTUS at <code>$AUGUSTUS_CONFIG_PATH</code>, i.e. the subfolders <code>species</code> with its contents (at least <code>generic</code>) and <code>extrinsic</code>! Providing a writable but empty folder at <code>$AUGUSTUS_CONFIG_PATH</code> will not work for BRAKER. If you need to separate augustus binary and <code>$AUGUSTUS_CONFIG_PATH</code>, we recommend that you recursively copy the un-writable config contents to a writable location.</p>
<p>If you have a system-wide installation of AUGUSTUS at <code>/usr/bin/augustus</code>, an unwritable copy of <code>config</code> sits at <code>/usr/bin/augustus_config/</code>. The folder <code>/home/yours/</code> is writable to you. Copy with the following command (and additionally set the then required variables):</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs awk">cp -r <span class="hljs-regexp">/usr/</span>bin<span class="hljs-regexp">/Augustus/</span>config<span class="hljs-regexp">/ /</span>home<span class="hljs-regexp">/yours/</span><br>export AUGUSTUS_CONFIG_PATH=<span class="hljs-regexp">/home/y</span>ours/augustus_config<br>export AUGUSTUS_BIN_PATH=<span class="hljs-regexp">/usr/</span>bin<br>export AUGUSTUS_SCRIPTS_PATH=<span class="hljs-regexp">/usr/</span>bin/augustus_scripts<br></code></pre></td></tr></table></figure>

<h5 id="Modification-of-PATH"><a href="#Modification-of-PATH" class="headerlink" title="Modification of $PATH"></a>Modification of $PATH</h5><p>Adding directories of AUGUSTUS binaries and scripts to your <code>$PATH</code> variable enables your system to locate these tools, automatically. It is not a requirement for running BRAKER to do this, because BRAKER will try to guess them from the location of another environment variable (<code>$AUGUSTUS_CONFIG_PATH</code>), or both directories can be supplied as command line arguments to <code>braker.pl</code>, but we recommend to add them to your <code>$PATH</code> variable. For your current bash session, type:</p>
<figure class="highlight elixir"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs elixir"><span class="hljs-title class_">PATH</span>=<span class="hljs-symbol">:/your_path_to_augustus/bin/</span><span class="hljs-symbol">:/your_path_to_augustus/scripts/</span><span class="hljs-symbol">:</span><span class="hljs-variable">$PATH</span><br>export <span class="hljs-title class_">PATH</span><br></code></pre></td></tr></table></figure>

<p>For all your BASH sessions, add the above lines to a startup script (e.g.<code>~/.bashrc</code>).</p>
<h4 id="Python3"><a href="#Python3" class="headerlink" title="Python3"></a>Python3</h4><p>On Ubuntu, Python3 is usually installed by default, <code>python3</code> will be in your <code>$PATH</code> variable, by default, and BRAKER will automatically locate it. However, you have the option to specify the <code>python3</code> binary location in two other ways:</p>
<ol>
<li><p>Export an environment variable <code>$PYTHON3_PATH</code>, e.g. in your <code>~/.bashrc</code> file:</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs awk">export PYTHON3_PATH=<span class="hljs-regexp">/path/</span>to<span class="hljs-regexp">/python3/</span><br></code></pre></td></tr></table></figure>
</li>
<li><p>Specify the command line option <code>--PYTHON3_PATH=/path/to/python3/</code> to <code>braker.pl</code>.</p>
</li>
</ol>
<h4 id="Bamtools"><a href="#Bamtools" class="headerlink" title="Bamtools"></a>Bamtools</h4><p>Download BAMTOOLS (e.g. <code>git clone https://github.com/pezmaster31/bamtools.git</code>). Install BAMTOOLS by typing the following in your shell:</p>
<figure class="highlight vim"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs vim"><span class="hljs-keyword">cd</span> your-bamtools-directory <span class="hljs-built_in">mkdir</span> build <span class="hljs-keyword">cd</span> build cmake .. <span class="hljs-keyword">make</span><br></code></pre></td></tr></table></figure>

<p>If already in your <code>$PATH</code> variable, BRAKER will find bamtools, automatically. Otherwise, BRAKER can locate the bamtools binary either by using an environment variable <code>$BAMTOOLS_PATH</code>, or by taking a command line argument (<code>--BAMTOOLS_PATH=/your_path_to_bamtools/bin/</code><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#g6">F6</a>). In order to set the environment variable e.g. for your current bash session, type:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs routeros"><span class="hljs-built_in">export</span> <span class="hljs-attribute">BAMTOOLS_PATH</span>=/your_path_to_bamtools/bin/<br></code></pre></td></tr></table></figure>

<p>Add the above line to a startup script (e.g. <code>~/.bashrc</code>) in order to set the environment variable for all bash sessions.</p>
<h4 id="NCBI-BLAST-or-DIAMOND"><a href="#NCBI-BLAST-or-DIAMOND" class="headerlink" title="NCBI BLAST+ or DIAMOND"></a>NCBI BLAST+ or DIAMOND</h4><p>You can use either NCBI BLAST+ or DIAMOND for removal of redundant training genes. You do not need both tools. If DIAMOND is present, it will be preferred because it is much faster.</p>
<p>Obtain and unpack DIAMOND as follows:</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs awk">wget http:<span class="hljs-regexp">//gi</span>thub.com<span class="hljs-regexp">/bbuchfink/</span>diamond<span class="hljs-regexp">/releases/</span>download<span class="hljs-regexp">/v0.9.24/</span>diamond-linux64.tar.gz<br>tar xzf diamond-linux64.tar.gz<br></code></pre></td></tr></table></figure>

<p>If already in your <code>$PATH</code> variable, BRAKER will find diamond, automatically. Otherwise, BRAKER can locate the diamond binary either by using an environment variable <code>$DIAMOND_PATH</code>, or by taking a command line argument (<code>--DIAMOND_PATH=/your_path_to_diamond</code>). In order to set the environment variable e.g. for your current bash session, type:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs routeros"><span class="hljs-built_in">export</span> <span class="hljs-attribute">DIAMOND_PATH</span>=/your_path_to_diamond/<br></code></pre></td></tr></table></figure>

<p>Add the above line to a startup script (e.g. <code>~/.bashrc</code>) in order to set the environment variable for all bash sessions.</p>
<p>If you decide for BLAST+, install NCBI BLAST+ with <code>sudo apt-get install ncbi-blast+</code>.</p>
<p>If already in your <code>$PATH</code> variable, BRAKER will find blastp, automatically. Otherwise, BRAKER can locate the blastp binary either by using an environment variable <code>$BLAST_PATH</code>, or by taking a command line argument (<code>--BLAST_PATH=/your_path_to_blast/</code>). In order to set the environment variable e.g. for your current bash session, type:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs routeros"><span class="hljs-built_in">export</span> <span class="hljs-attribute">BLAST_PATH</span>=/your_path_to_blast/<br></code></pre></td></tr></table></figure>

<p>Add the above line to a startup script (e.g. <code>~/.bashrc</code>) in order to set the environment variable for all bash sessions.</p>
<h4 id="ProtHint"><a href="#ProtHint" class="headerlink" title="ProtHint"></a>ProtHint</h4><p>ProtHint is a pipeline for generating hints for GeneMark-EX and AUGUSTUS from proteins of any evolutionary distance. If protein sequences are given on input, BRAKER runs ProtHint automatically. Alternatively, ProtHint can be executed as a separate step during data preparation. ProtHint is available from <a target="_blank" rel="noopener" href="https://github.com/gatech-genemark/ProtHint">https://github.com/gatech-genemark/ProtHint</a>. Download as follows:</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs awk">git clone https:<span class="hljs-regexp">//gi</span>thub.com<span class="hljs-regexp">/gatech-genemark/</span>ProtHint.git<br></code></pre></td></tr></table></figure>

<p>or by getting the latest release from <a target="_blank" rel="noopener" href="https://github.com/gatech-genemark/ProtHint/releases">https://github.com/gatech-genemark/ProtHint/releases</a>.</p>
<p>ProtHint has software requirements of its own. In addition to the Perl modules required by BRAKER, it needs</p>
<figure class="highlight ebnf"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs ebnf"><span class="hljs-attribute">threads</span><br></code></pre></td></tr></table></figure>

<p>You can easily verify ProtHint’s installation by running the test in <a target="_blank" rel="noopener" href="https://github.com/gatech-genemark/ProtHint/tree/master/example">https://github.com/gatech-genemark/ProtHint/tree/master/example</a>.</p>
<p>ProtHint requires DIAMOND and Spaln, both of which come with ProtHint’s installation. ProtHint’s requirement of GeneMark-ES will already be fulfilled if you installed BRAKER dependencies above. For further installation instructions, please check <a target="_blank" rel="noopener" href="https://github.com/gatech-genemark/ProtHint">https://github.com/gatech-genemark/ProtHint</a>.</p>
<p>If already in your <code>$PATH</code> variable, BRAKER will find prothint.py, automatically. Otherwise, BRAKER will try to locate the prothint.py executable by using an environment variable <code>$PROTHINT_PATH</code>. Alternatively, this can be supplied as a command line argument <code>--PROTHINT_PATH=/your/path/to/ProtHint/bin</code>.</p>
<h3 id="Optional-tools"><a href="#Optional-tools" class="headerlink" title="Optional tools"></a>Optional tools</h3><h4 id="Samtools"><a href="#Samtools" class="headerlink" title="Samtools"></a>Samtools</h4><p>Samtools is not required for running BRAKER if all your files are formatted, correctly (i.e. all sequences should have short and unique fasta names). If you are not sure whether all your files are fomatted correctly, it might be helpful to have Samtools installed because BRAKER can automatically fix certain format issues by using Samtools.</p>
<p>As a prerequisite for Samtools, download and install <code>htslib</code> (e.g. <code>git clone https://github.com/samtools/htslib.git</code>, follow the <code>htslib</code> documentation for installation).</p>
<p>Download and install Samtools (e.g. <code>git clone git://github.com/samtools/samtools.git</code>), subsequently follow Samtools documentation for installation).</p>
<p>If already in your <code>$PATH</code> variable, BRAKER will find samtools, automatically. Otherwise, BRAKER can find Samtools either by taking a command line argument (<code>--SAMTOOLS_PATH=/your_path_to_samtools/</code>), or by using an environment variable <code>$SAMTOOLS_PATH</code>. For exporting the variable, e.g. for your current bash session, type:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs routeros"><span class="hljs-built_in">export</span> <span class="hljs-attribute">SAMTOOLS_PATH</span>=/your_path_to_samtools/<br></code></pre></td></tr></table></figure>

<p>Add the above line to a startup script (e.g. <code>~/.bashrc</code>) in order to set the environment variable for all bash sessions.</p>
<h4 id="Biopython"><a href="#Biopython" class="headerlink" title="Biopython"></a>Biopython</h4><p>If Biopython is installed, BRAKER can generate FASTA-files with coding sequences and protein sequences predicted by AUGUSTUS and generate track data hubs for visualization of a BRAKER run with MakeHub <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f16">R16</a>. These are optional steps. The first can be disabled with the command-line flag <code>--skipGetAnnoFromFasta</code>, the second can be activated by using the command-line options <code>--makehub --email=your@mail.de</code>, Biopython is not required if neither of these optional steps shall be performed.</p>
<p>On Ubuntu, install Python3 package manager with:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs routeros">`sudo apt-<span class="hljs-built_in">get</span> install python3-pip`<br></code></pre></td></tr></table></figure>

<p>Then, install Biopython with:</p>
<figure class="highlight arcade"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs arcade"><span class="hljs-string">`sudo pip3 install biopython`</span><br></code></pre></td></tr></table></figure>

<h4 id="cdbfasta"><a href="#cdbfasta" class="headerlink" title="cdbfasta"></a>cdbfasta</h4><p>cdbfasta and cdbyank are required by BRAKER for correcting AUGUSTUS genes with in frame stop codons (spliced stop codons) using the AUGUSTUS script fix_in_frame_stop_codon_genes.py. This can be skipped with <code>--skip_fixing_broken_genes</code>.</p>
<p>On Ubuntu, install cdbfasta with:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs routeros">`sudo apt-<span class="hljs-built_in">get</span> install cdbfasta`<br></code></pre></td></tr></table></figure>

<p>For other systems, you can for example obtain cdbfasta from <a target="_blank" rel="noopener" href="https://github.com/gpertea/cdbfasta">https://github.com/gpertea/cdbfasta</a>, e.g.:</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs awk">git clone https:<span class="hljs-regexp">//gi</span>thub.com<span class="hljs-regexp">/gpertea/</span>cdbfasta.git`<br>cd cdbfasta<br>make all<br></code></pre></td></tr></table></figure>

<p>On Ubuntu, cdbfasta and cdbyank will be in your <code>$PATH</code> variable after installation, and BRAKER will automatically locate them. However, you have the option to specify the <code>cdbfasta</code> and <code>cdbyank</code> binary location in two other ways:</p>
<ol>
<li>Export an environment variable <code>$CDBTOOLS_PATH</code>, e.g. in your <code>~/.bashrc</code> file:</li>
</ol>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs awk">export CDBTOOLS_PATH=<span class="hljs-regexp">/path/</span>to<span class="hljs-regexp">/cdbtools/</span><br></code></pre></td></tr></table></figure>

<ol>
<li>Specify the command line option <code>--CDBTOOLS_PATH=/path/to/cdbtools/</code> to <code>braker.pl</code>.</li>
</ol>
<h4 id="GenomeThreader"><a href="#GenomeThreader" class="headerlink" title="GenomeThreader"></a>GenomeThreader</h4><p><strong>Note:</strong> Support of GenomeThreader within BRAKER is deprecated.</p>
<p>This tool is required, only, if you would like to run protein to genome alignments with BRAKER using GenomeThreader. This is a suitable approach only if an annotated species of short evolutionary distance to your target genome is available. Download GenomeThreader from <a target="_blank" rel="noopener" href="http://genomethreader.org/">http://genomethreader.org/</a>. Unpack and install according to <code>gth/README</code>.</p>
<p>BRAKER will try to locate the GenomeThreader executable by using an environment variable <code>$ALIGNMENT_TOOL_PATH</code>. Alternatively, this can be supplied as command line argument (<code>--ALIGNMENT_TOOL_PATH=/your/path/to/gth</code>).</p>
<h4 id="Spaln"><a href="#Spaln" class="headerlink" title="Spaln"></a>Spaln</h4><p><strong>Note:</strong> Support of stand-alone Spaln (ouside of ProtHint) within BRAKER is deprecated.</p>
<p>This tool is required if you run ProtHint or if you would like to run protein to genome alignments with BRAKER using Spaln outside of ProtHint. Using Spaln outside of ProtHint is a suitable approach only if an annotated species of short evolutionary distance to your target genome is available. We recommend running Spaln through ProtHint for BRAKER. ProtHint brings along a Spaln binary. If that does not work on your system, download Spaln from <a target="_blank" rel="noopener" href="https://github.com/ogotoh/spaln">https://github.com/ogotoh/spaln</a>. Unpack and install according to <code>spaln/doc/SpalnReadMe22.pdf</code>.</p>
<p>BRAKER will try to locate the Spaln executable by using an environment variable <code>$ALIGNMENT_TOOL_PATH</code>. Alternatively, this can be supplied as command line argument (<code>--ALIGNMENT_TOOL_PATH=/your/path/to/spaln</code>).</p>
<h4 id="Exonerate"><a href="#Exonerate" class="headerlink" title="Exonerate"></a>Exonerate</h4><p><strong>Note:</strong> Support of Exonerate within BRAKER is deprecated.</p>
<p>This tool is required, only, if you would like to run protein to genome alignments with BRAKER using Exonerate. This is a suitable approach only if an annotated species of short evolutionary distance to your target genome is available. (We recommend the usage of GenomeThreader instad of Exonerate because Exonerate is comparably slower and has lower specificity than GenomeThreader.) Download Exonerate from <a target="_blank" rel="noopener" href="https://github.com/nathanweeks/exonerate">https://github.com/nathanweeks/exonerate</a>. Unpack and install according to <code>exonerate/README</code>. (On Ubuntu, download and install by typing <code>sudo apt-get install exonerate</code>.)</p>
<p>BRAKER will try to locate the Exonerate executable by using an environment variable <code>$ALIGNMENT_TOOL_PATH</code>. Alternatively, this can be supplied as command line argument (<code>--ALIGNMENT_TOOL_PATH=/your/path/to/exonerate</code>).</p>
<h4 id="GUSHR"><a href="#GUSHR" class="headerlink" title="GUSHR"></a><font color="red">GUSHR</font></h4><p>This tool is only required if you want either add UTRs (from RNA-Seq data) to predicted genes or if you want to train UTR parameters for AUGUSTUS and predict genes with UTRs. In any case, GUSHR requires the input of RNA-Seq data.</p>
<p>GUSHR is available for download at <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/GUSHR">https://github.com/Gaius-Augustus/GUSHR</a>. Obtain it by typing:</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs awk">git clone https:<span class="hljs-regexp">//gi</span>thub.com<span class="hljs-regexp">/Gaius-Augustus/</span>GUSHR.git<br></code></pre></td></tr></table></figure>

<p>GUSHR executes a GeMoMa jar file <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f19">R19, </a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f20">R20, </a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#f21">R21</a>, and this jar file requires Java 1.8. On Ubuntu, you can install Java 1.8 with the following command:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs routeros">sudo apt-<span class="hljs-built_in">get</span> install openjdk-8-jdk<br></code></pre></td></tr></table></figure>

<p>If you have several java versions installed on your system, make sure that you enable 1.8 prior running BRAKER with java by running</p>
<figure class="highlight mipsasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs mipsasm">sudo update-alternatives --<span class="hljs-built_in">config</span> <span class="hljs-keyword">java </span><br></code></pre></td></tr></table></figure>

<p>and selecting the correct version.</p>
<h4 id="Tools-from-UCSC"><a href="#Tools-from-UCSC" class="headerlink" title="Tools from UCSC"></a><font color="red">Tools from UCSC</font></h4><p>If you switch <code>--UTR=on</code>, bamToWig.py will require the following tools that can be downloaded from <a target="_blank" rel="noopener" href="http://hgdownload.soe.ucsc.edu/admin/exe">http://hgdownload.soe.ucsc.edu/admin/exe</a>:</p>
<ul>
<li>twoBitInfo</li>
<li>faToTwoBit</li>
</ul>
<p>It is optional to install these tools into your $PATH. If you don’t, and you switch <code>--UTR=on</code>, bamToWig.py will automatically download them into the working directory.</p>
<h4 id="MakeHub"><a href="#MakeHub" class="headerlink" title="MakeHub"></a>MakeHub</h4><p>If you wish to automaticaly generate a track data hub of your BRAKER run, the MakeHub software, available at <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/MakeHub">https://github.com/Gaius-Augustus/MakeHub</a> is required. Download the software (either by running <code>git clone https://github.com/Gaius-Augustus/MakeHub.git</code>, or by picking a release from <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/MakeHub/releases">https://github.com/Gaius-Augustus/MakeHub/releases</a>. Extract the release package if you downloaded a release (e.g. <code>unzip MakeHub.zip</code> or <code>tar -zxvf MakeHub.tar.gz</code>.</p>
<p>BRAKER will try to locate the make_hub.py script by using an environment variable <code>$MAKEHUB_PATH</code>. Alternatively, this can be supplied as command line argument (<code>--MAKEHUB_PATH=/your/path/to/MakeHub/</code>). BRAKER can also try to guess the location of MakeHub on your system.</p>
<h1 id="Running-BRAKER"><a href="#Running-BRAKER" class="headerlink" title="Running BRAKER"></a><font color="red">Running BRAKER</font></h1><h2 id="Different-BRAKER-pipeline-modes"><a href="#Different-BRAKER-pipeline-modes" class="headerlink" title="Different BRAKER pipeline modes"></a>Different BRAKER pipeline modes</h2><p>In the following, we describe “typical” BRAKER calls for different input data types. In general, we recommend that you run BRAKER <font color="blue">on genomic sequences that have been <strong>softmasked</strong> for Repeats</font>. If your genome has been softmasked, include the <code>--softmasking</code> flag in your BRAKER call!</p>
<h3 id="BRAKER-with-RNA-Seq-data"><a href="#BRAKER-with-RNA-Seq-data" class="headerlink" title="BRAKER with RNA-Seq data"></a>BRAKER with <font color="red">RNA-Seq data</font></h3><p>This approach is suitable for genomes of species for which <font color="blue">RNA-Seq libraries with a good coverage</font> of the transcriptome are available. The pipeline is illustrated in Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig1">2</a>.</p>
<p>BRAKER can either extract RNA-Seq spliced alignment information from <code>bam</code> files, or it can use such extracted information, directly.</p>
<p>In order to run BRAKER with RNA-Seq data supplied as <code>bam</code> file(s) (in case of multiple files, separate them by comma), run:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs routeros">braker.pl <span class="hljs-attribute">--species</span>=yourSpecies <span class="hljs-attribute">--genome</span>=genome.fasta \<br>   <span class="hljs-attribute">--bam</span>=file1.bam,file2.bam<br></code></pre></td></tr></table></figure>

<p>In order to run BRAKER with <font color="blue">RNA-Seq <strong>spliced alignment information</strong> that has already been extracted</font>, run:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs routeros">braker.pl <span class="hljs-attribute">--species</span>=yourSpecies <span class="hljs-attribute">--genome</span>=genome.fasta \<br>   <span class="hljs-attribute">--hints</span>=hints1.gff,hints2.gff<br></code></pre></td></tr></table></figure>

<p>The format of such a hints file must be as follows (tabulator separated file):</p>
<figure class="highlight tap"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs tap">chrName b2h intron <span class="hljs-number"> 6591 </span>  <span class="hljs-number"> 8003 </span>  <span class="hljs-number"> 1 </span>  +   .   pri=4;src=E<br>chrName b2h intron <span class="hljs-number"> 6136 </span>  <span class="hljs-number"> 9084 </span>  <span class="hljs-number"> 11 </span> +   .   mult=11;pri=4;src=E<br>...<br></code></pre></td></tr></table></figure>

<p><font color="blue">The source <strong>b2h</strong> in the <strong>second column</strong> and the source tag <strong>src&#x3D;E</strong> in the <strong>last column</strong> are <strong>essential for BRAKER to determine</strong> whether a <strong>hint</strong> has been <strong>generated from RNA-Seq</strong> data.</font></p>
<h3 id="BRAKER-with-proteins-of-any-evolutionary-distance"><a href="#BRAKER-with-proteins-of-any-evolutionary-distance" class="headerlink" title="BRAKER with proteins of any evolutionary distance"></a>BRAKER with <font color="red">proteins of any evolutionary distance</font></h3><p>This approach is <font color="blue"><strong>suitable for</strong> genomes of species for which <strong>no RNA-Seq libraries</strong></font> are available. A large database of proteins (with possibly longer evolutionary distance to the target species) should be used in this case. This mode is illustrated in figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig8">9</a>.</p>
<img src="BRAKER/f9.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<p>Figure 9: BRAKER with proteins of any evolutionary distance. ProtHint protein mapping pipelines is used to generate protein hints. <font color="blue"><strong>ProtHint</strong> automatically <strong>determines</strong> which <strong>alignments</strong> are <strong>from close relatives</strong>, and which are <strong>from rather distant relatives.</strong></font></p>
<p>For running BRAKER in this mode, type:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span>=genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--prot_seq</span>=proteins<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--softmasking</span><br></code></pre></td></tr></table></figure>

<p>We <font color="blue"><strong>recommend</strong> using OrthoDB</font> as basis for <code>proteins.fa</code>. The instructions on <font color="blue"><strong>how to prepare the input OrthoDB proteins</strong></font> are documented here: <a target="_blank" rel="noopener" href="https://github.com/gatech-genemark/ProtHint#protein-database-preparation">https://github.com/gatech-genemark/ProtHint#protein-database-preparation</a>.</p>
<p>You can of course add additional protein sequences to that file, or try with a completely different database. Any database will need several representatives for each protein, though.</p>
<p>Instead of having BRAKER run ProtHint, you <font color="blue">can also start BRAKER with hints already produced by ProtHint</font>, by providing ProtHint’s <code>prothint_augustus.gff</code> output:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span>=genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--hints</span>=prothint_augustus<span class="hljs-selector-class">.gff</span> <span class="hljs-attr">--softmasking</span><br></code></pre></td></tr></table></figure>

<p>The format of <code>prothint_augustus.gff</code> in this mode looks like this:</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">2R</span> ProtHint intron <span class="hljs-number">11506230</span> <span class="hljs-number">11506648</span> <span class="hljs-number">4</span> + . src=M;mult=<span class="hljs-number">4</span>;pri=<span class="hljs-number">4</span><br><span class="hljs-attribute">2R</span> ProtHint intron <span class="hljs-number">9563406</span>  <span class="hljs-number">9563473</span>  <span class="hljs-number">1</span> + . grp=<span class="hljs-number">69004</span>_0:<span class="hljs-number">001</span>de1_702_g;src=C;pri=<span class="hljs-number">4</span>;<br><span class="hljs-attribute">2R</span> ProtHint intron <span class="hljs-number">8446312</span>  <span class="hljs-number">8446371</span>  <span class="hljs-number">1</span> + . grp=<span class="hljs-number">43151</span>_0:<span class="hljs-number">001</span>cae_473_g;src=C;pri=<span class="hljs-number">4</span>;<br><span class="hljs-attribute">2R</span> ProtHint intron <span class="hljs-number">8011796</span>  <span class="hljs-number">8011865</span>  <span class="hljs-number">2</span> - . src=P;mult=<span class="hljs-number">1</span>;pri=<span class="hljs-number">4</span>;al_score=<span class="hljs-number">0</span>.<span class="hljs-number">12</span>;<br><span class="hljs-attribute">2R</span> ProtHint start  <span class="hljs-number">234524</span>   <span class="hljs-number">234526</span>   <span class="hljs-number">1</span> + . src=P;mult=<span class="hljs-number">1</span>;pri=<span class="hljs-number">4</span>;al_score=<span class="hljs-number">0</span>.<span class="hljs-number">08</span>;<br></code></pre></td></tr></table></figure>

<p>The prediction of all hints with <font color="blue"><strong>src&#x3D;M</strong></font> will be enforced. Hints with <font color="blue"><strong>src&#x3D;C</strong></font> are ‘chained evidence’, i.e. they will only be incorporated if all members of the group (grp&#x3D;…) can be incorporated in a single transcript. All other hints have <font color="blue"><strong>src&#x3D;P</strong></font> in the last column. Supported features in column 3 are <code>intron</code>, <code>start</code>, <code>stop</code> and <code>CDSpart</code>.</p>
<h4 id="Training-and-prediction-of-UTRs-integration-of-coverage-information"><a href="#Training-and-prediction-of-UTRs-integration-of-coverage-information" class="headerlink" title="Training and prediction of UTRs, integration of coverage information"></a>Training and prediction of <font color="red">UTRs</font>, integration of coverage information</h4><p>If RNA-Seq (and only RNA-Seq) data is provided to BRAKER as a bam-file, and <font color="blue">if the genome is <strong>softmasked for repeats</strong></font>, BRAKER can automatically train UTR parameters for AUGUSTUS. After successful training of UTR parameters, BRAKER will automatically predict genes including coverage information form RNA-Seq data. Example call:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs routeros">braker.pl <span class="hljs-attribute">--species</span>=yourSpecies <span class="hljs-attribute">--genome</span>=genome.fasta \<br>   <span class="hljs-attribute">--bam</span>=file.bam --softmasking <span class="hljs-attribute">--UTR</span>=on<br></code></pre></td></tr></table></figure>

<p><strong>Warnings:</strong></p>
<ol>
<li>This feature is experimental!</li>
<li>–UTR&#x3D;on is currently not compatible with bamToWig.py as released in AUGUSTUS 3.3.3; it requires the current development code version from the github repository (git clone <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/Augustus.git">https://github.com/Gaius-Augustus/Augustus.git</a>).</li>
<li>–UTR&#x3D;on increases memory consumption of AUGUSTUS. Carefully monitor jobs if your machine was close to maxing RAM without –UTR&#x3D;on! Reducing the number of cores will also reduce RAM consumption.</li>
<li>UTR prediction sometimes improves coding sequence prediction accuracy, but not always.<font color="red"> <strong>If you try this feature, carefully compare results with and without UTR parameters</strong>, </font>afterwards (e.g. in UCSC Genome Browser).</li>
</ol>
<h4 id="Stranded-RNA-Seq-alignments"><a href="#Stranded-RNA-Seq-alignments" class="headerlink" title="Stranded RNA-Seq alignments"></a>Stranded RNA-Seq alignments</h4><p>For running BRAKER without UTR parameters, <strong>it is not very important whether RNA-Seq data was generated by a <em>stranded</em> protocol</strong> (because spliced alignments are ’artificially stranded’ by checking the splice site pattern). <font color="red"><strong>However, for UTR training and prediction, stranded libraries may provide information that is valuable for BRAKER</strong>.</font></p>
<p>After alignment of the <strong>stranded RNA-Seq libraries</strong>, separate the resulting bam file entries into two files: <strong>one for plus strand mappings, one for minus strand mappings</strong>. Call BRAKER as follows:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs routeros">braker.pl <span class="hljs-attribute">--species</span>=yourSpecies <span class="hljs-attribute">--genome</span>=genome.fasta \<br>   --softmasking <span class="hljs-attribute">--bam</span>=plus.bam,minus.bam <span class="hljs-attribute">--stranded</span>=+,- \<br>    <span class="hljs-attribute">--UTR</span>=on<br></code></pre></td></tr></table></figure>

<p>You may additionally include bam files from <strong>unstranded libraries. Those files will not used for generating UTR training examples, but they will be included in the final gene prediction step as unstranded coverage information</strong>, example call:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--species</span>=yourSpecies <span class="hljs-attr">--genome</span>=genome<span class="hljs-selector-class">.fasta</span> \<br>   <span class="hljs-attr">--softmasking</span> <span class="hljs-attr">--bam</span>=plus<span class="hljs-selector-class">.bam</span>,minus<span class="hljs-selector-class">.bam</span>,unstranded<span class="hljs-selector-class">.bam</span> \<br>   <span class="hljs-attr">--stranded</span>=+,-,. <span class="hljs-attr">--UTR</span>=on<br></code></pre></td></tr></table></figure>

<p><strong>Warning:</strong> This feature is experimental and currently has low priority on our maintenance list!</p>
<h3 id="BRAKER-with-proteins-of-short-evolutionary-distance"><a href="#BRAKER-with-proteins-of-short-evolutionary-distance" class="headerlink" title="BRAKER with proteins of short evolutionary distance"></a>BRAKER with <font color="red">proteins of short evolutionary distance</font></h3><p>This is a deprecated pipeline that was before the only suitable approach if RNA-Seq data for the species of the target genome is not available and if a well annotated and very closely related reference species is available and you don’t want to use the approach for proteins of any evolutionary distance (where we back then weren’t sure how it’d perform on proteins of short evolutionary distance).</p>
<p>For running BRAKER in this mode, type:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs routeros">braker.pl <span class="hljs-attribute">--species</span>=yourSpecies <span class="hljs-attribute">--genome</span>=genome.fasta \<br>   <span class="hljs-attribute">--prot_seq</span>=proteins.fa <span class="hljs-attribute">--prg</span>=gth \<br>   <span class="hljs-attribute">--ALIGNMENT_TOOL_PATH</span>=/path/to/gth/binary \<br>   --trainFromGth<br></code></pre></td></tr></table></figure>

<p><font color="blue">It is possible to generate protein alignments <strong>externally</strong>,</font> prior running BRAKER, itself. The compatible command for running GenomeThreader prior running BRAKER, is:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs stylus">gth -genomic genome<span class="hljs-selector-class">.fa</span>  -protein protein<span class="hljs-selector-class">.fa</span> -gff3out \<br>   -skipalignmentout -o gth.aln<br></code></pre></td></tr></table></figure>

<p>In order to use such externally created alignment files, run: <font color="blue"><strong>[aln out]</strong></font></p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs routeros">braker.pl <span class="hljs-attribute">--species</span>=yourSpecies <span class="hljs-attribute">--genome</span>=genome.fasta \<br>   <span class="hljs-attribute">--prot_aln</span>=proteins.aln <span class="hljs-attribute">--prg</span>=gth --trainFromGth<br></code></pre></td></tr></table></figure>

<p>It is also possible to run BRAKER in this mode using an already prepared hints file. In this case, run: <font color="blue"><strong>[aln trans hint]</strong></font></p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs routeros">braker.pl <span class="hljs-attribute">--species</span>=yourSpecies <span class="hljs-attribute">--genome</span>=genome.fasta \<br>   <span class="hljs-attribute">--hints</span>=hints.gff <span class="hljs-attribute">--prg</span>=gth --trainFromGth<br></code></pre></td></tr></table></figure>

<p><font color="red"><strong>–prg&#x3D;gth –trainFromGth? if i have aug,rna,pro hints, alos set this???</strong></font></p>
<p>Format of the hints file should look like this:</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">chrName</span>   gth2h   CDSpart <span class="hljs-number">105984</span>  <span class="hljs-number">106633</span>  .     -    .    src=P;grp=FBpp0285205;pri=<span class="hljs-number">4</span><br><span class="hljs-attribute">chrName</span>   gth2h   start   <span class="hljs-number">106646</span>  <span class="hljs-number">106648</span>  .     -    .    src=P;grp=FBpp0285205;pri=<span class="hljs-number">4</span><br></code></pre></td></tr></table></figure>

<p>Supported features in column 3 are intron, CDSpart, start, stop.</p>
<h3 id="BRAKER-with-RNA-Seq-and-protein-data"><a href="#BRAKER-with-RNA-Seq-and-protein-data" class="headerlink" title="BRAKER with RNA-Seq and protein data"></a>BRAKER with <font color="red">RNA-Seq and protein data</font></h3><p>Even though BRAKER supports the combination of RNA-Seq and protein data within the BRAKER pipeline, we strongly recommend to run BRAKER twice (<font color="blue"><strong>1x with RNA-Seq only</strong></font>, <font color="blue"><strong>1x with protein data only</strong></font>) and subsequently combine the results of both runs with <font color="blue"><strong>TSEBRA</strong></font>, the Transcript Selector for BRAKER (<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/TSEBRA">https://github.com/Gaius-Augustus/TSEBRA</a>). You find more information on TSEBRA at <a target="_blank" rel="noopener" href="https://www.biorxiv.org/content/10.1101/2021.06.07.447316v1">https://www.biorxiv.org/content/10.1101/2021.06.07.447316v1</a></p>
<p>In the following, we describe that it is possible to combine both data sources with BRAKER, alone, for the sake of completeness.</p>
<p>The  <font color="blue">native mode</font> for running BRAKER  <font color="blue">with RNA-Seq and protein</font> data is <code>--etpmode</code>. This will  <font color="blue">call GeneMark-ETP</font> (which is currently only available as a premature version by using current state GeneMark-ES&#x2F;ET&#x2F;EP&#x2F;EP+, improvements are to be expected, soon), which will  <font color="blue">use RNA-Seq and protein hints for training GeneMark-ETP</font>. Hints that are  <font color="blue"><strong>supported by both</strong> sources and proteins hints of particularly high quality are enforced in gene prediction with <strong>GeneMark-ETP</strong></font>. Subsequently,  <font color="blue"><strong>AUGUSTUS</strong></font> is trained on GeneMark-ETP predictions and genes with hints are predicted by AUGUSTUS. To call the pipeline in this mode, run:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span>=genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--prot_seq</span>=orthodb<span class="hljs-selector-class">.fa</span> \<br>   <span class="hljs-attr">--hints</span>=rnaseq<span class="hljs-selector-class">.hints</span> <span class="hljs-attr">--etpmode</span> <span class="hljs-attr">--softmasking</span><br></code></pre></td></tr></table></figure>

<p>You can or course replace the <code>rnaseq.gff</code> hints file by a BAM-file, e.g. <code>--bam=ranseq.bam</code>.</p>
<p>In addition, the following pipelines can be executed by BRAKER (deprecated pipelines):</p>
<ul>
<li>Adding protein data of short evolutionary distance to gene prediction step</li>
<li>Extending training gene set with proteins of short evolutionary distance</li>
</ul>
<h4 id="Adding-protein-data-of-short-evolutionary-distance-to-gene-prediction-step"><a href="#Adding-protein-data-of-short-evolutionary-distance-to-gene-prediction-step" class="headerlink" title="Adding protein data of short evolutionary distance to gene prediction step"></a>Adding protein data of short evolutionary distance to gene prediction step</h4><p>This pipeline is illustrated in Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig6">7</a>.</p>
<p>In general, add the options</p>
<figure class="highlight brainfuck"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs brainfuck"><span class="hljs-literal">--</span><span class="hljs-comment">prot_seq=proteins</span><span class="hljs-string">.</span><span class="hljs-comment">fa</span> <span class="hljs-literal">--</span><span class="hljs-comment">prg=(gth|exonerate|spaln)</span><br></code></pre></td></tr></table></figure>

<p>to the BRAKER call that is described in section <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#braker-with-rna-seq-data">BRAKER with RNA-Seq data</a>.  <font color="blue">Select one protein alignment tool</font> from <font color="blue"> <strong>GenomeThreader</strong> (<code>gth</code>, <strong>recommended</strong>), Spaln (<code>spaln</code>) or Exonerate (<code>exonerate</code>)</font>. Of course, you may also specify the protein information  <font color="blue">as protein alignment files or hints files as</font> described in section <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#braker-with-proteins-of-short-evolutionary-distance">BRAKER with proteins of short evolutionary distance</a>). This may result in a call similar to:</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs routeros">braker.pl <span class="hljs-attribute">--species</span>=yourSpecies <span class="hljs-attribute">--genome</span>=genome.fasta \<br>   <span class="hljs-attribute">--bam</span>=file1.bam,file2.bam <span class="hljs-attribute">--prot_seq</span>=proteins.fa \<br>   --prg=(gth|exonerate|spaln)<br></code></pre></td></tr></table></figure>

<h4 id="Extending-training-gene-set-with-proteins-of-short-evolutionary-distance"><a href="#Extending-training-gene-set-with-proteins-of-short-evolutionary-distance" class="headerlink" title="Extending training gene set with proteins of short evolutionary distance"></a>Extending training gene set with proteins of short evolutionary distance</h4><p> <font color="blue">If the <strong>number of training gene</strong> structures identified by RNA-Seq data, only, seems to be <strong>too small</strong></font>, you may add training gene structures generated by protein alignments with GenomeThreader to the training gene set. This pipeline is illustrated in Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig7">8</a>.</p>
<p>In general, add the options</p>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs routeros"><span class="hljs-attribute">--prot_seq</span>=proteins.fa <span class="hljs-attribute">--prg</span>=gth --gth2traingenes<br></code></pre></td></tr></table></figure>

<p>to the BRAKER call that is described in section <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#braker-with-rna-seq-data">BRAKER with RNA-Seq data</a>. This may result in a call similar to:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--species</span>=yourSpecies <span class="hljs-attr">--genome</span>=genome<span class="hljs-selector-class">.fasta</span> \<br>   <span class="hljs-attr">--bam</span>=file1<span class="hljs-selector-class">.bam</span>,file2<span class="hljs-selector-class">.bam</span> <span class="hljs-attr">--prot_seq</span>=proteins<span class="hljs-selector-class">.fa</span> \<br>   <span class="hljs-attr">--prg</span>=gth <span class="hljs-attr">--gth2traingenes</span><br></code></pre></td></tr></table></figure>

<h4 id="BRAKER-with-short-and-long-read-RNA-Seq-and-protein-data"><a href="#BRAKER-with-short-and-long-read-RNA-Seq-and-protein-data" class="headerlink" title="BRAKER with short and long read RNA-Seq and protein data"></a>BRAKER with short and long read RNA-Seq and protein data</h4><p>A preliminary protocol for integration of assembled subreads from PacBio ccs sequencing in combination with short read Illumina RNA-Seq and protein database is described at <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER/blob/master/docs/long_reads/long_read_protocol.md">https://github.com/Gaius-Augustus/BRAKER/blob/master/docs/long_reads/long_read_protocol.md</a></p>
<h2 id="Description-of-selected-BRAKER-command-line-options"><a href="#Description-of-selected-BRAKER-command-line-options" class="headerlink" title="Description of selected BRAKER command line options"></a>Description of selected BRAKER command line options</h2><p>Please run <code>braker.pl --help</code> to obtain a full list of options.</p>
<h3 id="–epmode"><a href="#–epmode" class="headerlink" title="–epmode"></a>–epmode</h3><p>Run BRAKER in <font color="blue"><strong>EP-mode</strong>, i.e. with proteins of any evolutionary distance as <strong>processed by ProtHint</strong> within BRAKER. This mode is turned on <strong>by default</strong> when <strong>only protein</strong> input is detected</font>. Should be provided with either <strong>–prot_seq&#x3D;orthodb.fa</strong> or protein hints <strong>–hints&#x3D;prothint_augustus.gff</strong>.</p>
<h3 id="–etpmode"><a href="#–etpmode" class="headerlink" title="–etpmode"></a>–etpmode</h3><p>Run BRAKER in <font color="red"><strong>ETP-mode</strong></font>, i.e. with proteins of any evolutionary distance processed by ProtHint, and with RNA-Seq data. Should to be provided with <code>prot_seq=orthodb.fa</code> and <code>--bam=rnaseq.bam</code>. <font color="blue"><strong>Alternatively, the RNA-Seq and protein hints can be provided as processed hints</strong></font> with the <code>--hints</code> opiton. Please consider using TSEBRA (<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/TSEBRA">https://github.com/Gaius-Augustus/TSEBRA</a>) instead of BRAKER in ETP-mode.</p>
<h3 id="–ab-initio"><a href="#–ab-initio" class="headerlink" title="–ab_initio"></a>–ab_initio</h3><p><strong>Compute AUGUSTUS <em>ab initio</em> predictions in addition</strong> to AUGUSTUS predictions with hints (additional output files: <code>augustus.ab_initio.*</code>. This may be useful for <strong>estimating the quality of training gene parameters</strong> when inspecting predictions in a Browser.</p>
<h3 id="–augustus-args-x3D-”–some-arg-x3D-bla”"><a href="#–augustus-args-x3D-”–some-arg-x3D-bla”" class="headerlink" title="–augustus_args&#x3D;”–some_arg&#x3D;bla”"></a>–augustus_args&#x3D;”–some_arg&#x3D;bla”</h3><p>One or several command line arguments to be passed to AUGUSTUS, if several arguments are given, separate them by whitespace, i.e. <code>&quot;--first_arg=sth --second_arg=sth&quot;</code>. This may be be useful if you know that gene prediction in your particular species benefits from a particular AUGUSTUS argument during the prediction step.</p>
<h3 id="–cores-x3D-INT"><a href="#–cores-x3D-INT" class="headerlink" title="–cores&#x3D;INT"></a>–cores&#x3D;INT</h3><p>Specifies the maximum number of cores that can be used during computation. BRAKER has to run some steps on a single core, others can take advantage of multiple cores. If you use more than 8 cores, this will not speed up all parallelized steps, in particular, the time consuming <code>optimize_augustus.pl</code> will not use more than <strong>8 cores</strong>. However, if you don’t mind some cores being idle, using more than 8 cores will speed up other steps.</p>
<h3 id="–fungus"><a href="#–fungus" class="headerlink" title="–fungus"></a>–fungus</h3><p>GeneMark-EX option: run algorithm with branch point model. Use this option <strong>if you genome is a fungus</strong>.</p>
<h3 id="–softmasking"><a href="#–softmasking" class="headerlink" title="–softmasking"></a>–softmasking</h3><p>Softmasking option for <strong>soft masked genome</strong> files. (Disabled by default.)</p>
<h3 id="–useexisting"><a href="#–useexisting" class="headerlink" title="–useexisting"></a>–useexisting</h3><p>【<strong>Use the present config and parameter files if they exist</strong> for ‘species’; will overwrite original parameters if BRAKER performs an AUGUSTUS training.】??</p>
<h3 id="–crf"><a href="#–crf" class="headerlink" title="–crf"></a>–crf</h3><p>Execute CRF training for AUGUSTUS; resulting parameters are only kept for final predictions if they show higher accuracy than HMM parameters. This increases runtime!</p>
<h3 id="–lambda-x3D-int"><a href="#–lambda-x3D-int" class="headerlink" title="–lambda&#x3D;int"></a>–lambda&#x3D;int</h3><p>Change the parameter $\lambda$ of the Poisson distribution that is used for downsampling training genes according to their number of introns (only genes with up to 5 introns are downsampled). The default value is $\lambda&#x3D;2$. You might want to set it to 0 for organisms that mainly have single-exon genes. (Generally, single-exon genes contribute less value to increasing AUGUSTUS parameters compared to genes with many exons.)</p>
<h3 id="–UTR-x3D-on"><a href="#–UTR-x3D-on" class="headerlink" title="–UTR&#x3D;on"></a>–UTR&#x3D;on</h3><p><font color="blue">Generate <strong>UTR training</strong> examples for AUGUSTUS from <strong>RNA-Seq coverage information</strong></font>, train AUGUSTUS UTR parameters and predict genes with AUGUSTUS and UTRs, including coverage information for RNA-Seq as evidence. This flag only works if –softmasking is also enabled. <em>This is an experimental feature!</em></p>
<p>If you performed a BRAKER run without –UTR&#x3D;on, you can add UTR parameter training and gene prediction with UTR parameters (and only RNA-Seq hints) with the following command:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span>=../genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--addUTR</span>=on <span class="hljs-attr">--softmasking</span> \<br>    <span class="hljs-attr">--bam</span>=../RNAseq<span class="hljs-selector-class">.bam</span> <span class="hljs-attr">--workingdir</span>=<span class="hljs-variable">$wd</span> \<br>    <span class="hljs-attr">--AUGUSTUS_hints_preds</span>=augustus<span class="hljs-selector-class">.hints</span><span class="hljs-selector-class">.gtf</span> \<br>    <span class="hljs-attr">--cores</span>=<span class="hljs-number">8</span> <span class="hljs-attr">--skipAllTraining</span> <span class="hljs-attr">--species</span>=somespecies<br></code></pre></td></tr></table></figure>

<p><font color="red"><strong>–AUGUSTUS_hints_preds</strong>. Modify <strong>augustus.hints.gtf</strong> to point to the <strong>AUGUSTUS predictions with hints from previous BRAKER run</strong></font>; <strong>modify flaning_DNA value to the flanking region from the log file???</strong> of your previous BRAKER run; modify some_new_working_directory to the location where BRAKER should store results of the additional BRAKER run; modify somespecies to the species name used in your previous BRAKER run.</p>
<p><font color="red">–AUGUSTUS_hints_preds</font>: only UTR training and prediction is performed if this option is given</p>
<h3 id="–addUTR-x3D-on"><a href="#–addUTR-x3D-on" class="headerlink" title="–addUTR&#x3D;on"></a>–addUTR&#x3D;on</h3><p>Add UTRs from RNA-Seq converage information to AUGUSTUS gene predictions using GUSHR. No training of UTR parameters and no gene prediction with UTR parameters is performed.</p>
<p>If you performed a BRAKER run without –addUTR&#x3D;on, you can add UTRs results of a previous BRAKER run with the following command:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span>=../genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--addUTR</span>=on <span class="hljs-attr">--softmasking</span> \<br>    <span class="hljs-attr">--bam</span>=../RNAseq<span class="hljs-selector-class">.bam</span> <span class="hljs-attr">--workingdir</span>=<span class="hljs-variable">$wd</span> \<br>    <span class="hljs-attr">--AUGUSTUS_hints_preds</span>=augustus<span class="hljs-selector-class">.hints</span><span class="hljs-selector-class">.gtf</span> <span class="hljs-attr">--cores</span>=<span class="hljs-number">8</span> \<br>    <span class="hljs-attr">--skipAllTraining</span> <span class="hljs-attr">--species</span>=somespecies<br></code></pre></td></tr></table></figure>

<p>Modify <code>augustus.hints.gtf</code> to point to the AUGUSTUS predictions with hints from previous BRAKER run; modify some_new_working_directory to the location where BRAKER should store results of the additional BRAKER run; this run will not modify AUGUSTUS parameters. <font color="red"> <strong>We recommend that you specify the original species of the original run with –species&#x3D;somespecies. Otherwise, BRAKER will create an unneeded species parameters directory Sp_</strong></font></p>
<h3 id="–stranded-x3D-…"><a href="#–stranded-x3D-…" class="headerlink" title="–stranded&#x3D;+,-,.,…"></a>–stranded&#x3D;+,-,.,…</h3><p>If <code>--UTR=on</code> is enabled, strand-separated bam-files can be provided with <code>--bam=plus.bam,minus.bam</code>. In that case, <code>--stranded=...</code> should hold the strands of the bam files (<code>+</code> for plus strand, <code>-</code> for minus strand, <code>.</code> for unstranded). Note that unstranded data will be used in the gene prediction step, only, if the parameter <code>--stranded=...</code> is set. <em>This is an experimental feature! GUSHR currently does not take advantage of stranded data.</em></p>
<h3 id="–makehub-email-x3D-x79-111-117-x72-64-x6d-x61-105-108-46-100-x65"><a href="#–makehub-email-x3D-x79-111-117-x72-64-x6d-x61-105-108-46-100-x65" class="headerlink" title="–makehub --email&#x3D;&#x79;&#111;&#117;&#x72;&#64;&#x6d;&#x61;&#105;&#108;&#46;&#100;&#x65;"></a>–makehub --email&#x3D;<a href="mailto:&#x79;&#111;&#117;&#x72;&#64;&#x6d;&#x61;&#105;&#108;&#46;&#100;&#x65;">&#x79;&#111;&#117;&#x72;&#64;&#x6d;&#x61;&#105;&#108;&#46;&#100;&#x65;</a></h3><p>If <code>--makehub</code> and <code>--email=your@mail.de</code> (with your valid e-mail adress) are provided, a track data hub for visualizing results with the UCSC Genome Browser will be generated using MakeHub (<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/MakeHub">https://github.com/Gaius-Augustus/MakeHub</a>).</p>
<h3 id="–gc-probability-x3D-DECIMAL"><a href="#–gc-probability-x3D-DECIMAL" class="headerlink" title="–gc_probability&#x3D;DECIMAL"></a>–gc_probability&#x3D;DECIMAL</h3><p>By default, GeneMark-EX uses a probability of 0.001 for predicting the donor splice site pattern GC (instead of GT). It may make sense to increase this value for species where this donor splice site is more common. For example, in the species <em>Emiliania huxleyi</em>, about 50% of donor splice sites have the pattern GC (<a target="_blank" rel="noopener" href="https://media.nature.com/original/nature-assets/nature/journal/v499/n7457/extref/nature12221-s2.pdf">https://media.nature.com/original/nature-assets/nature/journal/v499/n7457/extref/nature12221-s2.pdf</a>, page 5).</p>
<h1 id="Output-of-BRAKER"><a href="#Output-of-BRAKER" class="headerlink" title="Output of BRAKER"></a>Output of BRAKER</h1><p>BRAKER produces several important output files in the working directory.</p>
<ul>
<li><font color="red"><strong>augustus.hints.gtf</strong></font>: Genes predicted by AUGUSTUS with hints from given extrinsic evidence. This file will be missing if BRAKER was run with the option <code>--esmode</code>.</li>
<li><font color="blue"><strong>augustus.hints_utr.gtf</strong></font>: This file may contain different contents depending on how you called BRAKER:<ul>
<li>If you ran BRAKER with <font color="blue"><strong>–UTR&#x3D;on</strong></font>, then this file will <strong>contain genes predicted by AUGUSTUS with UTR parameters and coverage information from RNA-Seq</strong> data <strong>in GTF</strong> format.</li>
<li>If you ran BRAKER with <font color="blue"><strong>–addUTR&#x3D;on</strong></font>, then this file will contain genes predicted by AUGUSTUS <strong>without</strong> UTR parameters and without coverage information from RNA-Seq data. Instead, <strong>AUGUSTUS gene predictions with hints will only be extended by UTRs if RNA-Seq coverage allows</strong> it (i.e. no separate AUGUSTUS training or run was performed, UTRs are only added from running GUSHR). Genes in are in GTF format.</li>
</ul>
</li>
</ul>
<p>This file will only be present if BRAKER was executed with the options <code>--UTR=on</code> or <code>--addUTR=on</code> and a RNA-Seq BAM file.</p>
<ul>
<li><font color="blue"><strong>augustus.ab_initio.gtf</strong></font>: <strong>Genes predicted by AUGUSTUS in <em>ab initio</em> mode</strong> in GTF-format. The file will always be present if AUGUSTUS has been run with the option <font color="blue">–esmode</font>. Otherwise, it will only be present if BRAKER was run with the option <font color="blue">–AUGUSTUS_ab_initio</font>.</li>
<li><font color="blue"><strong>augustus.ab_initio_utr.gtf</strong></font>: This file may contain gene predictions with UTRs if you ran BRAKER with<font color="blue"> –UTR&#x3D;on</font>.</li>
</ul>
<p>This file will only be present if BRAKER was executed with the options <code>--UTR=on</code> or <code>--addUTR=on</code> and a RNA-Seq BAM-file, and with the option <code>--AUGUSTUS_ab_initio</code>.</p>
<ul>
<li><font color="blue"><em><em>GeneMark-E</em>&#x2F;genemark.gtf</em>*</font>: Genes predicted by GeneMark-ES&#x2F;ET&#x2F;EP&#x2F;EP+ in GTF-format. This file will be missing if BRAKER was executed with proteins of close homology and the option <code>--trainFromGth</code>.</li>
<li><font color="blue"><strong>braker.gtf</strong></font>: Union of augustus.hints.gtf and reliable GeneMark-EX predictions (genes fully supported by external evidence). In <code>--esmode</code>, this is the union of augustus.ab_initio.gtf and all GeneMark-ES genes. Thus, this set is generally more sensitive (more genes correctly predicted) and can be less specific (more false-positive predictions can be present).</li>
<li><font color="red"><strong>hintsfile.gff</strong></font>: The extrinsic evidence data extracted from RNAseq.bam and&#x2F;or protein data.</li>
</ul>
<p>AUGUSTUS output files may be present with the following name endings and formats:</p>
<ul>
<li><font color="red"><strong>GTF</strong>-format is <strong>always produced</strong></font>.</li>
<li><font color="red"><strong>GFF3</strong></font>-format is produced if the flag <font color="red"><strong>–gff3</strong></font> was specified to BRAKER.</li>
<li><font color="red">Coding sequences in FASTA-format <strong>augustus.hints.codingseq</strong></font> are produced if the flag <font color="blue">–skipGetAnnoFromFasta was not set</font>.</li>
<li><font color="red">Protein sequence files in FASTA-format <strong>augustus.hints.aa</strong></font> are produced if the flag <font color="blue">–skipGetAnnoFromFasta was not set.</font></li>
</ul>
<p>For details about gtf format, see <a target="_blank" rel="noopener" href="http://www.sanger.ac.uk/Software/formats/GFF/">http://www.sanger.ac.uk/Software/formats/GFF/</a>. A GTF-format file contains one line per predicted exon. Example:</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">HS04636</span> AUGUSTUS initial   <span class="hljs-number">966</span> <span class="hljs-number">1017</span> . + <span class="hljs-number">0</span> transcript_id <span class="hljs-string">&quot;g1.1&quot;</span>; gene_id <span class="hljs-string">&quot;g1&quot;</span>;<br><span class="hljs-attribute">HS04636</span> AUGUSTUS internal <span class="hljs-number">1818</span> <span class="hljs-number">1934</span> . + <span class="hljs-number">2</span> transcript_id <span class="hljs-string">&quot;g1.1&quot;</span>; gene_id <span class="hljs-string">&quot;g1&quot;</span>;<br></code></pre></td></tr></table></figure>

<p>The columns (fields) contain:</p>
<figure class="highlight armasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs armasm"><span class="hljs-symbol">seqname</span> source feature start <span class="hljs-meta">end</span> score strand frame transcript ID <span class="hljs-keyword">and</span> gene ID<br></code></pre></td></tr></table></figure>

<p>If the <code>--makehub</code> option was used and MakeHub is available on your system, a hub directory beginning with the name <code>hub_</code> will be created. Copy this directory to a publicly accessible web server. A file <code>hub.txt</code> resides in the directory. Provide the link to that file to the <font color="blue">UCSC Genome Browser for visualizing results</font>.</p>
<h1 id="Example-data"><a href="#Example-data" class="headerlink" title="Example data"></a>Example data</h1><p>An incomplete example data set is contained in the directory <code>BRAKER/example</code>. In order to complete the data set, please <strong>download the RNA-Seq alignment BAM file</strong> (134 MB) with <code>wget</code>:</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs awk">cd BRAKER/example<br>wget http:<span class="hljs-regexp">//</span>topaz.gatech.edu<span class="hljs-regexp">/GeneMark/</span>Braker/RNAseq.bam<br></code></pre></td></tr></table></figure>

<p>In case you have trouble accessing that file, there’s also a copy available from another server:</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs awk">cd BRAKER/example<br>wget http:<span class="hljs-regexp">//</span>bioinf.uni-greifswald.de<span class="hljs-regexp">/augustus/</span>datasets/RNAseq.bam<br></code></pre></td></tr></table></figure>

<p>The example data set was not compiled in order to achieve optimal prediction accuracy, but in order to quickly test pipeline components. The small subset of <strong>the genome used in these test examples is not long enough for BRAKER training</strong> to work well. <font color="blue">Too few genes</font></p>
<h2 id="Data-description"><a href="#Data-description" class="headerlink" title="Data description"></a>Data description</h2><p>Data corresponds to the last 1,000,000 nucleotides of <em>Arabidopsis thaliana</em>‘s chromosome Chr5, split into 8 artificial contigs.</p>
<p>RNA-Seq alignments were obtained by <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/VARUS">VARUS</a>.</p>
<p>The protein sequences are a subset of <a target="_blank" rel="noopener" href="https://v100.orthodb.org/download/odb10_plants_fasta.tar.gz">OrthoDB v10 plants proteins</a>.</p>
<p>List of files:</p>
<ul>
<li><code>genome.fa</code> - genome file in fasta format</li>
<li><code>RNAseq.bam</code> - RNA-Seq alignment file in bam format (this file is not a part of this repository, it must be downloaded separately from <a target="_blank" rel="noopener" href="http://topaz.gatech.edu/GeneMark/Braker/RNAseq.bam">http://topaz.gatech.edu/GeneMark/Braker/RNAseq.bam</a>)</li>
<li><code>RNAseq.hints</code> - RNA-Seq hints (<strong>can be used instead of RNAseq.bam as RNA-Seq input to BRAKER</strong>)</li>
<li><code>proteins.fa</code> - protein sequences in fasta format</li>
</ul>
<p><strong>Set $PATH.</strong></p>
<p>The example data set also contains scripts <code>tests/test*.sh</code> that will execute below listed commands for testing BRAKER with the example data set. You find example <font color="blue"><strong>results of AUGUSTUS and GeneMark-EX</strong></font> in the folder <code>results/test*</code>. Be aware that BRAKER contains several parts where random variables are used, i.e. results that you obtain when running the tests may not be exactly identical. To <strong>compare</strong> your test <strong>results with the reference ones</strong>, you can use the <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER/blob/master/scripts/compare_intervals_exact.pl">compare_intervals_exact.pl</a> script as follows:</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment"># Compare CDS features</span><br>compare_intervals_exact.pl --f1 augustus.hints.gtf --f2 ../../results/test<span class="hljs-variable">$&#123;N&#125;</span>/augustus.hints.gtf --verbose<br><span class="hljs-comment"># Compare transcripts</span><br>compare_intervals_exact.pl --f1 augustus.hints.gtf --f2 ../../results/test<span class="hljs-variable">$&#123;N&#125;</span>/augustus.hints.gtf --trans --verbose<br></code></pre></td></tr></table></figure>

<p>Several tests use <code>--gm_max_intergenic 10000</code> option to make the test runs faster. <font color="red"><strong>It is not recommended to use this option in real BRAKER runs, the speed increase achieved by adjusting this option is negligible on full-sized genomes.</strong></font></p>
<h2 id="Testing-BRAKER-with-RNA-Seq-data"><a href="#Testing-BRAKER-with-RNA-Seq-data" class="headerlink" title="Testing BRAKER with RNA-Seq data"></a>Testing BRAKER with RNA-Seq data</h2><p>The following command will run the pipeline according to Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig2">3</a>:</p>
<img src="BRAKER/f3.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span> genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--bam</span> RNAseq<span class="hljs-selector-class">.bam</span> <span class="hljs-attr">--softmasking</span> <span class="hljs-attr">--cores</span> N<br></code></pre></td></tr></table></figure>

<p>This test is implemented in <code>test1.sh</code>, expected runtime is ~20 minutes.</p>
<h2 id="Testing-BRAKER-with-proteins-of-any-evolutionary-distance"><a href="#Testing-BRAKER-with-proteins-of-any-evolutionary-distance" class="headerlink" title="Testing BRAKER with proteins of any evolutionary distance"></a>Testing BRAKER with proteins of any evolutionary distance</h2><p>The following command will run the pipeline according to Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig3">4</a>: <strong>ProHint</strong></p>
<img src="BRAKER/f4.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span> genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--prot_seq</span> proteins<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--softmasking</span> <span class="hljs-attr">--cores</span> N<br></code></pre></td></tr></table></figure>

<p>This test is implemented in <code>test2.sh</code>, expected runtime is ~20 minutes.</p>
<h2 id="Testing-BRAKER-with-proteins-of-any-evolutionary-distance-and-RNA-Seq"><a href="#Testing-BRAKER-with-proteins-of-any-evolutionary-distance-and-RNA-Seq" class="headerlink" title="Testing BRAKER with proteins of any evolutionary distance and RNA-Seq"></a>Testing BRAKER with proteins of any evolutionary distance and RNA-Seq</h2><p><font color="red"><strong>Please consider using TSEBRA instead of running BRAKER with both RNA-Seq and protein data</strong></font>: <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/TSEBRA">https://github.com/Gaius-Augustus/TSEBRA</a></p>
<p>The following command will run a pipeline that <font color="red">first trains <strong>GeneMark-ETP</strong> with <strong>protein</strong> and <strong>RNA</strong>-Seq hints</font> and subsequently trains AUGUSTUS on the basis of GeneMark-ETP predictions. <strong>AUGUSTUS predictions</strong> are also performed with hints from <strong>both sources</strong>, see Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig4">5</a>:</p>
<img src="BRAKER/f5.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span> genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--prot_seq</span> proteins<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--bam</span> ../RNAseq<span class="hljs-selector-class">.bam</span> <span class="hljs-attr">--etpmode</span> <span class="hljs-attr">--softmasking</span> <span class="hljs-attr">--cores</span> N<br></code></pre></td></tr></table></figure>

<p>This test is implemented in <code>test3.sh</code>, expected runtime is ~20 minutes.</p>
<p>You can <font color="red"><strong>add UTRs</strong> from RNA-Seq data</font> (<font color="red"><strong>no AUGUSTUS training</strong></font>) to <strong>results of a BRAKER run in ETP-mode</strong> the following way:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span>=../genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--addUTR</span>=on <span class="hljs-attr">--softmasking</span> \<br>    <span class="hljs-attr">--bam</span>=../RNAseq<span class="hljs-selector-class">.bam</span> <span class="hljs-attr">--workingdir</span>=<span class="hljs-variable">$wd</span> \<br>    <span class="hljs-attr">--AUGUSTUS_hints_preds</span>=augustus<span class="hljs-selector-class">.hints</span><span class="hljs-selector-class">.gtf</span> <span class="hljs-attr">--cores</span>=<span class="hljs-number">8</span> \<br>    <span class="hljs-attr">--skipAllTraining</span> <span class="hljs-attr">--species</span>=somespecies<br></code></pre></td></tr></table></figure>

<p>This is implemented in <code>test3_add_utrs.sh</code>, expected <strong>runtime is ~1 minute.</strong></p>
<h2 id="Testing-BRAKER-with-proteins-of-close-homology"><a href="#Testing-BRAKER-with-proteins-of-close-homology" class="headerlink" title="Testing BRAKER with proteins of close homology"></a>Testing BRAKER with proteins of close homology</h2><p>The following command will run the pipeline according to Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig5">6</a>:</p>
<img src="BRAKER/f6.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span> genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--prot_seq</span> proteins<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--prg</span> gth \<br>    <span class="hljs-attr">--trainFromGth</span> <span class="hljs-attr">--softmasking</span> <span class="hljs-attr">--cores</span> N<br></code></pre></td></tr></table></figure>

<p>This test is implemented in <code>test4.sh</code>, expected runtime is ~7 minutes. The fast runtime of this test is mostly caused by <font color="red"><strong>generating a low number of training genes [no GeneMark]</strong></font>. Note that this approach <font color="red"> not scale well with increasing genome size and the number of proteins in a protein database</font>. The runtime on a full genome will be much slower than with the command used in <code>test2.sh</code>.</p>
<h2 id="Testing-BRAKER-with-proteins-of-close-homology-and-RNA-Seq-data-RNA-Seq-supported-training"><a href="#Testing-BRAKER-with-proteins-of-close-homology-and-RNA-Seq-data-RNA-Seq-supported-training" class="headerlink" title="Testing BRAKER with proteins of close homology and RNA-Seq data (RNA-Seq supported training)"></a>Testing BRAKER with proteins of close homology and RNA-Seq data (<font color="red">RNA-Seq supported training</font>)</h2><p><font color="red">Please consider using <strong>TSBERA</strong> instead of running BRAKER with <strong>proteins</strong> and <strong>RNA</strong>-Seq</font>font&gt; data at the same time: <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/TSEBRA">https://github.com/Gaius-Augustus/TSEBRA</a></p>
<p>The following command will run the pipeline according to Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig6">7</a>:</p>
<img src="BRAKER/f7.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment">#no etp</span><br>braker.pl --genome genome.fa --prot_seq proteins.fa --prg gth \<br>    --bam RNAseq.bam --softmasking --cores N<br></code></pre></td></tr></table></figure>

<p>This test is implemented in <code>test5.sh</code>, expected runtime is ~20 minutes.</p>
<h2 id="Testing-BRAKER-with-proteins-of-close-homology-and-RNA-Seq-data-RNA-Seq-and-protein-supported-training"><a href="#Testing-BRAKER-with-proteins-of-close-homology-and-RNA-Seq-data-RNA-Seq-and-protein-supported-training" class="headerlink" title="Testing BRAKER with proteins of close homology and RNA-Seq data (RNA-Seq and protein supported training)"></a>Testing BRAKER with proteins of close homology and RNA-Seq data (<font color="red">RNA-Seq and protein supported training</font>)</h2><p>Please consider using TSBERA instead of running BRAKER with proteins and RNA-Seq data at the same time: <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/TSEBRA">https://github.com/Gaius-Augustus/TSEBRA</a></p>
<p>The following command will run the pipeline according to Figure <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#fig7">8</a>:</p>
<img src="BRAKER/f8.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment"># --gth2traingenes</span><br>braker.pl --genome genome.fa --prot_seq prot.fa --prg gth --bam RNAseq.bam \<br>    --gth2traingenes --softmasking --cores N<br></code></pre></td></tr></table></figure>

<p>This test is implemented in <code>test6.sh</code>, expected runtime is ~20 minutes.</p>
<h2 id="Testing-BRAKER-with-pre-trained-parameters"><a href="#Testing-BRAKER-with-pre-trained-parameters" class="headerlink" title="Testing BRAKER with pre-trained parameters"></a>Testing BRAKER with pre-trained parameters</h2><p>The training step of all pipelines can be skipped with the option <code>--skipAllTraining</code>. This means, <font color="red"><strong>only AUGUSTUS predictions</strong> will be <strong>performed</strong></font>, using pre-trained, already existing parameters. For example, you can predict genes with the command:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span>=genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--bam</span> RNAseq<span class="hljs-selector-class">.bam</span> <span class="hljs-attr">--species</span>=arabidopsis \<br>    <span class="hljs-attr">--skipAllTraining</span> <span class="hljs-attr">--softmasking</span> <span class="hljs-attr">--cores</span> N<br></code></pre></td></tr></table></figure>

<p>This test is implemented in <code>test7.sh</code>, expected runtime is ~1 minute.</p>
<h2 id="Testing-BRAKER-with-genome-sequence"><a href="#Testing-BRAKER-with-genome-sequence" class="headerlink" title="Testing BRAKER with genome sequence"></a>Testing BRAKER with genome sequence</h2><p>The following command will run the pipeline with <strong>no extrinsic evidence</strong>:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span>=genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--esmode</span> <span class="hljs-attr">--softmasking</span> <span class="hljs-attr">--cores</span> N<br></code></pre></td></tr></table></figure>

<p>This test is implemented in <code>test8.sh</code>, expected runtime is ~20 minutes.</p>
<h2 id="Testing-BRAKER-with-RNA-Seq-data-and-–UTR-x3D-on"><a href="#Testing-BRAKER-with-RNA-Seq-data-and-–UTR-x3D-on" class="headerlink" title="Testing BRAKER with RNA-Seq data and –UTR&#x3D;on"></a>Testing BRAKER with RNA-Seq data and –UTR&#x3D;on</h2><p>The following command will run BRAKER with <strong><font color="red">training</font> UTR parameters from RNA-Seq coverage data</strong>:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span> genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--bam</span> RNAseq<span class="hljs-selector-class">.bam</span> <span class="hljs-attr">--softmasking</span> <span class="hljs-attr">--UTR</span>=on <span class="hljs-attr">--cores</span> N<br></code></pre></td></tr></table></figure>

<p>This test is implemented in <code>test9.sh</code>, expected runtime is ~20 minutes.</p>
<h2 id="Testing-BRAKER-with-RNA-Seq-data-and-–addUTR-x3D-on"><a href="#Testing-BRAKER-with-RNA-Seq-data-and-–addUTR-x3D-on" class="headerlink" title="Testing BRAKER with RNA-Seq data and –addUTR&#x3D;on"></a>Testing BRAKER with RNA-Seq data and –addUTR&#x3D;on</h2><p>The following command will <strong><font color="red">add[ no training]</font> UTRs to augustus.hints.gtf from RNA-Seq coverage data</strong>:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">braker<span class="hljs-selector-class">.pl</span> <span class="hljs-attr">--genome</span> genome<span class="hljs-selector-class">.fa</span> <span class="hljs-attr">--bam</span> RNAseq<span class="hljs-selector-class">.bam</span> <span class="hljs-attr">--softmasking</span> <span class="hljs-attr">--addUTR</span>=on <span class="hljs-attr">--cores</span> N<br></code></pre></td></tr></table></figure>

<p>This test is implemented in <code>test10.sh</code>, expected runtime is ~20 minutes.</p>
<h1 id="Starting-BRAKER-on-the-basis-of-previously-existing-BRAKER-runs"><a href="#Starting-BRAKER-on-the-basis-of-previously-existing-BRAKER-runs" class="headerlink" title="Starting BRAKER on the basis of previously existing BRAKER runs"></a>Starting BRAKER on the basis of previously existing BRAKER runs</h1><p>There is currently no clean way to restart a failed BRAKER run (after solving some problem). However, it is possible to start a new BRAKER run based on results from a previous run – given that the old run produced the required intermediate results. We will in the following refer to the old working directory with variable <code>$&#123;BRAKER_OLD&#125;</code>, and to the new BRAKER working directory with <code>$&#123;BRAKER_NEW&#125;</code>. The file <code>what-to-cite.txt</code> will always only refer to the software that was actually called by a particular run. You might have to combine the contents of <code>$&#123;BRAKER_NEW&#125;/what-to-cite.txt</code> with <code>$&#123;BRAKER_OLD&#125;/what-to-cite.txt</code> for preparing a publication. The following figure illustrates at which points BRAKER run may be intercepted.</p>
<img src="BRAKER/f10.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="braker2-main-a[fig1]" style="zoom:50%;" />

<p>Figure 10: Points for intercepting a BRAKER run and reusing intermediate results in a new BRAKER run.</p>
<h2 id="Option-1-starting-BRAKER-with-existing-hints-file-s-before-training-my-opo-hints-file-generated-before-GeneMark-and-Aug"><a href="#Option-1-starting-BRAKER-with-existing-hints-file-s-before-training-my-opo-hints-file-generated-before-GeneMark-and-Aug" class="headerlink" title="Option 1: starting BRAKER with existing hints file(s) before training[my opo:hints file generated before GeneMark and Aug]"></a>Option 1: starting BRAKER with existing hints file(s) before training[my opo:hints file generated before GeneMark and Aug]</h2><p>If you have access to an existing BRAKER output that contains <strong>hintsfiles</strong> that were <strong>generated from</strong> extrinsic data, such as <strong>RNA</strong>-Seq or <strong>protein</strong> sequences, you can <strong>recycle these hints files in a new BRAKER run</strong>. Also, <strong>hints from a separate ProtHint run</strong> can be <strong>directly used in BRAKER</strong>.</p>
<p>The hints can be given to BRAKER with <font color="red"><strong>–hints ${BRAKER_OLD}&#x2F;hintsfile.gff</strong></font> option. This is illustrated in the test files <code>test1_restart1.sh</code>, <code>test2_restart1.sh</code>, <code>test3_restart1.sh</code>, <code>test5_restart1.sh</code>, and <code>test7_restart1.sh</code>. The other modes (for which this test is missing) cannot be restarted in this way.</p>
<h2 id="Option-2-starting-BRAKER-after-GeneMark-EX-had-finished-before-training-AUGUSTUS"><a href="#Option-2-starting-BRAKER-after-GeneMark-EX-had-finished-before-training-AUGUSTUS" class="headerlink" title="Option 2: starting BRAKER after GeneMark-EX had finished, before training AUGUSTUS"></a>Option 2: starting BRAKER after GeneMark-EX had finished, before training AUGUSTUS</h2><p>The GeneMark result can be given to BRAKER with <font color="red"><strong>–geneMarkGtf</strong> ${BRAKER_OLD}&#x2F;GeneMark-EX&#x2F;genemark.gtf</font> option. This is illustrated in the test files <code>test1_restart2.sh</code>, <code>test2_restart2.sh</code>, <code>test3_restart2.sh</code>, <code>test5_restart2.sh</code>, and <code>test8_restart2.sh</code>. The other modes (for which this test is missing) cannot be restarted in this way.</p>
<h2 id="Option-3-starting-BRAKER-after-AUGUSTUS-training"><a href="#Option-3-starting-BRAKER-after-AUGUSTUS-training" class="headerlink" title="Option 3: starting BRAKER after AUGUSTUS training"></a>Option 3: starting BRAKER after AUGUSTUS training</h2><p>The trained species parameters for AGUSTUS can be passed with <font color="red"><strong>–skipAllTraining and –species $speciesName</strong></font> options. This is illustrated in <code>test*_restart3.sh</code> files.</p>
<p>My opo: each training creat a new dir, can be use again.</p>
<h1 id="Bug-reporting"><a href="#Bug-reporting" class="headerlink" title="Bug reporting"></a>Bug reporting</h1><p>Before reporting bugs, please check that you are using the most recent versions of GeneMark-EX, AUGUSTUS and BRAKER. Also, check the list of <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER#common-problems">Common problems</a>, and the Issue list on GitHub before reporting bugs. We do monitor open issues on GitHub. Sometimes, we are unable to help you, immediately, but we try hard to solve your problems.</p>
<h2 id="Reporting-bugs-on-GitHub"><a href="#Reporting-bugs-on-GitHub" class="headerlink" title="Reporting bugs on GitHub"></a>Reporting bugs on GitHub</h2><p>If you found a bug, please open an issue at <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER/issues">https://github.com/Gaius-Augustus/BRAKER/issues</a> (or contact <a href="mailto:&#107;&#x61;&#x74;&#104;&#97;&#x72;&#105;&#x6e;&#97;&#46;&#x68;&#111;&#x66;&#x66;&#64;&#117;&#x6e;&#x69;&#x2d;&#x67;&#114;&#101;&#x69;&#102;&#x73;&#119;&#x61;&#108;&#x64;&#x2e;&#100;&#101;">&#107;&#x61;&#x74;&#104;&#97;&#x72;&#105;&#x6e;&#97;&#46;&#x68;&#111;&#x66;&#x66;&#64;&#117;&#x6e;&#x69;&#x2d;&#x67;&#114;&#101;&#x69;&#102;&#x73;&#119;&#x61;&#108;&#x64;&#x2e;&#100;&#101;</a> or <a href="mailto:&#98;&#x72;&#117;&#x6e;&#97;&#x2e;&#x74;&#111;&#109;&#97;&#115;&#x40;&#x67;&#97;&#116;&#101;&#99;&#x68;&#x2e;&#x65;&#100;&#117;">&#98;&#x72;&#117;&#x6e;&#97;&#x2e;&#x74;&#111;&#109;&#97;&#115;&#x40;&#x67;&#97;&#116;&#101;&#99;&#x68;&#x2e;&#x65;&#100;&#117;</a>).</p>
<p>Information worth mentioning in your bug report:</p>
<p>Check in <code>braker/yourSpecies/braker.log</code> at which step <code>braker.pl</code> crashed.</p>
<p>There are a number of other files that might be of interest, depending on where in the pipeline the problem occurred. Some of the following files will not be present if they did not contain any <strong>errors</strong>.</p>
<ul>
<li><code>braker/yourSpecies/errors/bam2hints.*.stderr</code> - will give details on a bam2hints crash (<strong>step for converting bam file to intron gff file</strong>)</li>
<li><code>braker/yourSpecies/hintsfile.gff</code> - is this file empty? If yes, something went wrong during hints generation - does this file contain hints from source “<strong>b2h</strong>” and of type “<strong>intron</strong>”? If not: <strong>GeneMark-ET</strong>(RNA) will not be able to <strong>execute</strong> properly. Conversely, <strong>GeneMark-EP+</strong>(pro) will not be able to execute correctly if hints from the source “<strong>ProtHint</strong>“ are missing.</li>
<li><code>braker/yourSpecies/(align_gthalign_exoneratealign_spaln)/*err</code> - <strong>errors reported by the alignment tools gth&#x2F;exonerate&#x2F;spaln</strong></li>
<li><code>braker/yourSpecies/errors/GeneMark-&#123;ET,EP&#125;.stderr</code> - <strong>errors reported by GeneMark-ET&#x2F;EP+</strong></li>
<li><code>braker/yourSpecies/errors/GeneMark-&#123;ET,EP).stdout</code> - may give clues about the point at <strong>which errors in GeneMark-ET&#x2F;EP+ occured</strong></li>
<li><code>braker/yourSpecies/GeneMark-&#123;ET,EP&#125;/genemark.gtf</code> - is this file empty? If yes, something went wrong during executing GeneMark-ET&#x2F;EP+ <strong>My is gene mark_hintsfile.gff? no ,in GeneMark-ET</strong></li>
<li><code>braker/yourSpecies/GeneMark-&#123;ET,EP&#125;/genemark.f.good.gtf</code> - is this file empty? If yes, something went wrong during <font color="red"><strong>filtering GeneMark-ET&#x2F;EP+ genes for training AUGUSTUS no??</strong></font></li>
<li><code>braker/yourSpecies/genbank.good.gb</code> - try a “grep -c LOCUS genbank.good.gb” to determine the number of training genes for training AUGUSTUS, should not be low. <font color="red"><strong>no??</strong></font></li>
<li><code>braker/yourSpecies/errors/firstetraining.stderr</code> - contains errors from first iteration of training AUGUSTUS</li>
<li><code>braker/yourSpecies/errors/secondetraining.stderr</code> - contains errors from second iteration of training AUGUSTUS</li>
<li><code>braker/yourSpecies/errors/optimize_augustus.stderr</code> - contains errors <strong>optimize_augustus.pl</strong> (<strong>additional training set for AUGUSTUS</strong>)</li>
<li><code>braker/yourSpecies/errors/augustus*.stderr</code> - contain <strong>AUGUSTUS execution</strong> errors</li>
<li><code>braker/yourSpecies/startAlign.stderr</code> - if you provided a protein fasta file and <code>--prg</code> option and this file is not empty, something went <strong>wrong</strong> during <strong>protein alignment</strong></li>
<li><code>braker/yourSpecies/startAlign.stdout</code> - may give clues on at which point protein alignment went wrong</li>
</ul>
<h2 id="Common-problems"><a href="#Common-problems" class="headerlink" title="Common problems"></a>Common problems</h2><ul>
<li><p><em>BRAKER complains that the RNA-Seq file does not correspond to the provided genome file, but I am sure the files correspond to each other!</em></p>
<p>Please check the headers of the genome FASTA file. If the headers are long and contain whitespaces, some RNA-Seq alignment tools will truncate sequence names in the BAM file. This leads to an error with BRAKER. Solution: shorten&#x2F;simplify FASTA headers in the genome file before running the RNA-Seq alignment and BRAKER.</p>
</li>
<li><p><em>There are duplicate Loci in the train.gb file (after using GenomeThreader)!</em></p>
<p>This issue arises if outdated versions of AUGUSTUS and BRAKER are used. Solution: Please update AUGUSTUS and BRAKER from github (<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/Augustus">https://github.com/Gaius-Augustus/Augustus</a>, <a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER">https://github.com/Gaius-Augustus/BRAKER</a>).</p>
</li>
<li><p><em>GeneMark fails!</em></p>
<p>(a) GeneMark requires a valid hidden key file in your home directory (<code>~/.gm_key</code>). The file expires after 200 days. Please check whether you have a valid key file before reporting an issue about this. Also, BRAKER may issue a WARNING that GeneMark is likely going to fail due to limited extrinsic evidence. If you see that warning, please don’t open an issue but try a different approach towards annotating your genome. For example, you can add more evidence data, you can try the protein mapping pipeline approach, you can try running <code>--esmode</code> without extrinsic evidence, …</p>
<p>(b) GeneMark by default only uses contigs longer than 50k for training. If you have a highly fragmented assembly, this might lead to “no data” for training. You can override the default minimal length by setting the BRAKER argument <code>--min_contig=10000</code>.</p>
<p>(c) see “[something] failed to execute” below.</p>
</li>
<li><p><em>[something] failed to execute!</em></p>
<p>When providing paths to software to BRAKER, please use absolute, non-abbreviated paths. For example, BRAKER might have problems with <code>--SAMTOOLS_PATH=./samtools/</code> or <code>--SAMTOOLS_PATH=~/samtools/</code>. Please use <code>SAMTOOLS_PATH=/full/absolute/path/to/samtools/</code>, instead. This applies to all path specifications as command line options to <code>braker.pl</code>. Relative paths and absolute paths will not pose problems if you export a bash variable, instead, or if you append the location of tools to your $PATH variable.</p>
</li>
<li><p><em>BRAKER cannot find the Augustus script XYZ…</em></p>
<p>Update Augustus from github with <code>git clone https://github.com/Gaius-Augustus/Augustus.git</code>. Do not use Augustus from other sources. BRAKER is highly dependent on an up-to-date Augustus. Augustus releases happen rather rarely, updates to the Augustus scripts folder occur rather frequently.</p>
</li>
<li><p><em>Does BRAKER depend on Python3?</em></p>
<p>It does. The python scripts employed by BRAKER are not compatible with Python2.</p>
</li>
<li><p><em>Why does BRAKER predict more genes than I expected?</em></p>
<p>If transposable elements (or similar) have not been masked appropriately, AUGUSTUS tends to predict those elements as protein coding genes. This can lead to a huge number genes. You can check whether this is the case for your project by BLASTing (or DIAMONDing) the predicted protein sequences against themselves (all vs. all) and counting how many of the proteins have a high number of high quality matches. You can use the output of this analysis to divide your gene set into two groups: the protein coding genes that you want to find and the repetitive elements that were additionally predicted.</p>
</li>
<li><p><em>I am running BRAKER in Anaconda and something fails…</em></p>
<p>Update AUGUSTUS and BRAKER from github with <code>git clone https://github.com/Gaius-Augustus/Augustus.git</code> and <code>git clone https://github.com/Gaius-Augustus/BRAKER.git</code>. The Anaconda installation is great, but it relies on releases of AUGUSTUS and BRAKER - which are often lagging behind. Please use the current GitHub code, instead.</p>
</li>
</ul>
<h1 id="Citing-BRAKER-and-software-called-by-BRAKER"><a href="#Citing-BRAKER-and-software-called-by-BRAKER" class="headerlink" title="Citing BRAKER and software called by BRAKER"></a>Citing BRAKER and software called by BRAKER</h1><p>Since BRAKER is a pipeline that calls several Bioinformatics tools, publication of results obtained by BRAKER requires that not only BRAKER is cited, but also the tools that are called by BRAKER. BRAKER will output a file <code>what-to-cite.txt</code> in the BRAKER working directory, informing you about which exact sources apply to your run.</p>
<ul>
<li>Always cite:<ul>
<li>Bruna, T., Hoff, K.J., Lomsadze, A., Stanke, M., &amp; Borodovsky, M. (2021). BRAKER2: Automatic Eukaryotic Genome Annotation with GeneMark-EP+ and AUGUSTUS Supported by a Protein Database. NAR Genomics and Bioinformatics 3(1):lqaa108, doi: 10.1093&#x2F;nargab&#x2F;lqaa108.</li>
<li>Hoff, K.J., Lomsadze, A., Borodovsky, M. and Stanke, M. (2019). Whole-Genome Annotation with BRAKER. Methods Mol Biol. 1962:65-95, doi: 10.1007&#x2F;978-1-4939-9173-0_5.</li>
<li>Hoff, K.J., Lange, S., Lomsadze, A., Borodovsky, M. and Stanke, M. (2016). BRAKER1: unsupervised RNA-Seq-based genome annotation with GeneMark-ET and AUGUSTUS. Bioinformatics, 32(5):767-769.</li>
<li>Stanke, M., Diekhans, M., Baertsch, R. and Haussler, D. (2008). Using native and syntenically mapped cDNA alignments to improve de novo gene finding. Bioinformatics, doi: 10.1093&#x2F;bioinformatics&#x2F;btn013.</li>
<li>Stanke. M., Schöffmann, O., Morgenstern, B. and Waack, S. (2006). Gene prediction in eukaryotes with a generalized hidden Markov model that uses hints from external sources. BMC Bioinformatics 7, 62.</li>
</ul>
</li>
<li>If any kind of AUGUSTUS training was performed by BRAKER, check carefully whether you configured BRAKER to use NCBI BLAST or DIAMOND. One of them was used to filter out redundant training gene structures.<ul>
<li>If you used NCBI BLAST, please cite:<ul>
<li>Altschul, A.F., Gish, W., Miller, W., Myers, E.W. and Lipman, D.J. (1990). A basic local alignment search tool. J Mol Biol 215:403–410.</li>
<li>Camacho, C., Coulouris, G., Avagyan, V., Ma, N., Papadopoulos, J., Bealer, K., and Madden, T.L. (2009). Blast+: architecture and applications. BMC bioinformatics, 10(1):421.</li>
</ul>
</li>
<li>If you used DIAMOND, please cite:<ul>
<li>Buchfink, B., Xie, C., Huson, D.H. (2015). Fast and sensitive protein alignment using DIAMOND. Nature Methods 12:59-60.</li>
</ul>
</li>
</ul>
</li>
<li>If BRAKER was executed with a genome file and no extrinsic evidence, cite, then GeneMark-ES was used, cite:<ul>
<li>Lomsadze, A., Ter-Hovhannisyan, V., Chernoff, Y.O. and Borodovsky, M. (2005). Gene identification in novel eukaryotic genomes by self-training algorithm. Nucleic Acids Research, 33(20):6494–6506.</li>
<li>Ter-Hovhannisyan, V., Lomsadze, A., Chernoff, Y.O. and Borodovsky, M. (2008). Gene prediction in novel fungal genomes using an ab initio algorithm with unsupervised training. Genome research, pages gr–081612, 2008.</li>
</ul>
</li>
<li>If BRAKER was run with proteins of any phylogenetic distance (–epmode or –etpmode), please cite all tools that are used by the ProtHint pipeline to generate hints:<ul>
<li>Bruna, T., Lomsadze, A., &amp; Borodovsky, M. (2020). GeneMark-EP+: eukaryotic gene prediction with self-training in the space of genes and proteins. NAR Genomics and Bioinformatics, 2(2), lqaa026.</li>
<li>Buchfink, B., Xie, C., Huson, D.H. (2015). Fast and sensitive protein alignment using DIAMOND. Nature Methods 12:59-60.</li>
<li>Lomsadze, A., Ter-Hovhannisyan, V., Chernoff, Y.O. and Borodovsky, M. (2005). Gene identification in novel eukaryotic genomes by self-training algorithm. Nucleic Acids Research, 33(20):6494–6506.</li>
<li>Iwata, H., and Gotoh, O. (2012). Benchmarking spliced alignment programs including Spaln2, an extended version of Spaln that incorporates additional species-specific features. Nucleic acids research, 40(20), e161-e161.</li>
<li>Gotoh, O., Morita, M., Nelson, D. R. (2014). Assessment and refinement of eukaryotic gene structure prediction with gene-structure-aware multiple protein sequence alignment. BMC bioinformatics, 15(1), 189.</li>
</ul>
</li>
<li>If BRAKER was executed with RNA-Seq alignments in bam-format, then SAMtools was used, cite:<ul>
<li>Li, H., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., Marth, G., Abecasis, G., Durbin, R.; 1000 Genome Project Data Processing Subgroup (2009). The Sequence Alignment&#x2F;Map format and SAMtools. Bioinformatics, 25(16):2078-9.</li>
<li>Barnett, D.W., Garrison, E.K., Quinlan, A.R., Strömberg, M.P. and Marth G.T. (2011). BamTools: a C++ API and toolkit for analyzing and managing BAM files. Bioinformatics, 27(12):1691-2</li>
</ul>
</li>
<li>If BRAKER used RNA-Seq alignments for generating a training gene set, cite GeneMark-ET:<ul>
<li>Lomsadze, A., Paul D.B., and Mark B. (2014) Integration of Mapped Rna-Seq Reads into Automatic Training of Eukaryotic Gene Finding Algorithm. Nucleic Acids Research 42(15): e119–e119</li>
</ul>
</li>
<li>If BRAKER was executed with proteins of closely related species, cite GenomeThreader:<ul>
<li>Gremme, G. (2013). Computational Gene Structure Prediction. PhD thesis, Universität Hamburg.</li>
</ul>
</li>
<li>If BRAKER called MakeHub for creating a track data hub for visualization of BRAKER results with the UCSC Genome Browser, cite:<ul>
<li>Hoff, K.J. (2019) MakeHub: Fully automated generation of UCSC Genome Browser Assembly Hubs. Genomics, Proteomics and Bioinformatics, in press 2020, preprint on bioarXive, doi: <a target="_blank" rel="noopener" href="https://doi.org/10.1101/550145">https://doi.org/10.1101/550145</a>.</li>
</ul>
</li>
<li>If BRAKER called GUSHR for generating UTRs, cite:<ul>
<li>Keilwagen, J., Hartung, F., Grau, J. (2019) GeMoMa: Homology-based gene prediction utilizing intron position conservation and RNA-seq data. Methods Mol Biol. 1962:161-177, doi: 10.1007&#x2F;978-1-4939-9173-0_9.</li>
<li>Keilwagen, J., Wenk, M., Erickson, J.L., Schattat, M.H., Grau, J., Hartung F. (2016) Using intron position conservation for homology-based gene prediction. Nucleic Acids Research, 44(9):e89.</li>
<li>Keilwagen, J., Hartung, F., Paulini, M., Twardziok, S.O., Grau, J. (2018) Combining RNA-seq data and homology-based gene prediction for plants, animals and fungi. BMC Bioinformatics, 19(1):189.</li>
</ul>
</li>
</ul>
<h1 id="License"><a href="#License" class="headerlink" title="License"></a>License</h1><p>All source code, i.e. <code>scripts/*.pl</code> or <code>scripts/*.py</code> are under the Artistic License (see <a target="_blank" rel="noopener" href="http://www.opensource.org/licenses/artistic-license.php">http://www.opensource.org/licenses/artistic-license.php</a>).</p>
<h1 id=""><a href="#" class="headerlink" title=""></a></h1><hr>
<h1 id="TSEBRA"><a href="#TSEBRA" class="headerlink" title="TSEBRA"></a>TSEBRA</h1><p><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/TSEBRA">https://github.com/Gaius-Augustus/TSEBRA</a></p>
<h1 id="TSEBRA-Transcript-Selector-for-BRAKER"><a href="#TSEBRA-Transcript-Selector-for-BRAKER" class="headerlink" title="TSEBRA: Transcript Selector for BRAKER"></a>TSEBRA: Transcript Selector for BRAKER</h1><h2 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h2><p><a target="_blank" rel="noopener" href="https://doi.org/10.1186/s12859-021-04482-0">TSEBRA</a> is a combiner tool that <font color="blue">selects transcripts from gene predictions</font> based on the support by extrisic evidence in form of <font color="blue">introns and start&#x2F;stop codons</font>. It was developed to combine BRAKER1<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/TSEBRA#ref1">1</a> and BRAKER2<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/TSEBRA#ref2">2</a>predicitons to increase their accuracies.</p>
<h2 id="Prerequisites"><a href="#Prerequisites" class="headerlink" title="Prerequisites"></a>Prerequisites</h2><p>Python 3.5.2 or higher is required.</p>
<h2 id="Installation-1"><a href="#Installation-1" class="headerlink" title="Installation"></a>Installation</h2><p>Download TSEBRA:</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs awk">git clone https:<span class="hljs-regexp">//gi</span>thub.com<span class="hljs-regexp">/Gaius-Augustus/</span>TSEBRA<br></code></pre></td></tr></table></figure>

<h2 id="Usage"><a href="#Usage" class="headerlink" title="Usage"></a>Usage</h2><p>The main script is <code>./bin/tsebra.py</code>. For usage information run <code>./bin/tsebra.py --help</code>.</p>
<h2 id="Input-Files"><a href="#Input-Files" class="headerlink" title="Input Files"></a>Input Files</h2><p>TSEBRA takes a list of <font color="blue">gene prediciton files</font>, a list of <font color="blue">hintfiles</font> and a <font color="blue">configuration file</font> as mandatory input.</p>
<h4 id="1-Gene-Predictions"><a href="#1-Gene-Predictions" class="headerlink" title="1 Gene Predictions"></a>1 Gene Predictions</h4><p>The gene prediction files have to be in <font color="blue">gtf format</font>. This is the standard output format of a BRAKER or AUGUSTUS<a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/TSEBRA#ref3">3,</a><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/TSEBRA#ref4">4</a> gene prediciton.</p>
<p>Example:</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">2L</span>      AUGUSTUS        gene    <span class="hljs-number">83268</span>   <span class="hljs-number">87026</span>   <span class="hljs-number">0</span>.<span class="hljs-number">88</span>    -       .       g5332<br><span class="hljs-attribute">2L</span>      AUGUSTUS        transcript      <span class="hljs-number">83268</span>   <span class="hljs-number">87026</span>   <span class="hljs-number">0</span>.<span class="hljs-number">88</span>    -       .       g5332.t1<br><span class="hljs-attribute">2L</span>      AUGUSTUS        intron  <span class="hljs-number">84278</span>   <span class="hljs-number">87019</span>   <span class="hljs-number">1</span>       -       .       transcript_id <span class="hljs-string">&quot;file_1_file_1_g5332.t1&quot;</span>; gene_id <span class="hljs-string">&quot;file_1_file_1_g5332&quot;</span>;<br><span class="hljs-attribute">2L</span>      AUGUSTUS        CDS     <span class="hljs-number">87020</span>   <span class="hljs-number">87026</span>   <span class="hljs-number">0</span>.<span class="hljs-number">88</span>    -       <span class="hljs-number">0</span>       transcript_id <span class="hljs-string">&quot;file_1_file_1_g5332.t1&quot;</span>; gene_id <span class="hljs-string">&quot;file_1_file_1_g5332&quot;</span>;<br><span class="hljs-attribute">2L</span>      AUGUSTUS        exon    <span class="hljs-number">87020</span>   <span class="hljs-number">87026</span>   .       -       .       transcript_id <span class="hljs-string">&quot;file_1_file_1_g5332.t1&quot;</span>; gene_id <span class="hljs-string">&quot;file_1_file_1_g5332&quot;</span>;<br></code></pre></td></tr></table></figure>

<h4 id="2-Hint-Files"><a href="#2-Hint-Files" class="headerlink" title="2 Hint Files"></a>2 Hint Files</h4><p>The hints files have to be in <font color="blue">gff format</font>, the last column must include an attribute for <font color="blue">the source for the hint with <strong>‘src&#x3D;’</strong></font> and can include <font color="blue">the number of hints supporting the gene structure segment with <strong>‘mult&#x3D;’</strong></font>. This is the standard file format of the <code>hintsfile.gff</code> in a BRAKER working directory.</p>
<p>Example:</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">2L</span>      ProtHint        intron  <span class="hljs-number">279806</span>  <span class="hljs-number">279869</span>  <span class="hljs-number">2</span>       +       .       src=P;mult=<span class="hljs-number">25</span>;pri=<span class="hljs-number">4</span>;al_score=<span class="hljs-number">0</span>.<span class="hljs-number">437399</span>;<br><span class="hljs-attribute">2L</span>      ProtHint        intron  <span class="hljs-number">275252</span>  <span class="hljs-number">275318</span>  <span class="hljs-number">2</span>       -       .       src=P;mult=<span class="hljs-number">19</span>;pri=<span class="hljs-number">4</span>;al_score=<span class="hljs-number">0</span>.<span class="hljs-number">430006</span>;<br><span class="hljs-attribute">2L</span>      ProtHint        stop    <span class="hljs-number">293000</span>  <span class="hljs-number">293002</span>  <span class="hljs-number">1</span>       +       <span class="hljs-number">0</span>       grp=<span class="hljs-number">7220</span>_0:<span class="hljs-number">002</span>b08_g42;src=C;pri=<span class="hljs-number">4</span>;<br><span class="hljs-attribute">2L</span>      ProtHint        intron  <span class="hljs-number">207632</span>  <span class="hljs-number">207710</span>  <span class="hljs-number">1</span>       +       .       grp=<span class="hljs-number">7220</span>_0:<span class="hljs-number">002</span>afa_g26;src=C;pri=<span class="hljs-number">4</span>;<br><span class="hljs-attribute">2L</span>      ProtHint        start   <span class="hljs-number">207512</span>  <span class="hljs-number">207514</span>  <span class="hljs-number">1</span>       +       <span class="hljs-number">0</span>       grp=<span class="hljs-number">7220</span>_0:<span class="hljs-number">002</span>afa_g26;src=C;pri=<span class="hljs-number">4</span>;<br></code></pre></td></tr></table></figure>

<h4 id="3-Configuration-File"><a href="#3-Configuration-File" class="headerlink" title="3 Configuration File"></a>3 Configuration File</h4><p>The configuration file has to include three different sets of parameter:</p>
<ol>
<li><p>Weights for <font color="blue">all sources of hints</font>. The source of a hint is specified by the mandatory ‘src&#x3D;’ attribute in the last column of the <code>hintsfile.gff</code> (see section ‘Hint Files’). See section ‘Transcript scores’ in <a target="_blank" rel="noopener" href="https://doi.org/10.1101/2021.06.07.447316">TSEBRA</a> for more information on how these weigths are used.<br>A weight is set to 1, if the weight for a hint source is not specified in the configuration file.</p>
<ul>
<li><p><em>Notes on adjusting these parameters: Increase the weight of the hint sources that have the highest quality. For example, if the protein database includes only species that are remotely related to the target species, the hints produced by BRAKER2 might be less accurate than the RNA-seq evidence. Then, you should increase the weight of the source related to the RNA-seq hints.</em></p>
<figure class="highlight livecodeserver"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><code class="hljs livecodeserver"><span class="hljs-comment"># src weights</span><br>P <span class="hljs-number">0.1</span><br>E <span class="hljs-number">10</span><br>C <span class="hljs-number">5</span><br><span class="hljs-comment"># E设置10，C设置5，8，10没区别？？？</span><br>M <span class="hljs-number">1</span><br><br>E = RNA-seq hints<br>M = manual hints, these are hints that are enforced during <span class="hljs-keyword">the</span> prediction step <span class="hljs-keyword">of</span> BRAKER,   手动强制预测。<br>C = protein hints <span class="hljs-built_in">from</span> proteins <span class="hljs-keyword">with</span> <span class="hljs-keyword">a</span> <span class="hljs-string">&#x27;high&#x27;</span> spliced alignment score.<br>P = protein hints <span class="hljs-built_in">from</span> proteins that have <span class="hljs-keyword">a</span> <span class="hljs-string">&#x27;good&#x27;</span> spliced alignment score, <br>     but that is <span class="hljs-built_in">lower</span> than <span class="hljs-keyword">the</span> score <span class="hljs-built_in">from</span> <span class="hljs-keyword">the</span> ones <span class="hljs-keyword">in</span> <span class="hljs-string">&#x27;C&#x27;</span>. <br></code></pre></td></tr></table></figure></li>
</ul>
</li>
<li><p>Required fractions of <strong>supported introns</strong> or <strong>supported start&#x2F;stop-codons</strong> for a transcript. A transcript is not included in the TSEBRA result if the fractions of introns and start&#x2F;stop codons supported by extrinsic evidence are lower than the thresholds.</p>
<ul>
<li><p><em>Notes on adjusting these parameters: The low evidence support thresholds for low evidence support are quite strict in the default configuration file. In this configuration, <strong>only transcripts with very high evidence support are allowed in the TSBERA result</strong>. In some cases, the default setting might be too strict, so that too many transcripts are filtered out. In this case, you should reduce the threshold of ‘intron_support’ (e.g., to 0.2).</em></p>
</li>
<li><p>Transcript scores: Likely exon-intron borders are inferred from the alignments to create intron hints. The start and stop codons of the protein alignments are used to create start and stop codon hints, respectively. The transcript scores utilize them to quantify the support of the transcript structure.</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-comment"># Low evidence support</span><br><span class="hljs-attribute">intron_support</span> <span class="hljs-number">0</span>.<span class="hljs-number">75</span><br><span class="hljs-attribute">stasto_support</span> <span class="hljs-number">1</span><br></code></pre></td></tr></table></figure></li>
</ul>
</li>
<li><p>Allowed difference between two overlapping transcripts for the four transcript scores (see section ‘Transcript scores’ in <a target="_blank" rel="noopener" href="https://doi.org/10.1101/2021.06.07.447316">TSEBRA</a> for a definition of the different transcript scores and section ‘Pairwise transcript comparison rule’ for more information on how transcripts are compared). TSEBRA compares transcripts via their transcript scores and removes the one with the lower score if their difference exceeds the respective threshold.<br>Note that e_1, e_2 are fractions (e_1, e_2 ∊ [0,1]), and e_3, e_4 can be larger than 1 (e_3, e_4 ≥ 0).</p>
<ul>
<li><p><em>Notes on adjusting these parameters: The higher the thresholds are set the less transcripts are filtered by the respective rule. With these thresholds one can adjust the effect of each filtering rule of TSEBRA. As these thresholds are increased, more transcripts are included in the TSEBRA result, in particular, more alternatively spliced isoforms per gene are contained in the result.</em></p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-comment"># Allowed difference for each feature </span><br><span class="hljs-comment"># Values have to be in [0,1]</span><br><span class="hljs-attribute">e_1</span> <span class="hljs-number">0</span><br><span class="hljs-attribute">e_2</span> <span class="hljs-number">0</span>.<span class="hljs-number">5</span><br><span class="hljs-comment"># Values have to be &gt;0</span><br><span class="hljs-attribute">e_3</span> <span class="hljs-number">25</span><br><span class="hljs-attribute">e_4</span> <span class="hljs-number">10</span><br></code></pre></td></tr></table></figure></li>
</ul>
</li>
</ol>
<p>The name and the value of a parameter are separated by a space, and each parameter is listed in a different line.<br>Example:</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-comment"># src weights</span><br><span class="hljs-attribute">P</span> <span class="hljs-number">0</span>.<span class="hljs-number">1</span><br><span class="hljs-attribute">E</span> <span class="hljs-number">10</span><br><span class="hljs-attribute">C</span> <span class="hljs-number">5</span><br><span class="hljs-attribute">M</span> <span class="hljs-number">1</span><br><span class="hljs-comment"># Low evidence support</span><br><span class="hljs-attribute">intron_support</span> <span class="hljs-number">0</span>.<span class="hljs-number">75</span><br><span class="hljs-attribute">stasto_support</span> <span class="hljs-number">1</span><br><span class="hljs-comment"># Feature differences</span><br><span class="hljs-attribute">e_1</span> <span class="hljs-number">0</span><br><span class="hljs-attribute">e_2</span> <span class="hljs-number">0</span>.<span class="hljs-number">5</span><br><span class="hljs-attribute">e_3</span> <span class="hljs-number">25</span><br><span class="hljs-attribute">e_4</span> <span class="hljs-number">10</span><br></code></pre></td></tr></table></figure>

<p>Description of evidence sources in default BRAKER1 and BRAKER2 outputs:</p>
<figure class="highlight livecodeserver"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs livecodeserver">E = RNA-seq hints<br>M = manual hints, these are hints that are enforced during <span class="hljs-keyword">the</span> prediction step <span class="hljs-keyword">of</span> BRAKER,<br>C = protein hints <span class="hljs-built_in">from</span> proteins <span class="hljs-keyword">with</span> <span class="hljs-keyword">a</span> <span class="hljs-string">&#x27;high&#x27;</span> spliced alignment score.<br>P = protein hints <span class="hljs-built_in">from</span> proteins that have <span class="hljs-keyword">a</span> <span class="hljs-string">&#x27;good&#x27;</span> spliced alignment score, <br>     but that is <span class="hljs-built_in">lower</span> than <span class="hljs-keyword">the</span> score <span class="hljs-built_in">from</span> <span class="hljs-keyword">the</span> ones <span class="hljs-keyword">in</span> <span class="hljs-string">&#x27;C&#x27;</span>. <br></code></pre></td></tr></table></figure>

<h2 id="Use-Case"><a href="#Use-Case" class="headerlink" title="Use Case"></a>Use Case</h2><p>The recommended and most common usage for TSEBRA is to combine the resulting <code>augustus.hints.gtf</code> files of a BRAKER1 and a BRAKER2 run using the hintsfile.gff from both working directories. However, TSEBRA can be applied to any number (&gt;1) of gene predictions and hint files as long as they are in the correct format.</p>
<p>A common case might be that a user wants to annotate a novel genome with BRAKER and has:</p>
<ul>
<li>a novel genome with repeats masked: <code>genome.fasta.masked</code>,</li>
<li>hints for intron positions from RNA-seq reads<code>rna_seq_hints.gff</code>,</li>
<li>database of homologous proteins: <code>proteins.fa</code>.</li>
</ul>
<ol>
<li>Run BRAKER1 and BRAKER2 for example with</li>
</ol>
<figure class="highlight routeros"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs routeros"><span class="hljs-comment">### BRAKER1</span><br>braker.pl <span class="hljs-attribute">--genome</span>=genome.fasta.masked <span class="hljs-attribute">--hints</span>=rna_seq_hints.gff \<br>            --softmasking <span class="hljs-attribute">--species</span>=species_name <span class="hljs-attribute">--workingdir</span>=braker1_out<br><span class="hljs-comment">### BRAKER2</span><br>braker.pl <span class="hljs-attribute">--genome</span>=genome.fasta.masked <span class="hljs-attribute">--prot_seq</span>=proteins.fa \<br>    --softmasking <span class="hljs-attribute">--species</span>=species_name --epmode \<br>    <span class="hljs-attribute">--workingdir</span>=braker2_out<br></code></pre></td></tr></table></figure>

<ol>
<li>Combine predicitons with TSEBRA</li>
</ol>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs stylus">./bin/tsebra<span class="hljs-selector-class">.py</span> -g braker1_out/augustus<span class="hljs-selector-class">.hints</span><span class="hljs-selector-class">.gtf</span>,braker2_out/augustus<span class="hljs-selector-class">.hints</span><span class="hljs-selector-class">.gtf</span> -c default<span class="hljs-selector-class">.cfg</span> \<br>    -e braker1_out/hintsfile<span class="hljs-selector-class">.gff</span>,braker2_out/hintsfile<span class="hljs-selector-class">.gff</span> \<br>    -o braker1+<span class="hljs-number">2</span>_combined.gtf<br></code></pre></td></tr></table></figure>

<p>The combined gene prediciton is <code>braker1+2_combined.gtf</code>.</p>
<h2 id="Example"><a href="#Example" class="headerlink" title="Example"></a>Example</h2><p>A small example is located at <code>example/</code>. Run <code>./example/run_prevco_example.sh</code> to execute the example and to check if TSEBRA runs properly.</p>
<h2 id="Other-scripts-in-the-TSEBRA-repository"><a href="#Other-scripts-in-the-TSEBRA-repository" class="headerlink" title="Other scripts in the TSEBRA repository"></a>Other scripts in the TSEBRA repository</h2><h3 id="Renaming-transcripts-from-a-TSEBRA-output"><a href="#Renaming-transcripts-from-a-TSEBRA-output" class="headerlink" title="Renaming transcripts from a TSEBRA output"></a>Renaming transcripts from a TSEBRA output</h3><p>The IDs of the transcripts and genes in the TSEBRA output can be renamed such that the gene and transcript ID match. Genes and transcript are numbered consecutively and for example, the second transcript of gene “g12” has the ID “g12.t2”. If a prefix is set then it will be added before all IDs, for example, the transcript ID is “dmel_g12.t2” if the prefix is set to “dmel”. Additionally, a translation table can be produced that provides the mapping from old to new transcript IDs.</p>
<p>Example for renaming <code>tsebra_result.gtf</code>:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">./bin/rename_gtf<span class="hljs-selector-class">.py</span> <span class="hljs-attr">--gtf</span> tsebra_result<span class="hljs-selector-class">.gtf</span> <span class="hljs-attr">--prefix</span> dmel <span class="hljs-attr">--translation_tab</span> translation<span class="hljs-selector-class">.tab</span> <span class="hljs-attr">--out</span> tsebra_result_renamed.gtf<br></code></pre></td></tr></table></figure>

<p>The arguments <code>--prefix</code> and <code>--translation_tab</code> are optional.</p>
<h3 id="Fixing-the-formatting-issue-of-braker-gtf"><a href="#Fixing-the-formatting-issue-of-braker-gtf" class="headerlink" title="Fixing the formatting issue of braker.gtf"></a>Fixing the formatting issue of <code>braker.gtf</code></h3><p>A BRAKER run produces a second complete gene set named <code>braker.gtf</code>, besides the official output <code>augustus.hints.gtf</code>. The <code>braker.gtf</code> is the result of merging <code>augustus.hints.gtf</code> with some ‘high-confidents’ genes from the GeneMark prediction. However, the merging process leads to a formatting issue in <code>braker.gtf</code>. A quick fix for this formatting issue is the script <code>fix_gtf_ids.py</code>, e.g.:</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">./bin/fix_gtf_ids<span class="hljs-selector-class">.py</span> <span class="hljs-attr">--gtf</span> braker_out/braker<span class="hljs-selector-class">.gtf</span> <span class="hljs-attr">--out</span> braker1_fixed.gtf<br></code></pre></td></tr></table></figure>

<p>Take note that the <code>braker.gtf</code> and <code>fix_gtf_ids.py</code> haven’t been tested sufficently and there is no guarantee that this gene set is superior to <code>augustus.hints.gtf</code>.</p>
<h2 id="-1"><a href="#-1" class="headerlink" title=""></a></h2><h1 id="braker问题"><a href="#braker问题" class="headerlink" title="braker问题"></a>braker问题</h1><p>预测结果文件braker.gtf中，转录本和基因ID，前面可能会自动加上file_1_file_1_，如g4417.t1变为file_1_file_1_g4417.t1，导致生成的gff（或者你用其他软件，如augustus转换脚本gtf2gff.pl，实际上你在参数中指定–gff3，流程调用的也是agustus&#x2F;gtf2gff.pl）转化得到的gff3文件中没有基因特征。所以，如果你用的结果是braker.gtf，含有这个问题，必须人为去掉。augustus.hints.*则是正常，目前没发现这个问题。<br><a target="_blank" rel="noopener" href="https://www.biostars.org/p/9464353/">https://www.biostars.org/p/9464353/</a></p>
<p>第二个比较关心的问题是，braker流程出来一堆结果，我到底该用哪个？虽然官网有一些解释，但总有些不能理解。<br>几个比较关键的结果： augustus.hints.* 是AUGUSTUS最终蛋白hints结果。<br>而braker.gtf&#x2F;gff3是 AUGUSTUS和GeneMark-EP+预测（braker流程中的蛋白预测）的并集，因此该结果是高敏感性低特异性（更多基因被预测以及更多假阳性）。<br>总体来说，二者结果是相近的，如果侧重于敏感性，则用braker.gtf结果。否则用augustus。（个人建议还是用augustus的结果，相当于二次预测）<br><a target="_blank" rel="noopener" href="https://github.com/Gaius-Augustus/BRAKER/issues/194">https://github.com/Gaius-Augustus/BRAKER/issues/194</a></p>

              
            </div>
            <hr/>
            <div>
              <div class="post-metas my-3">
  
    <div class="post-meta mr-3 d-flex align-items-center">
      <i class="iconfont icon-category"></i>
      

<span class="category-chains">
  
  
    
      <span class="category-chain">
        
  <a href="/GeekFocus/categories/Software/" class="category-chain-item">Software</a>
  
  

      </span>
    
  
</span>

    </div>
  
  
    <div class="post-meta">
      <i class="iconfont icon-tags"></i>
      
        <a href="/GeekFocus/tags/annotation/">#annotation</a>
      
        <a href="/GeekFocus/tags/Braker2/">#Braker2</a>
      
    </div>
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>BRAKER2</div>
      <div>https://wangc1106.github.io/GeekFocus/2022/04/21/2022-04-28-BRAKER/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>Author</div>
          <div>banner</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>Posted on</div>
          <div>April 21, 2022</div>
        </div>
      
      
      <div class="license-meta-item">
        <div>Licensed under</div>
        <div>
          
            
            
              <a target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
              <span class="hint--top hint--rounded" aria-label="BY - Attribution">
                <i class="iconfont icon-by"></i>
              </span>
              </a>
            
          
        </div>
      </div>
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/GeekFocus/2022/05/11/2022-05-11-hisat2/" title="RNA-seq【hisat2】">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">RNA-seq【hisat2】</span>
                        <span class="visible-mobile">Previous</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/GeekFocus/2022/04/10/2022-04-10-MAKER-anno/" title="MAKER">
                        <span class="hidden-mobile">MAKER</span>
                        <span class="visible-mobile">Next</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="margin-left: -1rem">
    <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;Table of Contents</p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>
  </div>
</div>





  



  



  



  



  






    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">Search</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">Keyword</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <span>banner</span> <i class="iconfont icon-love"></i> <span>2021-2022©</span> 
    </div>
  
  
    <div class="statistics">
  
  

  
    
      <span id="leancloud-site-pv-container" style="display: none">
        总访问量 
        <span id="leancloud-site-pv"></span>
         次
      </span>
    
    
      <span id="leancloud-site-uv-container" style="display: none">
        总访客数 
        <span id="leancloud-site-uv"></span>
         人
      </span>
    
    

  
</div>

  
  
  
</div>

    <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
    </div>
  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.0/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/GeekFocus/js/events.js" ></script>
<script  src="/GeekFocus/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>






  
    <script  src="/GeekFocus/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.18.2/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      headingSelector : CONFIG.toc.headingSelector || 'h1,h2,h3,h4,h5,h6',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      collapseDepth   : CONFIG.toc.collapseDepth || 0,
      scrollSmooth    : true,
      headingsOffset  : -boardTop
    });
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }
  });
</script>


  <script>
  (function() {
    var enableLang = CONFIG.code_language.enable && CONFIG.code_language.default;
    var enableCopy = CONFIG.copy_btn;
    if (!enableLang && !enableCopy) {
      return;
    }

    function getBgClass(ele) {
      return Fluid.utils.getBackgroundLightness(ele) >= 0 ? 'code-widget-light' : 'code-widget-dark';
    }

    var copyTmpl = '';
    copyTmpl += '<div class="code-widget">';
    copyTmpl += 'LANG';
    copyTmpl += '</div>';
    jQuery('.markdown-body pre').each(function() {
      var $pre = jQuery(this);
      if ($pre.find('code.mermaid').length > 0) {
        return;
      }
      if ($pre.find('span.line').length > 0) {
        return;
      }

      var lang = '';

      if (enableLang) {
        lang = CONFIG.code_language.default;
        if ($pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2 && $pre.children().hasClass('hljs')) {
          lang = $pre[0].children[0].classList[1];
        } else if ($pre[0].getAttribute('data-language')) {
          lang = $pre[0].getAttribute('data-language');
        } else if ($pre.parent().hasClass('sourceCode') && $pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2) {
          lang = $pre[0].children[0].classList[1];
          $pre.parent().addClass('code-wrapper');
        } else if ($pre.parent().hasClass('markdown-body') && $pre[0].classList.length === 0) {
          $pre.wrap('<div class="code-wrapper"></div>');
        }
        lang = lang.toUpperCase().replace('NONE', CONFIG.code_language.default);
      }
      $pre.append(copyTmpl.replace('LANG', lang).replace('code-widget">',
        getBgClass($pre[0]) + (enableCopy ? ' code-widget copy-btn" data-clipboard-snippet><i class="iconfont icon-copy"></i>' : ' code-widget">')));

      if (enableCopy) {
        Fluid.utils.createScript('https://lib.baomitu.com/clipboard.js/2.0.10/clipboard.min.js', function() {
          var clipboard = new window.ClipboardJS('.copy-btn', {
            target: function(trigger) {
              var nodes = trigger.parentNode.childNodes;
              for (var i = 0; i < nodes.length; i++) {
                if (nodes[i].tagName === 'CODE') {
                  return nodes[i];
                }
              }
            }
          });
          clipboard.on('success', function(e) {
            e.clearSelection();
            e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-copy', 'icon-success');
            setTimeout(function() {
              e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-success', 'icon-copy');
            }, 2000);
          });
        });
      }
    });
  })();
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/GeekFocus/js/local-search.js" ></script>

  <script defer src="/GeekFocus/js/leancloud.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/GeekFocus/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">Blog works best with JavaScript enabled</div>
  </noscript>
</body>
</html>
