

<!DOCTYPE html>
<html lang="en" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/GeekFocus/img/fluid.png">
  <link rel="icon" href="/GeekFocus/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="banner">
  <meta name="keywords" content="Bioinformatics">
  
    <meta name="description" content="Juicer is a platform for analyzing kilobase resolution Hi-C data. In this distribution, we include the pipeline for generating Hi-C maps from fastq raw data files and command line tools for feature an">
<meta property="og:type" content="article">
<meta property="og:title" content="【Hi-C】【Juicer】">
<meta property="og:url" content="https://wangc1106.github.io/GeekFocus/2022/03/16/2022-03-16-juicer/index.html">
<meta property="og:site_name" content="DrBanner">
<meta property="og:description" content="Juicer is a platform for analyzing kilobase resolution Hi-C data. In this distribution, we include the pipeline for generating Hi-C maps from fastq raw data files and command line tools for feature an">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-03-16-juicer/1.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-03-16-juicer/1.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-03-16-juicer/2.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-03-16-juicer/2.png">
<meta property="article:published_time" content="2022-03-16T14:06:04.197Z">
<meta property="article:modified_time" content="2022-06-08T06:31:17.019Z">
<meta property="article:author" content="banner">
<meta property="article:tag" content="assembly">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="https://wangc1106.github.io/GeekFocus/2022-03-16-juicer/1.png">
  
  
  
  <title>【Hi-C】【Juicer】 - DrBanner</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/GeekFocus/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/GeekFocus/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/GeekFocus/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/GeekFocus/","version":"1.9.1","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"right","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":false,"follow_dnt":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":"https://wangc1106.github.io/GeekFocus","path":"window.location.pathname","ignore_local":false}},"search_path":"/GeekFocus/local-search.xml"};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/GeekFocus/js/utils.js" ></script>
  <script  src="/GeekFocus/js/color-schema.js" ></script>

  
<meta name="generator" content="Hexo 6.2.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 70vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/GeekFocus/">
      <strong>DrBanner</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/">
                <i class="iconfont icon-home-fill"></i>
                Home
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/archives/">
                <i class="iconfont icon-archive-fill"></i>
                Archives
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/categories/">
                <i class="iconfont icon-category-fill"></i>
                Categories
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/tags/">
                <i class="iconfont icon-tags-fill"></i>
                Tags
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/about/">
                <i class="iconfont icon-user-fill"></i>
                About
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              &nbsp;<i class="iconfont icon-search"></i>&nbsp;
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/GeekFocus/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="【Hi-C】【Juicer】"></span>
          
        </div>

        
          
  <div class="mt-3">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-author" aria-hidden="true"></i>
        banner
      </span>
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2022-03-16 22:06" pubdate>
          March 16, 2022 pm
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          12k words
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          104 mins
        
      </span>
    

    
    
      
        <span id="leancloud-page-views-container" class="post-meta" style="display: none">
          <i class="iconfont icon-eye" aria-hidden="true"></i>
          <span id="leancloud-page-views"></span> views
        </span>
        
      
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">【Hi-C】【Juicer】</h1>
            
              <p class="note note-info">
                
                  
                    Last updated on June 8, 2022 pm
                  
                
              </p>
            
            <div class="markdown-body">
              
              <p>Juicer is a platform for analyzing kilobase resolution Hi-C data. In this distribution, we include the pipeline for <strong>generating Hi-C maps from fastq raw data</strong> files and <strong>command line tools for feature annotation on the Hi-C maps</strong>.</p>
<span id="more"></span>


<p><a target="_blank" rel="noopener" href="https://github.com/aidenlab/juicer">https://github.com/aidenlab/juicer</a></p>
<h1 id="Read-this-first"><a href="#Read-this-first" class="headerlink" title="Read this first!!"></a>Read this first!!</h1><p>To access Juicer 1.6 (last <strong>stable</strong> release), please see <a target="_blank" rel="noopener" href="https://github.com/aidenlab/juicer/releases/tag/1.6">the Github Release</a>. If you clone the Juicer repo directly from Github, it will clone Juicer 2, which is under active development. If you encounter any bugs, please let us know.</p>
<h1 id="About-Juicer"><a href="#About-Juicer" class="headerlink" title="About Juicer"></a>About Juicer</h1><p>Juicer is a platform for analyzing kilobase resolution Hi-C data. In this distribution, we include the pipeline for <strong>generating Hi-C maps from fastq raw data files</strong> and <strong>command line tools for feature annotation on the Hi-C maps</strong>.</p>
<p><strong>If you use Juicer in your research, please cite: Neva C. Durand, Muhammad S. Shamim, Ido Machol, Suhas S. P. Rao, Miriam H. Huntley, Eric S. Lander, and Erez Lieberman Aiden. “Juicer provides a one-click system for analyzing loop-resolution Hi-C experiments.” Cell Systems 3(1), 2016.</strong></p>
<h1 id="Documentation"><a href="#Documentation" class="headerlink" title="Documentation"></a>Documentation</h1><p>Please see <a target="_blank" rel="noopener" href="https://github.com/aidenlab/juicer/wiki">the wiki</a> for extensive documentation.</p>
<h1 id="Questions"><a href="#Questions" class="headerlink" title="Questions?"></a>Questions?</h1><p>For FAQs, or for asking new questions, please see our forum: <a target="_blank" rel="noopener" href="http://aidenlab.org/forum.html">aidenlab.org&#x2F;forum.html</a>.</p>
<hr>
<h2 id="Distribution"><a href="#Distribution" class="headerlink" title="Distribution"></a>Distribution</h2><p>In this repository, we include the scripts for running Juicer on AWS, LSF, Univa Grid Engine, SLURM, and a single CPU</p>
<p>&#x2F;AWS - scripts for running pipeline and postprocessing on AWS</p>
<p>&#x2F;UGER - scripts for running pipeline and postprocessing on UGER (Univa)</p>
<p>&#x2F;SLURM - scripts for running pipeline and postprocessing on SLURM</p>
<p>&#x2F;LSF - scripts for running pipeline and postprocessing on LSF <strong>BETA</strong></p>
<p>&#x2F;CPU - scripts for running pipeline and postprocessing on a single CPU <strong>BETA</strong></p>
<p>&#x2F;misc - miscellaneous helpful scripts</p>
<hr>
<h2 id="Hardware-and-Software-Requirements"><a href="#Hardware-and-Software-Requirements" class="headerlink" title="Hardware and Software Requirements"></a>Hardware and Software Requirements</h2><p>Juicer is a pipeline optimized for parallel computation on a cluster. Juicer consists of two parts: <strong>the pipeline that creates Hi-C files from raw data</strong>, and <strong>the post-processing command line tools</strong>.</p>
<h3 id="Cluster-requirements"><a href="#Cluster-requirements" class="headerlink" title="Cluster requirements:"></a>Cluster requirements:</h3><p>Juicer requires the use of a cluster, with <strong>ideally &gt;&#x3D; 4 cores</strong> (min 1 core) and <strong>&gt;&#x3D; 64 GB RAM</strong> (min 16 GB RAM)</p>
<p>Juicer currently works with the following resource management software:</p>
<ul>
<li><a target="_blank" rel="noopener" href="http://www.openlava.org/">OpenLava</a></li>
<li><a target="_blank" rel="noopener" href="https://www.ibm.com/systems/spectrum-computing/products/lsf">LSF</a></li>
<li><a target="_blank" rel="noopener" href="https://slurm.schedmd.com/download.html">SLURM</a></li>
<li>GridEngine (Univa, etc. any flavor)</li>
</ul>
<h3 id="Juicer-tools-requirements"><a href="#Juicer-tools-requirements" class="headerlink" title="Juicer tools requirements"></a>Juicer tools requirements</h3><p>The minimum software requirement to run Juicer is a working Java installation (version &gt;&#x3D; 1.8) on Windows, Linux, and Mac OSX. We recommend using the latest Java version available, but please do not use the Java Beta Version. Minimum system requirements for running Java can be found at<a target="_blank" rel="noopener" href="https://java.com/en/download/help/sysreq.xml">https://java.com/en/download/help/sysreq.xml</a></p>
<p>To download and install the latest Java Runtime Environment (JRE), please go to <a target="_blank" rel="noopener" href="https://www.java.com/download">https://www.java.com/download</a></p>
<h3 id="GNU-CoreUtils"><a href="#GNU-CoreUtils" class="headerlink" title="GNU CoreUtils"></a>GNU CoreUtils</h3><p>The latest version of GNU coreutils can be downloaded from <a target="_blank" rel="noopener" href="https://www.gnu.org/software/coreutils/manual/">https://www.gnu.org/software/coreutils/manual/</a></p>
<h3 id="Burrows-Wheeler-Aligner-BWA"><a href="#Burrows-Wheeler-Aligner-BWA" class="headerlink" title="Burrows-Wheeler Aligner (BWA)"></a>Burrows-Wheeler Aligner (BWA)</h3><p>The <strong>latest version of BWA</strong> should be installed from <a target="_blank" rel="noopener" href="http://bio-bwa.sourceforge.net/">http://bio-bwa.sourceforge.net/</a></p>
<h3 id="CUDA-for-HiCCUPS-peak-calling"><a href="#CUDA-for-HiCCUPS-peak-calling" class="headerlink" title="CUDA (for HiCCUPS peak calling)"></a>CUDA (for HiCCUPS peak calling)</h3><p>You must have an NVIDIA GPU to install CUDA.</p>
<p>Instructions for installing the latest version of CUDA can be found on the <a target="_blank" rel="noopener" href="https://developer.nvidia.com/cuda-downloads">NVIDIA Developer site</a>.</p>
<p>The native libraries included with Juicer are compiled for CUDA 7 or CUDA 7.5. See the <a target="_blank" rel="noopener" href="https://github.com/theaidenlab/juicer/wiki/Download">download page for Juicer Tools</a>.</p>
<p>Other versions of CUDA can be used, but you will need to download the respective native libraries from <a target="_blank" rel="noopener" href="http://www.jcuda.org/downloads/downloads.html">JCuda</a>.</p>
<p>For best performance, use a dedicated GPU. You may also be able to obtain access to GPU clusters through Amazon Web Services or a local research institution.</p>
<p>If you cannot access a GPU, you can run the <a target="_blank" rel="noopener" href="https://github.com/aidenlab/juicer/wiki/CPU-HiCCUPS">CPU version of HiCCUPS</a> directly using the <code>.hic</code> file and Juicer Tools.</p>
<h3 id="Building-new-jars"><a href="#Building-new-jars" class="headerlink" title="Building new jars"></a>Building new jars</h3><p>See the Juicebox documentation at <a target="_blank" rel="noopener" href="https://github.com/theaidenlab/Juicebox">https://github.com/theaidenlab/Juicebox</a> for details on building new jars of the juicer_tools.</p>
<hr>
<h2 id="Quick-Start"><a href="#Quick-Start" class="headerlink" title="Quick Start"></a><font color="red">Quick Start</font></h2><p>Run the Juicer pipeline on your cluster of choice with “juicer.sh [options]”</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br></pre></td><td class="code"><pre><code class="hljs sh">Usage: juicer.sh [-g genomeID] [-d topDir] [-q queue] [-l long queue] [-s site]<br>                 [-a about] [-R end] [-S stage] [-p chrom.sizes path]<br>                 [-y restriction site file] [-z reference genome file]<br>                 [-C chunk size] [-D Juicer scripts directory]<br>                 [-Q queue time <span class="hljs-built_in">limit</span>] [-L long queue time <span class="hljs-built_in">limit</span>] [-e] [-h] [-x]<br>* [-g genomeID] must be defined <span class="hljs-keyword">in</span> the script, e.g. <span class="hljs-string">&quot;hg19&quot;</span> or <span class="hljs-string">&quot;mm10&quot;</span> (default<br>  <span class="hljs-string">&quot;hg19&quot;</span>); alternatively, it can be defined using the -z <span class="hljs-built_in">command</span><br>* [<span class="hljs-string">&#x27;-d topDir&#x27;</span>] is the top level directory (default<br>  <span class="hljs-string">&quot;/Users/nchernia/Downloads/neva-muck/UGER&quot;</span>)<br>     [topDir]/fastq must contain the fastq files<br>     [topDir]/splits will be created to contain the temporary <span class="hljs-built_in">split</span> files<br>     [topDir]/aligned will be created <span class="hljs-keyword">for</span> the final alignment<br>* [-q queue] is the queue <span class="hljs-keyword">for</span> running alignments (default <span class="hljs-string">&quot;short&quot;</span>)<br>* [-l long queue] is the queue <span class="hljs-keyword">for</span> running longer <span class="hljs-built_in">jobs</span> such as the hic file<br>  creation (default <span class="hljs-string">&quot;long&quot;</span>)<br>* [site] must be defined <span class="hljs-keyword">in</span> the script, e.g.  <span class="hljs-string">&quot;HindIII&quot;</span> or <span class="hljs-string">&quot;MboI&quot;</span><br>  (default <span class="hljs-string">&quot;none&quot;</span>)<br>* [about]: enter description of experiment, enclosed <span class="hljs-keyword">in</span> single quotes<br>* [stage]: must be one of <span class="hljs-string">&quot;chimeric&quot;</span>, <span class="hljs-string">&quot;merge&quot;</span>, <span class="hljs-string">&quot;dedup&quot;</span>, <span class="hljs-string">&quot;final&quot;</span>, <span class="hljs-string">&quot;postproc&quot;</span>, or <span class="hljs-string">&quot;early&quot;</span>.<br>    -Use <span class="hljs-string">&quot;chimeric&quot;</span> when alignments are <span class="hljs-keyword">done</span> but chimeric handling has not finished<br>    -Use <span class="hljs-string">&quot;merge&quot;</span> when alignment has finished but the merged_sort file has not<br>     yet been created.<br>    -Use <span class="hljs-string">&quot;dedup&quot;</span> when the files have been merged into merged_sort but<br>     merged_nodups has not yet been created.<br>    -Use <span class="hljs-string">&quot;final&quot;</span> when the reads have been deduped into merged_nodups but the<br>     final stats and hic files have not yet been created.<br>    -Use <span class="hljs-string">&quot;postproc&quot;</span> when the hic files have been created and only<br>     postprocessing feature annotation remains to be completed.<br>    -Use <span class="hljs-string">&quot;early&quot;</span> <span class="hljs-keyword">for</span> an early <span class="hljs-built_in">exit</span>, before the final creation of the stats and<br>     hic files<br>* [<span class="hljs-string">&#x27;chrom.sizes path&#x27;</span>]: enter path <span class="hljs-keyword">for</span> chrom.sizes file<br>* [<span class="hljs-string">&#x27;restriction site file&#x27;</span>]: enter path <span class="hljs-keyword">for</span> restriction site file (locations of<br>  restriction sites <span class="hljs-keyword">in</span> genome; can be generated with the script<br>  (misc/generate_site_positions.py) )<br>* [<span class="hljs-string">&#x27;reference genome file&#x27;</span>]: enter path <span class="hljs-keyword">for</span> reference sequence file, BWA index<br>  files must be <span class="hljs-keyword">in</span> same directory<br>* [chunk size]: number of lines <span class="hljs-keyword">in</span> <span class="hljs-built_in">split</span> files, must be multiple of 4<br>  (default 90000000, <span class="hljs-built_in">which</span> equals 22.5 million reads)<br>* [<span class="hljs-string">&#x27;Juicer scripts directory&#x27;</span>]: <span class="hljs-built_in">set</span> the Juicer directory,<br>  <span class="hljs-built_in">which</span> should have scripts/ references/ and restriction_sites/ underneath it<br>  (default /broad/aidenlab)<br>* [queue time <span class="hljs-built_in">limit</span>]: time <span class="hljs-built_in">limit</span> <span class="hljs-keyword">for</span> queue, i.e. -W 12:00 is 12 hours<br>  (default 1200)<br>* [long queue time <span class="hljs-built_in">limit</span>]: time <span class="hljs-built_in">limit</span> <span class="hljs-keyword">for</span> long queue, i.e. -W 168:00 is one week<br>  (default 3600)<br>* -f: include fragment-delimited maps from hic file creation<br>* -e: early <span class="hljs-built_in">exit</span><br>* -h: <span class="hljs-built_in">print</span> this <span class="hljs-built_in">help</span> and <span class="hljs-built_in">exit</span><br></code></pre></td></tr></table></figure>

<hr>
<h2 id="Juicer-Usage"><a href="#Juicer-Usage" class="headerlink" title="Juicer Usage"></a>Juicer Usage</h2><ul>
<li><strong>Running Juicer with no arguments</strong> will run it with genomeID hg19 and site MboI</li>
<li><strong>Providing a genome ID</strong>: if not defined in the script, you can either directly modify the script or provide the script with the files needed. You would provide the script with the files needed via “-z reference_sequence_path” (needs to have the BWA index files in same directory), “-p chrom_sizes_path” (these are the chromosomes you want included in .hic file), and “-s site_file” (this is the listing of all the restriction site locations, one line per chromosome). Note that ligation junction won’t be defined in this case. The script (misc&#x2F;generate_site_positions.py) can help you generate the file</li>
<li><strong>Providing a restriction enzyme</strong>: if not defined in the script, you can either directly modify the script or provide the files needed via the “-s site_file” flag, as above. Alternatively, if you don’t want to do any fragment-level analysis (as with a DNAse experiment), you should assign the site “none”, as in <code>juicer.sh -s none</code></li>
<li><strong>Directory structure</strong>: Juicer expects the fastq files to be stored in a directory underneath the top-level directory. E.g. HIC001&#x2F;fastq. By default, the top-level directory is the directory where you are when you launch Juicer; you can change this via the -d flag. Fastqs can be zipped. [topDir]&#x2F;splits will be created to contain the temporary split files and should be deleted once your run is completed. [topDir]&#x2F;aligned will be created for the final files, including the hic files, the statistics, the valid pairs (merged_nodups), the collisions, and the feature annotations.</li>
<li><strong>Queues</strong> are complicated and it’s likely that you’ll have to modify the script for your system, though we did our best to avoid this. By default there’s a short queue and a long queue. We also allow you to pass in wait times for those queues; this is currently ignored by the UGER and SLURM versions. The short queue should be able to complete alignment of one split file. The long queue is for jobs that we expect to take a while, like writing out the merged_sort file</li>
<li><strong>Chunk size</strong> is intimitely associated with your queues; a smaller chunk size means more alignment jobs that complete in a faster time. If you have a hard limit on the number of jobs, you don’t want too small of a chunk size. If your short queue has a very limited runtime ceiling, you don’t want too big of a chunk size. Run time for alignment will also depend on the particulars of your cluster. We launch ~5 jobs per chunk. Chunk size must be a multiple of 4.</li>
<li><strong>Relaunch</strong> via the same script. Type <code>juicer.sh [options] -S stage</code> where “stage” is one of merge, dedup, final, postproc, or early. “merge” is for when alignment has finished but merged_sort hasn’t been created; “dedup” is for when merged_sort is there but not merged_nodups (this will relaunch all dedup jobs); “final” is for when merged_nodups is there and you want the stats and hic files; “postproc” is for when you have the hic files and just want feature annotations; and “early” is for early exit, before hic file creation. If your jobs failed at the alignment stage, run <code>relaunch_prep.sh</code> and then run juicer.sh.</li>
<li><strong>Miscelleaneous options</strong> include -a ‘experiment description’, which will add the experiment description to the statistics file and the meta data in the hic file; -r, which allows you to use bwa aln instead of bwa mem, useful for shorter reads; -R [end], in case you have one read end that’s short and one that’s long and you want to align the short end with bwa aln and the long end with bwa mem; and -D [Juicer scripts directory], to set an alternative Juicer directory; must have scripts&#x2F;, references&#x2F;, and restriction_sites&#x2F; underneath it</li>
</ul>
<hr>
<h2 id="Command-Line-Tools-Usage"><a href="#Command-Line-Tools-Usage" class="headerlink" title="Command Line Tools Usage"></a>Command Line Tools Usage</h2><p>Detailed documentation about the command line tools can be found on the wiki:</p>
<ul>
<li><a target="_blank" rel="noopener" href="https://github.com/aidenlab/juicer/wiki/Feature-Annotation">Annotating features with Arrowhead, HiCCUPS, MotifFinder, APA, Eigenvector, and Pearsons</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/aidenlab/juicer/wiki/Pre">Creating .hic with Pre</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/aidenlab/straw">Extracting data from .hic files with straw</a></li>
</ul>
<p>To launch the command line tools, use the shell script “juicer_tools” on Unix&#x2F;MacOS or type</p>
<figure class="highlight mipsasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs mipsasm"><span class="hljs-keyword">java </span>-<span class="hljs-keyword">jar </span><span class="hljs-keyword">juicer_tools.jar </span>(command...) [flags...] &lt;parameters...&gt;`<br></code></pre></td></tr></table></figure>

<p>In the command line tools, there are several analysis functions:</p>
<ol>
<li><code>apa</code> for conducting aggregate peak analysis</li>
<li><code>hiccups</code> for annotating loops</li>
<li><code>motifs</code> for finding CTCF motifs</li>
<li><code>arrowhead</code> for annotating contact domains</li>
<li><code>eigenvector</code> for calculating the eigenvector (first PC) of the Pearson’s</li>
<li><code>pearsons</code> for calculating the Pearson’s</li>
</ol>
<p>The <code>juicer_tools</code> (Unix&#x2F;MacOS) script can be used in place of the unwieldy <code>java -Djava.library.path=path/to/natives/ -jar juicer_tools.jar</code></p>
<hr>
<h1 id="juicer的工作目录结构"><a href="#juicer的工作目录结构" class="headerlink" title="juicer的工作目录结构"></a>juicer的工作目录结构</h1><p>要求固定目录结构，新建<code>juicer</code>目录，该目录为软件安装目录，目录下必须有4个子目录</p>
<p><code>references</code>目录存放参考基因组文件，<code>work</code>存放样本序列文件和分析结果，<code>scripts</code>存放软件运行脚本，<code>restriction_sites</code>存放参考基因组酶切图谱。</p>
<h1 id="下载源码"><a href="#下载源码" class="headerlink" title="下载源码"></a>下载源码</h1><p>从github上下载juicer和jcuda的源代码，放置到<code>scripts</code>目录。</p>
<p>其中的<code>CPU</code>目录就是单机服务器，而<code>AWS</code>, <code>LSF</code>, <code>PBS</code>等对应公有云和不同的集群系统。</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">cd</span> scripts/common<br><span class="hljs-attribute">wget</span> https://hicfiles.tc4ga.com/public/juicer/juicer_tools.<span class="hljs-number">1</span>.<span class="hljs-number">9</span>.<span class="hljs-number">9</span>_jcuda.<span class="hljs-number">0</span>.<span class="hljs-number">8</span>.jar<br><span class="hljs-attribute">ln</span> -s juicer_tools.<span class="hljs-number">1</span>.<span class="hljs-number">9</span>.<span class="hljs-number">9</span>_jcuda.<span class="hljs-number">0</span>.<span class="hljs-number">8</span>.jar  juicer_tools.jar<br></code></pre></td></tr></table></figure>

<h1 id="参考基因组，酶切图谱"><a href="#参考基因组，酶切图谱" class="headerlink" title="参考基因组，酶切图谱"></a>参考基因组，酶切图谱</h1><p>fasta序列文件和bwa 索引</p>
<p>juicer&#x2F;misc&#x2F;generate_site_positions.py可以输出4种内切酶的酶切图谱。HindIII；DpnII；MboI；Sau3AI</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs awk">generate_site_positions.py HindIII hg19 hg19.fasta<br>python ..<span class="hljs-regexp">/juicer/mi</span>sc<span class="hljs-regexp">/generate_site_positions.py  MboI  hg38_MboI ../</span>references/Homo_sapiens_assembly38.fasta <span class="hljs-comment"># 生成了 hg38_MboI.txt 文件</span><br>awk <span class="hljs-string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;print $1, $NF&#125;&#x27;</span> hg38_MboI.txt &gt; hg38.chrom.sizes<br></code></pre></td></tr></table></figure>

<p>第一个参数内切酶名称，第二个参数自定义基因组版本，第三个参数基因组fasta文件路径，输出文件名称为第二个参数和第一个参数用下划线链接，后缀为<code>txt</code></p>
<h1 id="序列"><a href="#序列" class="headerlink" title="序列"></a>序列</h1><p>软件运行时对样本文件的存放位置也有要求，必须位于<code>work</code>目录下，以样本名作为一个子目录，序列文件存放于<code>fastq</code>目录下，示意如下</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs awk"><span class="hljs-regexp">/opt/</span>juicer<span class="hljs-regexp">/work/</span>MBR19/fastq<br><span class="hljs-regexp">/opt/</span>juicer<span class="hljs-regexp">/work/</span>MBR19<span class="hljs-regexp">/fastq/</span>chr19_R1.fastq.gz<br><span class="hljs-regexp">/opt/</span>juicer<span class="hljs-regexp">/work/</span>MBR19<span class="hljs-regexp">/fastq/</span>chr19_R2.fastq.gz<br></code></pre></td></tr></table></figure>

<hr>
<h1 id="软件比较"><a href="#软件比较" class="headerlink" title="软件比较"></a>软件比较</h1><p><img src="/GeekFocus/2022-03-16-juicer/1.png" srcset="/GeekFocus/img/loading.gif" lazyload><br><img src="/GeekFocus/img/2022-03-16-juicer/1.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<h1 id="脚本原理（juicer-sh）"><a href="#脚本原理（juicer-sh）" class="headerlink" title="脚本原理（juicer.sh）"></a>脚本原理（juicer.sh）</h1><p>fastq等文件输入 —》bwa比对 —》排序 —》合并 —》去PCR重复 —》生成 hic文件</p>
<p>Juicer的工作流程见下图，输入<strong>原始</strong>fastq文件，处理得到<strong>中间</strong>文件.hic, 之后对.hic文件用于<strong>下游</strong>分析，包括</p>
<p>Arrowhead: 寻找存在关联的区域<br>HiCCUPS: 分析局部富集peaks<br>MotifFinder: 用于锚定peaks<br>Persons: 计算观测&#x2F;期望的皮尔森相关系数矩阵<br>Eigenvector: 确定分隔</p>
<p><img src="/GeekFocus/2022-03-16-juicer/2.png" srcset="/GeekFocus/img/loading.gif" lazyload><br><img src="/GeekFocus/img/2022-03-16-juicer/2.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="juicer工作流程"></p>
<h1 id="结果文件"><a href="#结果文件" class="headerlink" title="结果文件"></a>结果文件</h1><p>结果文件都在 aligned 中，主要文件是 .hic 文件，其中inter_30.hic 是设置了 mapq threshold &gt;30 的结果。如果 GPU （cudd）可用，软件会自动使用自带 <strong>HiCCUPS</strong> 算法计算 contact domains 并保存在 inter_30_contact_domains 文件夹。如果GPU不可用，可手动用自带的 <strong>CPU HiCCUPS</strong> 算法来计算，后面章节有介绍。</p>
<p>hic文件格式请看 <a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=https://github.com/theaidenlab/juicebox/blob/master/HiC_format_v8.docx">https://github.com/theaidenlab/juicebox/blob/master/HiC_format_v8.docx</a></p>
<h1 id="示例代码"><a href="#示例代码" class="headerlink" title="示例代码"></a>示例代码</h1><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><code class="hljs sh">juicer.sh \<br>-z references/hg19.fa \<br>-p restriction_sites/hg19.chrom.sizes \<br>-y restriction_sites/hg19_HindIII.txt \<br>-d /home/pub/software/juicer/work/HIC003/ \<br>-D /home/pub/software/juicer \<br>-t 5<br><br>-d: juicer的目录 安装在/soft/，设置为/soft/juicer<br>-D: juicer scripts的目录，安装在/soft/，设置为/soft/juicer/CPU<br></code></pre></td></tr></table></figure>

<p>需要注意, 指定文件路径时，最好指定绝对路径，特别是fastq文件路径。因为软件运行过程中会使用软链接，相对路径会出错。</p>
<p><code>splits</code>目录下存放中间结果，由于hi-C数据量很大，所以会将原始序列<strong>拆分成多份</strong>，<strong>并行运算加快速度</strong>。默认每份包含22.5M的reads, 可以通过<code>-C</code>参数调整，该参数指定拆分文件的行数，默认是90000000?， 注意fastq文件4行代表一条序列，所以这个参数的值必须是4的倍数。拆分后序列的R1和R2端分别通过bwa比对基因组，然后合并，筛选嵌合体序列，去重复，生成预处理后的结果文件。</p>
<p><code>aligned</code>目录下存放的是最终结果，包含了可以导入juicebox的后缀为<code>hic</code>的图谱文件, <code>inter.hic</code>和<code>inter_30.hic</code>， 30表示通过<code>MAPQ &gt; 30</code>过滤之后的结果。完整流程还会进行后续处理，包括识别TAD, 染色质环等结构。其中识别染色质环的HICCUPs算法必须通过<a target="_blank" rel="noopener" href="https://cloud.tencent.com/product/gpu?from=10680">GPU</a>加速运行才可以，所以没有安装GPU卡的普通服务器无法运行这个步骤。</p>
<p>上述过程可看到，juicer使用简单。由于Hi-C数据测序量非常大，以及后续分析算法的复杂度，对服务器计算资源的要求相当高，必须高性能服务器才能满足要求，而该软件所需的GPU卡成本也非常高，一块的成本在2万元左右，这些因素一定程度制约了Hi-C的普及和发展。</p>
<h1 id="juicer下游分析"><a href="#juicer下游分析" class="headerlink" title="juicer下游分析"></a>juicer下游分析</h1><p><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/341206245">https://zhuanlan.zhihu.com/p/341206245</a></p>
<p>目前针对Hi-C数据的研究主要是三个方面，分别是<code>A/B comparment</code> ，<code>TADS</code>，<code>Loops</code>。</p>
<p><code>juicer_tools.jar</code> 功能介绍</p>
<p><code>arrowhead</code> 注释TAD</p>
<p><code>hiccups</code> 注释loop</p>
<p><code>motigs</code> 定位CTCF元件</p>
<p><code>hiccupsdiff</code> 从多个loos文件中找到不同的loop</p>
<p><code>apa</code> 聚合峰的分析</p>
<p><code>pearsons</code> 计算O&#x2F;E的皮尔森相关系数</p>
<p><code>eigenvector</code> 计算特征向量的皮尔森相关系数</p>
<p><code>dump</code> .hic文件互作矩阵提取</p>
<p><code>pre</code> 非juicer数据转.hic文件</p>
<h1 id="hic数据可视化"><a href="#hic数据可视化" class="headerlink" title="hic数据可视化"></a>hic数据可视化</h1><ul>
<li><p><strong>加载.hic文件</strong></p>
</li>
<li><p><strong>加载一维注释</strong></p>
</li>
<li><p><strong>加载二维注释，黄的的是TAD，天蓝色的是loop</strong></p>
</li>
</ul>

              
            </div>
            <hr/>
            <div>
              <div class="post-metas my-3">
  
    <div class="post-meta mr-3 d-flex align-items-center">
      <i class="iconfont icon-category"></i>
      

<span class="category-chains">
  
  
    
      <span class="category-chain">
        
  <a href="/GeekFocus/categories/Software/" class="category-chain-item">Software</a>
  
  

      </span>
    
  
</span>

    </div>
  
  
    <div class="post-meta">
      <i class="iconfont icon-tags"></i>
      
        <a href="/GeekFocus/tags/assembly/">#assembly</a>
      
    </div>
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>【Hi-C】【Juicer】</div>
      <div>https://wangc1106.github.io/GeekFocus/2022/03/16/2022-03-16-juicer/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>Author</div>
          <div>banner</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>Posted on</div>
          <div>March 16, 2022</div>
        </div>
      
      
      <div class="license-meta-item">
        <div>Licensed under</div>
        <div>
          
            
            
              <a target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
              <span class="hint--top hint--rounded" aria-label="BY - Attribution">
                <i class="iconfont icon-by"></i>
              </span>
              </a>
            
          
        </div>
      </div>
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/GeekFocus/2022/03/17/2022-03-17-3D-DNA/" title="【Hi-C】【3D-DNA】">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">【Hi-C】【3D-DNA】</span>
                        <span class="visible-mobile">Previous</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/GeekFocus/2022/03/16/2022-03-16-minimap2/" title="【long reads mapping】【minimap2】">
                        <span class="hidden-mobile">【long reads mapping】【minimap2】</span>
                        <span class="visible-mobile">Next</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="margin-left: -1rem">
    <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;Table of Contents</p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>
  </div>
</div>





  



  



  



  



  






    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">Search</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">Keyword</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <span>banner</span> <i class="iconfont icon-love"></i> <span>2021-2023©</span> 
    </div>
  
  
    <div class="statistics">
  
  

  
    
      <span id="leancloud-site-pv-container" style="display: none">
        总访问量 
        <span id="leancloud-site-pv"></span>
         次
      </span>
    
    
      <span id="leancloud-site-uv-container" style="display: none">
        总访客数 
        <span id="leancloud-site-uv"></span>
         人
      </span>
    
    

  
</div>

  
  
  
</div>

    <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
    </div>
  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.0/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/GeekFocus/js/events.js" ></script>
<script  src="/GeekFocus/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>






  
    <script  src="/GeekFocus/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.18.2/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      headingSelector : CONFIG.toc.headingSelector || 'h1,h2,h3,h4,h5,h6',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      collapseDepth   : CONFIG.toc.collapseDepth || 0,
      scrollSmooth    : true,
      headingsOffset  : -boardTop
    });
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }
  });
</script>


  <script>
  (function() {
    var enableLang = CONFIG.code_language.enable && CONFIG.code_language.default;
    var enableCopy = CONFIG.copy_btn;
    if (!enableLang && !enableCopy) {
      return;
    }

    function getBgClass(ele) {
      return Fluid.utils.getBackgroundLightness(ele) >= 0 ? 'code-widget-light' : 'code-widget-dark';
    }

    var copyTmpl = '';
    copyTmpl += '<div class="code-widget">';
    copyTmpl += 'LANG';
    copyTmpl += '</div>';
    jQuery('.markdown-body pre').each(function() {
      var $pre = jQuery(this);
      if ($pre.find('code.mermaid').length > 0) {
        return;
      }
      if ($pre.find('span.line').length > 0) {
        return;
      }

      var lang = '';

      if (enableLang) {
        lang = CONFIG.code_language.default;
        if ($pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2 && $pre.children().hasClass('hljs')) {
          lang = $pre[0].children[0].classList[1];
        } else if ($pre[0].getAttribute('data-language')) {
          lang = $pre[0].getAttribute('data-language');
        } else if ($pre.parent().hasClass('sourceCode') && $pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2) {
          lang = $pre[0].children[0].classList[1];
          $pre.parent().addClass('code-wrapper');
        } else if ($pre.parent().hasClass('markdown-body') && $pre[0].classList.length === 0) {
          $pre.wrap('<div class="code-wrapper"></div>');
        }
        lang = lang.toUpperCase().replace('NONE', CONFIG.code_language.default);
      }
      $pre.append(copyTmpl.replace('LANG', lang).replace('code-widget">',
        getBgClass($pre[0]) + (enableCopy ? ' code-widget copy-btn" data-clipboard-snippet><i class="iconfont icon-copy"></i>' : ' code-widget">')));

      if (enableCopy) {
        Fluid.utils.createScript('https://lib.baomitu.com/clipboard.js/2.0.10/clipboard.min.js', function() {
          var clipboard = new window.ClipboardJS('.copy-btn', {
            target: function(trigger) {
              var nodes = trigger.parentNode.childNodes;
              for (var i = 0; i < nodes.length; i++) {
                if (nodes[i].tagName === 'CODE') {
                  return nodes[i];
                }
              }
            }
          });
          clipboard.on('success', function(e) {
            e.clearSelection();
            e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-copy', 'icon-success');
            setTimeout(function() {
              e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-success', 'icon-copy');
            }, 2000);
          });
        });
      }
    });
  })();
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/GeekFocus/js/local-search.js" ></script>

  <script defer src="/GeekFocus/js/leancloud.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/GeekFocus/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">Blog works best with JavaScript enabled</div>
  </noscript>
</body>
</html>
