

<!DOCTYPE html>
<html lang="en" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/GeekFocus/img/fluid.png">
  <link rel="icon" href="/GeekFocus/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="banner">
  <meta name="keywords" content="Bioinformatics">
  
    <meta name="description" content="Genome annotation with Maker">
<meta property="og:type" content="article">
<meta property="og:title" content="MAKER">
<meta property="og:url" content="https://wangc1106.github.io/GeekFocus/2022/04/10/2022-04-10-MAKER-anno/index.html">
<meta property="og:site_name" content="DrBanner">
<meta property="og:description" content="Genome annotation with Maker">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2022-04-10T08:37:30.052Z">
<meta property="article:modified_time" content="2022-06-05T12:18:34.007Z">
<meta property="article:author" content="banner">
<meta property="article:tag" content="annotation">
<meta property="article:tag" content="Maker2">
<meta name="twitter:card" content="summary_large_image">
  
  
  
  <title>MAKER - DrBanner</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/GeekFocus/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/GeekFocus/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/GeekFocus/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/GeekFocus/","version":"1.9.1","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"right","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":false,"follow_dnt":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":"https://wangc1106.github.io/GeekFocus","path":"window.location.pathname","ignore_local":false}},"search_path":"/GeekFocus/local-search.xml"};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/GeekFocus/js/utils.js" ></script>
  <script  src="/GeekFocus/js/color-schema.js" ></script>

  
<meta name="generator" content="Hexo 6.2.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 70vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/GeekFocus/">
      <strong>DrBanner</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/">
                <i class="iconfont icon-home-fill"></i>
                Home
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/archives/">
                <i class="iconfont icon-archive-fill"></i>
                Archives
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/categories/">
                <i class="iconfont icon-category-fill"></i>
                Categories
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/tags/">
                <i class="iconfont icon-tags-fill"></i>
                Tags
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/about/">
                <i class="iconfont icon-user-fill"></i>
                About
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              &nbsp;<i class="iconfont icon-search"></i>&nbsp;
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/GeekFocus/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="MAKER"></span>
          
        </div>

        
          
  <div class="mt-3">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-author" aria-hidden="true"></i>
        banner
      </span>
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2022-04-10 16:37" pubdate>
          April 10, 2022 pm
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          13k words
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          107 mins
        
      </span>
    

    
    
      
        <span id="leancloud-page-views-container" class="post-meta" style="display: none">
          <i class="iconfont icon-eye" aria-hidden="true"></i>
          <span id="leancloud-page-views"></span> views
        </span>
        
      
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">MAKER</h1>
            
              <p class="note note-info">
                
                  
                    Last updated on June 5, 2022 pm
                  
                
              </p>
            
            <div class="markdown-body">
              
              <h1 id="Genome-annotation-with-Maker"><a href="#Genome-annotation-with-Maker" class="headerlink" title="Genome annotation with Maker"></a><a target="_blank" rel="noopener" href="https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">Genome annotation with Maker</a></h1><span id="more"></span>

<p>Maker is able to annotate both prokaryotes and eukaryotes. It works by <strong>aligning as many evidences as possible along the genome sequence</strong>, and then reconciliating all these signals to determine probable gene structures.</p>
<p>The evidences can be <strong>transcript or protein sequences</strong> from the <font color="red"><strong>same (or closely related) organism</strong></font>. These sequences can come from <strong>public databases</strong> (like NR or GenBank) or from your <strong>own experimental data</strong> (transcriptome assembly from an RNASeq experiment for example). Maker is also able to <strong>take into account repeated elements</strong>.</p>
<p>Maker uses <strong>ab-initio predictors</strong> (like <a target="_blank" rel="noopener" href="http://bioinf.uni-greifswald.de/augustus/">Augustus</a> or <a target="_blank" rel="noopener" href="https://github.com/KorfLab/SNAP">SNAP</a>) to improve its predictions: these software tools are able to make gene structure predictions by analysing only the genome sequence with a <strong>statistical model</strong>.</p>
<p>In this tutorial you will learn <strong>how to perform a genome annotation</strong>, and <strong>how to evaluate its quality</strong>. You will see how training ab-initio predictors is an important step to produce good results. Finally, you will learn how to use the <a target="_blank" rel="noopener" href="http://jbrowse.org/">JBrowse</a> genome browser to visualise the results.</p>
<p>More information about Maker can be found <a target="_blank" rel="noopener" href="http://www.yandell-lab.org/software/maker.html">here</a>.</p>
<p>This tutorial was inspired by the MAKER Tutorial for <a target="_blank" rel="noopener" href="http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/MAKER_Tutorial_for_WGS_Assembly_and_Annotation_Winter_School_2018">WGS Assembly and Annotation Winter School 2018</a>, don’t hesitate to consult it for more information on Maker, and on how to run it with command line.</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment">#install</span><br>不是maker/src/bin/maker<br>注意maker/bin/maker<br>不用conda perl [系统perl]<br><br>tar xf maker-2.31.10.tgz<br><span class="hljs-built_in">cd</span> maker/src<br>perl Build.PL   检查依赖的库<br>./Build installdeps<br>./Build installexes<br>./Build install<br>./Build status<br><span class="hljs-built_in">mkdir</span> test01<br><span class="hljs-built_in">cp</span> data/dpp* test01<br><span class="hljs-built_in">cd</span> test01<br>maker -CPL<br><span class="hljs-comment">#MPICC 多线程，所用的computer都需要？？？网络冲突 【已安装</span><br>https://github.com/bioconda/bioconda-recipes/issues/16501<br>ERROR: Could not determine <span class="hljs-keyword">if</span> RepBase is installed<br>！！MASKER checks xxx/RepeatMasker/Libraries/RepeatMaskerLib.embl <span class="hljs-keyword">for</span> Repbase.<br>If you have older versions of Repbase like RMRBSeqs.embl, create symlink <span class="hljs-keyword">in</span> xxx/RepeatMasker/Libraries/. For example, <span class="hljs-built_in">ln</span> -s RMRBSeqs.embl RepeatMaskerLib.embl. But <span class="hljs-keyword">in</span> that <span class="hljs-keyword">case</span>, model_org= should be <span class="hljs-built_in">set</span> to the org that exists <span class="hljs-keyword">in</span> the database, instead of all. CAUTION: I don<span class="hljs-string">&#x27;t know whether the result is reliable enough in this way.</span><br><span class="hljs-string"></span><br><span class="hljs-string">Can&#x27;</span>t locate List/MoreUtils.pm <span class="hljs-keyword">in</span> @INC (@INC contains: <br>conda <span class="hljs-built_in">env</span> install ， no conda <span class="hljs-built_in">env</span> no error？<br>conda install perl-list-moreutils [<span class="hljs-built_in">link</span>](https://www.biostars.org/p/278249/)<br></code></pre></td></tr></table></figure>



<h2 id="需要的输入文件"><a href="#需要的输入文件" class="headerlink" title="需要的输入文件"></a>需要的输入文件</h2><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><code class="hljs sh">-1. 【genome.fasta】<br>    基因组序列文件。<br>-2. 【inchworm.fasta】<br>    转录组组装结果文件。此文件内容代表转录子序列或 EST 序列。为了增加其准确性，选用了 trinity 中 inchworm 的结果文件，而不是转录组最终的组装结果文件。<br>-3. 【proteins.fasta】<br>    临近物种的蛋白质序列。可以从 NCBI 的 Taxonomy 数据库中下载。<br>-4. 【consensi.fa.classified】<br>    适合本物种的重复序列数据库。该文件是一个fasta文件，其序列代表着本物种的高重复序列。该文件使用 RepeatModeler 制作。<br>-5. 【Augustus hmm files [species]】<br>    适合本物种的 Augustus 的 HMM 文件。该文件以文件夹形式存放在 /opt/biosoft/augustus-3.0.3/config/species/ 目录下。<br>    将 genome.fasta 和 inchworm.fasta 输入 PASA，得到用于 trainning 的基因<br>    然后使用 Augustus 进行 training，得到 HMM 文件。<br>-6. 【Snap hmm file [species.hmm]】<br>    适合本物种的 SNAP 的 HMM 文件。该文件是一个单独的文件。将 genome.fasta 和 inchworm.fasta 输入 PASA，得到用于 trainning 的基因，然后使用 SNAP 进行 training，得到 HMM 文件。<br>-7. 【Genemark_es hmm file [es.mod]】<br>    适合本物种的 Genemark_es 的 HMM 文件。将 genome.fasta 输入 genemark_es 软件进行自我训练得到的 HMM 文件。<br>-8. 【rRNA.fa】<br>    rRNA 的序列文件。该文件使用 RNAmmer 对基因组进行分析得到。<br></code></pre></td></tr></table></figure>

<p>使用 MAKER 命令得到初始化的 3 个配置文件。</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs stylus">$ maker -CTL<br><br>得到<span class="hljs-number">3</span>个配置文件： maker_exe<span class="hljs-selector-class">.ctl</span>, maker_bopts<span class="hljs-selector-class">.ctl</span> 和 maker_opts.ctl。<br>maker_exe.ctl： 配置 Maker 需要调用的程序的路径；maker_exe<span class="hljs-selector-class">.ctl</span>: 执行程序的路径<br>maker_bopts.ctl： 配置 blast 的各种阈值；maker_bopt<span class="hljs-selector-class">.ctl</span>: BLAST和Exonerat的过滤参数<br>maker_opts.ctl： MAKER 的主要配置文件，配置输入文件的路径，以及 MAKER 的使用流程。<br></code></pre></td></tr></table></figure>

<h2 id="注释"><a href="#注释" class="headerlink" title="注释"></a>注释</h2><p>先修改 maker_opts.ctl 配置文件，需要修改的内容如下：</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><code class="hljs sh">genome=genome.fasta            <span class="hljs-comment"># 基因组序列文件</span><br>est=inchworm.fasta             <span class="hljs-comment"># EST 序列文件</span><br>protein=proteins.fasta         <span class="hljs-comment"># 临近物种的蛋白质序列文件</span><br>model_org=fungi                <span class="hljs-comment"># 选择repebase数据库的物种，对真菌则用fungi，也可以使用 all</span><br>rmlib=consensi.fa.classified   <span class="hljs-comment"># 适合本物种的重复序列数据库</span><br>snaphmm=species.hmm            <span class="hljs-comment"># snap traing 的 HMM 文件</span><br>gmhmm=es.mod                   <span class="hljs-comment"># genemark_es 的 HMM 文件</span><br>augustus_species=species       <span class="hljs-comment"># augustus 的 HMM 文件</span><br>est2genome=1                   <span class="hljs-comment"># 使用 EST 序列进行基因预测</span><br>protein2genome=1               <span class="hljs-comment"># 使用蛋白质序列进行基因预测</span><br>trna=1                         <span class="hljs-comment"># 使用 tRNAscan 进行 tRNA 预测</span><br>snoscan_rrna=rRNA.fa           <span class="hljs-comment"># 【使用 Snoscan 利用 rRNA 序列进行 C/D box 类型的 snoRNAs 预测，不过我加入了这个信息后，maker运行不成功，不知道为什么。】</span><br>correct_est_fusion=1           <span class="hljs-comment"># 进行融合 EST 的校正</span><br></code></pre></td></tr></table></figure>

<p>并行化运行 maker。基因组序列比较大，MAKER 的计算量大，并行化能极大节约时间？？？</p>
<p>计算节点有mpicc吗</p>
<figure class="highlight crystal"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs crystal">mpd &amp;<br><span class="hljs-variable">$ </span>maker                  <span class="hljs-comment"># 不并行化运行</span><br><span class="hljs-variable">$ </span>mpiexec -n <span class="hljs-number">20</span> maker    <span class="hljs-comment"># 使用20个线程并行计算</span><br></code></pre></td></tr></table></figure>



<figure class="highlight ini"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs ini"><span class="hljs-comment">#change</span><br><span class="hljs-attr">genome</span>=dpp_contig.fasta<br><span class="hljs-attr">est</span>=dpp_est.fasta<br><span class="hljs-attr">protein</span>=dpp_protein.fasta<br><span class="hljs-attr">est2genome</span>=<span class="hljs-number">1</span><br></code></pre></td></tr></table></figure>

<p>当前路径运行程序</p>
<p>输出结果见”dpp_contig.maker.output”, 重点是”dpp_contig_master_datastore_index.log”文件，由于maker会拆分数据集并行计算，因此该文件记录总体的运行情况，需要关注其中是否有<font color="red">”FAILED”,”RETRY”,”SKIPPED_SAMLL”,”DIED_SIPPED_PERMANET”</font>，因为这意味着有些数据出于某些原因没有运算。</p>
<p>最后，我们需要将并行运算的结果进行整合，导出GFF文件, 转录本序列和蛋白序列</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs awk">~<span class="hljs-regexp">/opt/</span>biosoft<span class="hljs-regexp">/maker/</span>bin/fasta_merge -d dpp_contig_master_datastore_index.log<br>~<span class="hljs-regexp">/opt/</span>biosoft<span class="hljs-regexp">/maker/</span>bin/gff3_merge -d dpp_contig_master_datastore_index.log<br></code></pre></td></tr></table></figure>

<p>在该目录下就会出现,<font color="red"> “dpp_contig.all.gff”,“dpp_contig.all.maker.proteins.fasta”,”dpp_contig.all.maker.transcripts.fasta”</font></p>
<p>其中GFF文件就需要用IGV，JBrowse, Apollo下展示来检查下注释是否正确。</p>
<hr>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/fe14cab264a7">https://www.jianshu.com/p/fe14cab264a7</a></p>
<p><a target="_blank" rel="noopener" href="http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/The_MAKER_control_files_explained">http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/The_MAKER_control_files_explained</a></p>
<h1 id="MAKER配置文件详解"><a href="#MAKER配置文件详解" class="headerlink" title="MAKER配置文件详解"></a>MAKER配置文件详解</h1><p>本文翻译自<a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/The_MAKER_control_files_explained">http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/The_MAKER_control_files_explained</a></p>
<p>MAKER会生成三个配置文件，如下</p>
<ul>
<li>maker_opts.ctl: 控制MAKER分析行为的主要配置文件</li>
<li>maker_bopts.ctl: BLAST和Exnerate的过滤阈值</li>
<li>maker_exe.ctl: MAKER运行过程中所依赖软件的路径</li>
</ul>
<p>这里主要介绍<code>maker_opts.ctl</code>, 其余配置文件直接看配置文件里面的注释信息即可。</p>
<p><font color="blue">A minimal input file set for MAKER would generally consist of a F<strong>ASTA file for the genomic</strong> sequence, a <strong>FASTA file of RNA</strong> (ESTs&#x2F;cDNA&#x2F;mRNA transcripts) from <strong>【the organism】</strong>, and a <strong>FASTA file of protein</strong> sequences from <strong>【the same or related organisms】</strong> (or a general protein database).</font></p>
<h3 id="基因组"><a href="#基因组" class="headerlink" title="基因组"></a>基因组</h3><p>用于设置被注释的基因组序列的位置和物种类型，包括<code>genome</code>和<code>organisam_type</code>两项</p>
<ul>
<li>genome: FASTA序列路径</li>
<li>organism_type: eukaryotic,prokaryotic二选一</li>
</ul>
<p>需要注意的是，基因组序列的N50需要超过预期基因长度的中位数，否则注释效果不好。另外最好保证基因组序列只包括A,T,C,G,N, 对于其他类型兼并碱基可以都改成N.</p>
<h3 id="使用MAKER得到GFF3进行重注释"><a href="#使用MAKER得到GFF3进行重注释" class="headerlink" title="使用MAKER得到GFF3进行重注释"></a>使用MAKER得到GFF3进行重注释</h3><p>这一项基本上我们用不上，它是在当你把MAKER的中间输出文件都删除了，仅保留了输出的GFF3文件时，你可以用之前相同的输入设置重新运行流程得到相同的输出</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-comment">#-----Re-annotation Using MAKER Derived GFF3</span><br>maker_gff= <span class="hljs-comment">#MAKER derived GFF3 file</span><br>est_pass=0 <span class="hljs-comment">#use ESTs in maker_gff: 1 = yes, 0 = no</span><br>altest_pass=0 <span class="hljs-comment">#use alternate organism ESTs in maker_gff: 1 = yes, 0 = no</span><br>protein_pass=0 <span class="hljs-comment">#use protein alignments in maker_gff: 1 = yes, 0 = no</span><br>rm_pass=0 <span class="hljs-comment">#use repeats in maker_gff: 1 = yes, 0 = no</span><br>model_pass=0 <span class="hljs-comment">#use gene models in maker_gff: 1 = yes, 0 = no</span><br>pred_pass=0 <span class="hljs-comment">#use ab-initio predictions in maker_gff: 1 = yes, 0 = no</span><br>other_pass=0 <span class="hljs-comment">#passthrough anyything else in maker_gff: 1 = yes, 0 = no</span><br></code></pre></td></tr></table></figure>

<h3 id="EST-x2F-转录本证据"><a href="#EST-x2F-转录本证据" class="headerlink" title="EST&#x2F;转录本证据"></a>EST&#x2F;转录本证据</h3><p>出于历史原因, MAKER还是用EST代表之前<font color="blue">EST数据</font>和目前的<font color="blue">转录组数据</font>。 此处不只是使用EST数据，而是可以使用<font color="blue">组装的mRNA-seq</font>, 组装的<font color="blue">全长cDNA</font>。 预期他们能够正确的组装，并联配到正确的剪切位点(对于FASTA格式，MAKER使用<code>exonerate</code>找到剪切位点）。用途如下:</p>
<ul>
<li><font color="red">直接<strong>推断</strong></font>基因模型(est2genome)</li>
<li>作为预测结果的<strong>支持证据</strong></li>
<li>修改结果和<strong>增加UTR</strong></li>
<li>鉴定<strong>可变转录本</strong></li>
<li>在某些情况下，这些数据和其他证据能帮助MAKER推断<strong>基因边界</strong></li>
<li>在预测步骤中辅助基因预测工具推断<strong>剪切位点</strong></li>
</ul>
<p>设置项如下:</p>
<ul>
<li><code>est</code>: EST, 全长cDNA, 组装的mRNA序列，可以通过<font color="blue">逗号分隔多个文件</font>。</li>
<li><code>altest</code>: 如果真的没有当前物种的转录组数据，也可以使用<font color="blue">同源物种序列</font>。这些序列会通过tblstx进行比对，会消耗大量的运算时间。</li>
<li><code>est_gff</code>: 预比对的转录本，以<strong>GFF3</strong>格式存放，通常来自于<font color="blue">CuffLinks&#x2F;StringTie的组装结果gtf</font>，或者是上一步的MAKER注释</li>
<li><code>altest_gff</code>: 和altest一样，只不过是比对后以GFF3存放，基本上也用不到</li>
</ul>
<p>Tophat2 + cufflinks 有参转录组组装得到<font color="red"><strong>gtf结果，转化为gff</strong></font>，用于estgff。<font color="red"><strong>转化fa</strong></font>用于est。</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs sh">cufflinks-gffread<br>gffread transcripts.gtf -g reference.fasta -w transcripts.fasta<br>gffread hp1.est.gtf -o- &gt; hp1.est.gff3<br></code></pre></td></tr></table></figure>







<h3 id="同源蛋白证据"><a href="#同源蛋白证据" class="headerlink" title="同源蛋白证据"></a>同源蛋白证据</h3><p>和之前的转录组数据类似，用途如下</p>
<ul>
<li>直接推断基因模型(protein2genome), 仅在它们能够正确的<font color="blue">联配到剪切位点附件</font>。</li>
<li>作为预测结果的<font color="blue">支持证据</font>(MAKER会检查CDS，保证基因预测结果和蛋白联配是<font color="blue">相同的阅读框</font>)</li>
<li>某些情况下，用于推断<font color="blue">基因边界</font></li>
<li>在预测步骤中，<font color="blue">使用从蛋白推断的ORF辅助从头预测软件</font></li>
</ul>
<p><strong>建议</strong>使用 <font color="blue">uniprot&#x2F;swiss-prot 或 RefSeq</font>上的NP数据，因为经过人工审查，可信度较高。<font color="blue">不建议用UniProt&#x2F;tremble或者Genbank上的数据</font>，这些数据的可信度较低。你可以挑选几个同源物种的高可信度蛋白。或者使用MAKER注释的其他物种AED小于0.5的转录本产物。</p>
<p>由于许多注释里包含一些<font color="blue">死亡转座子(dead transposons)或伪基因(pseudogenes)</font>，<font color="blue">因此<strong>不建议</strong>使用临近物种的所有注释蛋白</font>。想象一个比较糟糕的情况，如果有邻近物种的死亡转座子，当构建重复序列屏蔽文库时，发现其中一个条目和该序列匹配。 于是你假设这是一个真实的基因，于是你从屏蔽文库中删除了该条目。当注释基因组的时候，该基因变成了注释集中的一整个基因组家族，但这其实是糟糕的证据和重复序列屏蔽所导致的后果。</p>
<p>需要设置的就两项：</p>
<ul>
<li><code>protein</code>: 以FASTA存放的蛋白序列位置</li>
<li><code>protein_gff</code>: 以GFF3格式记录的蛋白序列预先比对结果，通常来自于之前MAKER输出</li>
</ul>
<h3 id="重复序列重复屏蔽"><a href="#重复序列重复屏蔽" class="headerlink" title="重复序列重复屏蔽"></a>重复序列重复屏蔽</h3><p>通过屏蔽重复序列来避免EST和蛋白比对到重复区域，防止基因预测算法在这些区域预测外显子。由于许多重复序列会编码真实的蛋白(例如反转座子等)，基因预测工具和比对工具会被他们所迷惑(会在一个基因中错误的加上外显子)</p>
<ul>
<li><code>model_org</code>: 默认是all, 可以设置成RepeatMasker中<font color="blue">RepBase数据库的其中一个</font></li>
<li><code>rmlib</code>: <font color="blue">自己构建的重复序列文库</font></li>
<li><code>repeat_protein</code>: 转座因子的蛋白序列</li>
<li><code>rm_gff</code>: 以GFF3格式记录的重复序列位置信息</li>
<li><code>prok_rm</code>: 不需要修改，因为原核生物不需要考虑重复序列</li>
<li><code>softmask</code>: 不需要修改</li>
</ul>
<h3 id="从头基因预测"><a href="#从头基因预测" class="headerlink" title="从头基因预测"></a>从头基因预测</h3><p>如果你需要从MAKER以外获取基因模型，则需要在这一节添加相应的配置。根据可信度高低，MAKER会对这些基因模型采取不同的行为。</p>
<ul>
<li>通过软件预测的基因结构可信度低，它们不会影响证据簇(evidence cluster). MAKER会保留预测结果，或者根据EST证据调整外显子，如果有证据支持，那么他们会保留在最终的注释集中。如果有多个注释结果，MAKER会对其进行比较，从中挑选出最优结果。</li>
<li><code>model_gff</code>提供的基因模型的可信度最高，会影响证据簇。在一些基因边界判定中，MAKER在证据簇的影响下，更倾向于保留之前的基因模型而非替换。它们也会保留名字。MAKER不会修改模型，要么删除要么保留。最后，即便没有证据支持，MAKER还是会保留他们，而不会删除他们，只不过最终的AED会设置为1.</li>
<li>snaphmm: SNAP的HMM文件路径，允许多个输入，以逗号分隔</li>
<li>gmhmm: GeneMark的HMM文件文件路径，允许多个输入，以逗号分隔</li>
<li>augustus_species: Augustus的基因模型命名, 不容易训练，但是效果很好</li>
<li>fgenesh_par_file: FGENESH的HMM参数文件，收费工具，基本上用不到</li>
<li>pred_gff: 其他预测工具的输出结果，以GFF3格式保存</li>
<li>model_gff: 最高可信度的GFF输入</li>
<li>run_evm&#x3D;0: 是否让MAKER运行EVM，速度会变慢</li>
<li>est2genome: 让MAKER根据EST推测基因模型</li>
<li>protein2genome: 让MAKER根据蛋白序列推测基因模型</li>
<li>trna: 使用tRNAscan分析tRNA #find tRNAs with tRNAscan, 1 &#x3D; yes, 0 &#x3D; no</li>
<li>snoscan_rrna: Snoscan分析snoRNA所需的rRNA文件&#x3D; #rRNA file to have Snoscan find snoRNAs</li>
<li>snoscan_meth：Snoscan分析snoRNA所需的O-methylation site 文件</li>
<li>unmask: 是否在屏蔽序列中进行基因预测，默认是0</li>
<li>allow_overlap: 允许的基因重叠比例，从0到1，空白表示使用默认值</li>
</ul>
<h3 id="其他类型的注释"><a href="#其他类型的注释" class="headerlink" title="其他类型的注释"></a>其他类型的注释</h3><p>这一项功能很简单，就是提供一个GFF文件，在MAKER运行结束后增加里面的信息</p>
<ul>
<li><code>other_gff</code>: 其他类型注释的GFF文件路径</li>
</ul>
<h3 id="外部程序选项"><a href="#外部程序选项" class="headerlink" title="外部程序选项"></a>外部程序选项</h3><p>这里的两个参数用于影响外部程序，即BLAST的行为</p>
<ul>
<li><code>alt_peptide</code>: 对于非标准氨基酸的替换方法，默认是C(cysteine)</li>
<li><code>cpus</code>: BLAST的线程数，如果使用MPI，该值可以设置的小一些，默认是1.</li>
</ul>
<h3 id="MAKER行为选项"><a href="#MAKER行为选项" class="headerlink" title="MAKER行为选项"></a>MAKER行为选项</h3><p>这里的选项用于调整MAKER的行为，使其符合你的基因组特性</p>
<ul>
<li><code>mad_dna_len</code>至少要3倍于预期的最大内含子长度。在内存足够的情况下，对于<strong>脊椎动物</strong>可以考虑设置为<code>300000</code>，植物一般没有那么大的内含子</li>
<li><code>min_contig</code>: 低于该值的contig会被过滤掉，建议设置为10k.</li>
<li><code>pred_flank</code>: 在基因预测时，将证据簇在两端进行扩展，默认是200 bp. 对于比较紧凑的基因组，降低该值能够避免基因错误合并。对于比较稀疏的基因组，提高该值可以避免外显子缺失。</li>
<li><code>pred_stats</code>: 默认是0，只计算MAKER预测基因的AED和QI值，设置为1则计算所有从头预测的基因结构。</li>
<li><code>AED_threashold</code>: 根据AED(0-1)值来过滤输出的基因，默认是1，表示保留所有预测结果。</li>
<li><code>min_protein</code>: 一些时候，基因预测工具会生成许多短预测结果，而由于一些证据类型（例如mRNA-seq）存在噪音，导致这些预测结果看起来有证据支持，于是保留在最终的输出结果中(AED&gt;1). 通过限制预测蛋白的氨基酸数(amino acides, AA)，可以减少预测结果。</li>
<li><code>alt_splice</code>: 是否计算可变转录本</li>
<li><code>always_complete</code>: 这个是MAKER开发者在合作者的要求下加上的参数，用来确保基因模型始终有起始密码子和终止密码子，默认是0</li>
<li><code>map_forward</code>: 用于保留老版本的GFF文件的信息，映射到新的版本GFF中。</li>
<li><code>keep_preds</code>: 设置为1时表示保留所有的预测结果，默认是0.</li>
<li><code>split_hit</code>: 新版本的MAKER(&gt;2.28)不需要考虑该项。因为之前版本拆分后的contig是互不重叠，于是就有可能有外显子被刚好被拆成两端，设置该项可以保留该信息。</li>
<li><code>single_exon</code>: 如果一个EST只有只有单个外显子，默认情况下MAKER并不会把它当做支持基因模型的证据, 除非还有同源蛋白作为支持。单外显子EST和组装的mRNA-seq转录本通常是RNA制备过程中的基因组序列污染。关闭时会降低MAKER的敏感度(sensitivity), 但是当你打开它的时候，命中的特异性会比整体的准确度(accuracy)差得多。</li>
<li><code>single_length</code>: 在启用<code>single_exon</code>时，设置该项用于保留一些比较小的序列，但即便注释了也可能不是有功能的蛋白。</li>
<li><code>correct_est_fusion</code>用来避免因为UTR的重叠导致将基因模型的错误合并，在<strong>真菌基因组</strong>中比较常见。它会检查基因模型的5’ UTR长度是否超过基因长度的一半，如果是的话，那么MAKER会在起始密码位置打断基因，然后在5’UTR区重新预测基因。</li>
<li><code>tries</code>: 尝试次数，默认是2</li>
<li><code>clean_try</code>: 重新尝试时，是否删除之前的文件，默认是0，也就是不删除。建议设置为1</li>
<li><code>clean_up</code>: 删除分析过程中的文件，默认不删除</li>
<li><code>TMP</code>: 临时文件的目录，默认存放在<code>/tmp</code>下，建议设置一个容量比较大的目录</li>
</ul>
<hr>
<h1 id="maker-内含软件安装"><a href="#maker-内含软件安装" class="headerlink" title="maker 内含软件安装"></a>maker 内含软件安装</h1><h2 id="snoscan"><a href="#snoscan" class="headerlink" title="snoscan"></a>snoscan</h2><p><a target="_blank" rel="noopener" href="http://www.chenlianfu.com/?p=2325">http://www.chenlianfu.com/?p=2325</a></p>
<p><a target="_blank" rel="noopener" href="http://lowelab.ucsc.edu/snoscan/">http://lowelab.ucsc.edu/snoscan/</a></p>
<p><a target="_blank" rel="noopener" href="https://iamphioxus.org/2016/01/20/installing-trnascan-se-and-snoscan/">https://iamphioxus.org/2016/01/20/installing-trnascan-se-and-snoscan/</a></p>
<p>snoRNAs 主要有 2 种不同类型。其中 C&#x2F;D box 类型 snoRNAs 用于指导甲基化修饰(2’O-ribose-methylations)，该甲基化修饰位于核糖体 2′-OH上； H&#x2F;ACA box 类型 snoRNAs 用于指导假尿苷酸化修饰(Pseudouridylation)，即将尿嘧啶转换为假尿嘧啶。</p>
<p><strong>snoscan</strong>用于鉴定 C&#x2F;D box 类型 snoRNAs； <strong>snoGPS</strong>用于鉴定 H&#x2F;ACA box 类型的 snoRNAs。</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment">#wget lowelab.ucsc.edu/software/snoscan.tar.gz</span><br><span class="hljs-comment">#下载最新1.0</span><br>$ tar zxf snoscan.tar.gz -C /opt/biosoft/<br><span class="hljs-built_in">cd</span> snoscan-1.0<br><span class="hljs-built_in">cd</span> squid-1.5.11<br>$ perl -p -i -e <span class="hljs-string">&#x27;s/getline/get_line/g&#x27;</span> sqio.c<span class="hljs-comment">#另一个博客修改为getLine？？？</span><br>$ <span class="hljs-built_in">cd</span> ..<br>$ make<br><span class="hljs-comment">#no conda, fat error </span><br><span class="hljs-comment">#/usr/bin/ld: cannot find -lsquid</span><br><span class="hljs-comment">#collect2: error: ld returned 1 exit status</span><br><span class="hljs-comment">#no fat no conda:ok.</span><br>$ <span class="hljs-built_in">echo</span> <span class="hljs-string">&#x27;PATH=$PATH:/opt/biosoft/snoscan-1.0&#x27;</span> &gt;&gt; ~/.bashrc<br><span class="hljs-comment">#perl -p -i -e &#x27;s#/usr/local/bin/perl#/usr/bin/perl#&#x27; sort-snos</span><br>change sort-snos   <span class="hljs-comment">#!/usr/bin/perl</span><br></code></pre></td></tr></table></figure>

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs sh">$ snoscan rRNA.fa genome.fasta -s -o out<br>rRNA.fa      : rRNA序列的fasta格式<br>genome.fasta : 用于搜索 snoRNAs 的基因组序列<br></code></pre></td></tr></table></figure>

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><code class="hljs sh">-h<br>    打印帮助信息<br>-m string<br>    指定包含甲基化位点信息的文件。该文件的格式为：<br>        &gt;seq_A<br>        T 5 Verified meth site by TML<br>        C 12  Predicted by snoscan<br>        &gt;My-seq-B<br>        G 1   Partially modified site<br>        A 6   Mapped by Maden<br>    与fasta格式类似，头部是和 genome.fasta 序列名相同；内容部分分3列，以空格或制表符分割，第1列是位点的碱基，第2列是1-based的位点，第3列是60个字符以内的描述。<br>-o string<br>    将结构输出到指定的文件，默认输出到标准输出<br>-s<br>    是否保存 snoRNA 的序列到输出中<br></code></pre></td></tr></table></figure>

<blockquote>
<p>snoRNA 指导的 rRNA 的转录后修饰有 2 种： 甲基化(Methylation) 和 假尿苷酸化(Pseudouridylation)。</p>
<p>snoRNP(small nucleolar ribonucleoprotein): 每个 snoRNA 和至少4个蛋白分子形成复合物，来指导目标RNA的修饰。</p>
<p>snoRNAs 主要有 2 种不同类型。其中 C&#x2F;D box 类型用于指导甲基化修饰； H&#x2F;ACA box 类型用于指导假尿苷酸化修饰。</p>
<p>C&#x2F;D box 类型的 snoRNAs 包含 C(RUGAUGA)环 和 D(CUGA)环，分别在 snoRNA 的 5′ 和 3′ 端。</p>
<p>H&#x2F;ACA box 类型的 snoRNAs 由 2 个发夹结构和 2 个单链区组成，称为 hairpin-hinge-hairpin-tail 结构。结构中包含 H box(ANANNA) 和 ACA box(ACA)。</p>
</blockquote>
<h2 id="RNAmmer"><a href="#RNAmmer" class="headerlink" title="RNAmmer"></a>RNAmmer</h2><p><a target="_blank" rel="noopener" href="http://www.chenlianfu.com/?p=1979">http://www.chenlianfu.com/?p=1979</a></p>
<p>Appending installation info to &#x2F;home&#x2F;~&#x2F;perl5&#x2F;lib&#x2F;perl5&#x2F;x86_64-linux-thread-multi&#x2F;perllocal.pod<br>  JV&#x2F;Getopt-Long-2.52.tar.gz</p>
<p>Appending installation info to &#x2F;home&#x2F;~&#x2F;perl5&#x2F;lib&#x2F;perl5&#x2F;x86_64-linux-thread-multi&#x2F;perllocal.pod<br>  GRANTM&#x2F;XML-Simple-2.25.tar.gz</p>
<ol>
<li>perl的Getopt::Long和XML::Simple模块； 2. hmmsearch,需要安装hmmer-2.2g版本，使用最新版本会出错。</li>
</ol>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><code class="hljs sh">下载并安装hmmer-2.2g失败，查看https://blog.karinlag.no/2013/10/rnammer-install/，提示2.3<br>$ wget： http://hmmer.org/download.html<br>$ tar zxf hmmer-2.3<br><span class="hljs-built_in">mkdir</span> hmmer2.3<br>$ <span class="hljs-built_in">cd</span> hmmer-2.3<br>$ ./configure --prefix=~/opt/biosoft/hmmer2.3<br>$ <span class="hljs-built_in">mkdir</span> -p /opt/biosoft/hmmer-2.3/man/man1/ /opt/biosoft/hmmer-2.3/bin<br>$ make; make install<br><br>安装RNAmmer，该软件需要使用edu邮箱去申请。<br>$ <span class="hljs-built_in">mkdir</span> -p /opt/biosoft/rnammer-1.2<br>$ tar zxf rnammer-1.2.src.tar.Z -C /opt/biosoft/rnammer-1.2<br>修改rnammer中的<span class="hljs-variable">$INSTALL_PATH</span><br>修改rnammer中的<span class="hljs-variable">$HMMSEARCH_BINARY</span><br><br>检测是否正常运行<br>./rnammer -S bac -multi -f rRNA.fasta -h rRNA.hmmreport -xml rRNA.xml -gff rRNA.gff2 example/ecoli.fsa &gt; rnammer.log 2&gt;&amp;1 &amp;<br><span class="hljs-comment">#error：FATAL: POSIX threads support is not compiled into HMMER; --cpu doesn&#x27;t have any effect</span><br><span class="hljs-comment">#依然在官网：https://blog.karinlag.no/2013/10/rnammer-install/</span><br></code></pre></td></tr></table></figure>

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><code class="hljs sh">USAGE:<br>$ rnammer [options] sequence.fasta<br>-S  指定输入序列的物种所属的界： arc bac 或 euk<br>-gff  生成的gff文件结果<br>-m  所需要预测的moleculers： <span class="hljs-string">&#x27;tsu&#x27;</span> <span class="hljs-keyword">for</span> 5/8s rRNA, <span class="hljs-string">&#x27;ssu&#x27;</span> <span class="hljs-keyword">for</span> 16/18s rRNA, <span class="hljs-string">&#x27;lsu&#x27;</span> <span class="hljs-keyword">for</span> 23/28s rRNA。如果全部进行预测，则该参数后为 <span class="hljs-string">&#x27;tsu,ssu,lsu&#x27;</span>。<br>-multi  并行运算，预测正反两条链上所有的moleculers。最多并行运行6个计算。使用该参数，则不需要上一个参数。<br>-f  生成的rRNA的fasta结果文件<br>-h  生成的hmm报告结果<br>-gff  生成的rRNA的gff2文件<br>-x  生成的xml结果文件<br></code></pre></td></tr></table></figure>

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs sh">$ /opt/biosoft/rnammer-1.2/rnammer -S euk -multi -f rRNA.fasta\<br> -h rRNA.hmmreport -xml rRNA.xml -gff rRNA.gff2 genome.fasta<br> <span class="hljs-comment">#result: rRNA.fasta，rRNA.gff2，rRNA.hmmreport，rRNA.xml</span><br></code></pre></td></tr></table></figure>


              
            </div>
            <hr/>
            <div>
              <div class="post-metas my-3">
  
    <div class="post-meta mr-3 d-flex align-items-center">
      <i class="iconfont icon-category"></i>
      

<span class="category-chains">
  
  
    
      <span class="category-chain">
        
  <a href="/GeekFocus/categories/Software/" class="category-chain-item">Software</a>
  
  

      </span>
    
  
</span>

    </div>
  
  
    <div class="post-meta">
      <i class="iconfont icon-tags"></i>
      
        <a href="/GeekFocus/tags/annotation/">#annotation</a>
      
        <a href="/GeekFocus/tags/Maker2/">#Maker2</a>
      
    </div>
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>MAKER</div>
      <div>https://wangc1106.github.io/GeekFocus/2022/04/10/2022-04-10-MAKER-anno/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>Author</div>
          <div>banner</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>Posted on</div>
          <div>April 10, 2022</div>
        </div>
      
      
      <div class="license-meta-item">
        <div>Licensed under</div>
        <div>
          
            
            
              <a target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
              <span class="hint--top hint--rounded" aria-label="BY - Attribution">
                <i class="iconfont icon-by"></i>
              </span>
              </a>
            
          
        </div>
      </div>
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/GeekFocus/2022/04/21/2022-04-28-BRAKER/" title="BRAKER2">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">BRAKER2</span>
                        <span class="visible-mobile">Previous</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/GeekFocus/2022/03/28/2022-03-28-TPM/" title="count, TPM, FPKM, RPKM in RNA-seq">
                        <span class="hidden-mobile">count, TPM, FPKM, RPKM in RNA-seq</span>
                        <span class="visible-mobile">Next</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="margin-left: -1rem">
    <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;Table of Contents</p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>
  </div>
</div>





  



  



  



  



  






    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">Search</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">Keyword</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <span>banner</span> <i class="iconfont icon-love"></i> <span>2021-2023©</span> 
    </div>
  
  
    <div class="statistics">
  
  

  
    
      <span id="leancloud-site-pv-container" style="display: none">
        总访问量 
        <span id="leancloud-site-pv"></span>
         次
      </span>
    
    
      <span id="leancloud-site-uv-container" style="display: none">
        总访客数 
        <span id="leancloud-site-uv"></span>
         人
      </span>
    
    

  
</div>

  
  
  
</div>

    <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
    </div>
  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.0/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/GeekFocus/js/events.js" ></script>
<script  src="/GeekFocus/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>






  
    <script  src="/GeekFocus/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.18.2/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      headingSelector : CONFIG.toc.headingSelector || 'h1,h2,h3,h4,h5,h6',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      collapseDepth   : CONFIG.toc.collapseDepth || 0,
      scrollSmooth    : true,
      headingsOffset  : -boardTop
    });
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }
  });
</script>


  <script>
  (function() {
    var enableLang = CONFIG.code_language.enable && CONFIG.code_language.default;
    var enableCopy = CONFIG.copy_btn;
    if (!enableLang && !enableCopy) {
      return;
    }

    function getBgClass(ele) {
      return Fluid.utils.getBackgroundLightness(ele) >= 0 ? 'code-widget-light' : 'code-widget-dark';
    }

    var copyTmpl = '';
    copyTmpl += '<div class="code-widget">';
    copyTmpl += 'LANG';
    copyTmpl += '</div>';
    jQuery('.markdown-body pre').each(function() {
      var $pre = jQuery(this);
      if ($pre.find('code.mermaid').length > 0) {
        return;
      }
      if ($pre.find('span.line').length > 0) {
        return;
      }

      var lang = '';

      if (enableLang) {
        lang = CONFIG.code_language.default;
        if ($pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2 && $pre.children().hasClass('hljs')) {
          lang = $pre[0].children[0].classList[1];
        } else if ($pre[0].getAttribute('data-language')) {
          lang = $pre[0].getAttribute('data-language');
        } else if ($pre.parent().hasClass('sourceCode') && $pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2) {
          lang = $pre[0].children[0].classList[1];
          $pre.parent().addClass('code-wrapper');
        } else if ($pre.parent().hasClass('markdown-body') && $pre[0].classList.length === 0) {
          $pre.wrap('<div class="code-wrapper"></div>');
        }
        lang = lang.toUpperCase().replace('NONE', CONFIG.code_language.default);
      }
      $pre.append(copyTmpl.replace('LANG', lang).replace('code-widget">',
        getBgClass($pre[0]) + (enableCopy ? ' code-widget copy-btn" data-clipboard-snippet><i class="iconfont icon-copy"></i>' : ' code-widget">')));

      if (enableCopy) {
        Fluid.utils.createScript('https://lib.baomitu.com/clipboard.js/2.0.10/clipboard.min.js', function() {
          var clipboard = new window.ClipboardJS('.copy-btn', {
            target: function(trigger) {
              var nodes = trigger.parentNode.childNodes;
              for (var i = 0; i < nodes.length; i++) {
                if (nodes[i].tagName === 'CODE') {
                  return nodes[i];
                }
              }
            }
          });
          clipboard.on('success', function(e) {
            e.clearSelection();
            e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-copy', 'icon-success');
            setTimeout(function() {
              e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-success', 'icon-copy');
            }, 2000);
          });
        });
      }
    });
  })();
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/GeekFocus/js/local-search.js" ></script>

  <script defer src="/GeekFocus/js/leancloud.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/GeekFocus/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">Blog works best with JavaScript enabled</div>
  </noscript>
</body>
</html>
