

<!DOCTYPE html>
<html lang="en" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/GeekFocus/img/fluid.png">
  <link rel="icon" href="/GeekFocus/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="banner">
  <meta name="keywords" content="Bioinformatics">
  
    <meta name="description" content="鉴定蛋白质结构域保守位点，并进行家族分类">
<meta property="og:type" content="article">
<meta property="og:title" content="【FuncAnno】【InterProScan】">
<meta property="og:url" content="https://wangc1106.github.io/GeekFocus/2022/06/05/2022-06-05-FuncAnno-InterPro/index.html">
<meta property="og:site_name" content="DrBanner">
<meta property="og:description" content="鉴定蛋白质结构域保守位点，并进行家族分类">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2022-06-05T13:22:34.000Z">
<meta property="article:modified_time" content="2022-06-05T13:22:34.033Z">
<meta property="article:author" content="banner">
<meta property="article:tag" content="FuncAnno">
<meta name="twitter:card" content="summary_large_image">
  
  
  
  <title>【FuncAnno】【InterProScan】 - DrBanner</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/GeekFocus/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/GeekFocus/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/GeekFocus/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/GeekFocus/","version":"1.9.1","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"right","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":false,"follow_dnt":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":"https://wangc1106.github.io/GeekFocus","path":"window.location.pathname","ignore_local":false}},"search_path":"/GeekFocus/local-search.xml"};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/GeekFocus/js/utils.js" ></script>
  <script  src="/GeekFocus/js/color-schema.js" ></script>

  
<meta name="generator" content="Hexo 6.2.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 70vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/GeekFocus/">
      <strong>DrBanner</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/">
                <i class="iconfont icon-home-fill"></i>
                Home
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/archives/">
                <i class="iconfont icon-archive-fill"></i>
                Archives
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/categories/">
                <i class="iconfont icon-category-fill"></i>
                Categories
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/tags/">
                <i class="iconfont icon-tags-fill"></i>
                Tags
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/about/">
                <i class="iconfont icon-user-fill"></i>
                About
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              &nbsp;<i class="iconfont icon-search"></i>&nbsp;
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/GeekFocus/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="【FuncAnno】【InterProScan】"></span>
          
        </div>

        
          
  <div class="mt-3">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-author" aria-hidden="true"></i>
        banner
      </span>
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2022-06-05 21:22" pubdate>
          June 5, 2022 pm
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          16k words
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          138 mins
        
      </span>
    

    
    
      
        <span id="leancloud-page-views-container" class="post-meta" style="display: none">
          <i class="iconfont icon-eye" aria-hidden="true"></i>
          <span id="leancloud-page-views"></span> views
        </span>
        
      
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">【FuncAnno】【InterProScan】</h1>
            
              <p class="note note-info">
                
                  
                    Last updated on June 5, 2022 pm
                  
                
              </p>
            
            <div class="markdown-body">
              
              <h1 id="鉴定蛋白质结构域保守位点，并进行家族分类"><a href="#鉴定蛋白质结构域保守位点，并进行家族分类" class="headerlink" title="鉴定蛋白质结构域保守位点，并进行家族分类"></a>鉴定蛋白质结构域保守位点，并进行家族分类</h1><span id="more"></span>

<p><a target="_blank" rel="noopener" href="https://interproscan-docs.readthedocs.io/en/latest/HowToRun.html">https://interproscan-docs.readthedocs.io/en/latest/HowToRun.html</a></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/9d832ea9904f">https://www.jianshu.com/p/9d832ea9904f</a></p>
<p><a target="_blank" rel="noopener" href="https://sr-c.github.io/2018/09/13/interproscan-config/">https://sr-c.github.io/2018/09/13/interproscan-config/</a></p>
<h2 id="0-介绍"><a href="#0-介绍" class="headerlink" title="0. 介绍"></a>0. 介绍</h2><p>主要功能：</p>
<ol>
<li>预测蛋白质的结构域和功能位点</li>
<li>根据蛋白质含有的结构域和特殊位点进行分类</li>
</ol>
<p>默认包含所有的数据库，<font color="blue">结构域，特征序列，和信号肽</font>，可以选择部分数据库</p>
<h3 id="Familys-domains-sites-amp-repeats"><a href="#Familys-domains-sites-amp-repeats" class="headerlink" title="Familys, domains, sites &amp; repeats"></a>Familys, domains, sites &amp; repeats</h3><p>TIGRFAMS</p>
<p>SFLD</p>
<p>PANTHER</p>
<p>HAMAP</p>
<p>PROSITE profiles?fail</p>
<p>PROSITE patterns?</p>
<p>SMART</p>
<p>CDD</p>
<p>PRINTS</p>
<p>Pfam</p>
<p>PIRSF</p>
<h3 id="Structual-domains"><a href="#Structual-domains" class="headerlink" title="Structual domains"></a>Structual domains</h3><p>SUPERFAMILY</p>
<p>Gene3D</p>
<h3 id="Other-sequence-features"><a href="#Other-sequence-features" class="headerlink" title="Other sequence features"></a>Other sequence features</h3><p>SignalP</p>
<p>Coils</p>
<p>TMHMM</p>
<p>MobiDBLite</p>
<p>Phobius</p>
<h3 id="other-category"><a href="#other-category" class="headerlink" title="other category"></a>other category</h3><p>SIngnalP_EUK; Signal_GRAM_POSITIVE; </p>
<h3 id="单独配置分析"><a href="#单独配置分析" class="headerlink" title="单独配置分析"></a>单独配置分析</h3><p>分析信号肽，真核，Gram+细菌，Gram-细菌，跨膜结构，这部分软件需要自行配置</p>
<p>SignalP, TMHMM, 严格按照版本， deactivated 提示</p>
<h2 id="1-下载安装"><a href="#1-下载安装" class="headerlink" title="1. 下载安装"></a>1. 下载安装</h2><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment"># 此处为最新版本5.52-86.0，64位，下载MD5文件用来检验下载的完整性</span><br>wget https://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.52-86.0/interproscan-5.52-86.0-64-bit.tar.gz<br>wget https://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.52-86.0/interproscan-5.52-86.0-64-bit.tar.gz.md5<br><span class="hljs-built_in">md5sum</span> -c interproscan-5.52-86.0-64-bit.tar.gz.md5<br><span class="hljs-comment"># 返回ok则说明下载完整</span><br>tar -pxvzf interproscan-5.52-86.0-*-bit.tar.gz<br><br><span class="hljs-comment">###Latest version 5.56-89.0</span><br>https://github.com/ebi-pf-team/interproscan/releases<br>interproscan-5.56-89.0 Latest<br>release: interproscan-5.56-89.0<br>md5: f3288327a6c7be1ff42220c3b7a1de43<br>cpu: 64 bit<br>os: Linux<br>size: 7.5GB compressed<br><br>https://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.56-89.0/interproscan-5.56-89.0-64-bit.tar.gz<br>https://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.56-89.0/interproscan-5.56-89.0-64-bit.tar.gz.md5<br><span class="hljs-built_in">md5sum</span> -c interproscan-5.56-89.0-64-bit.tar.gz.md5<br>interproscan-5.56-89.0-64-bit.tar.gz: OK<br><span class="hljs-comment">#解压</span><br>tar -pxvzf interproscan-5.56-89.0-64-bit.tar.gz<br><br><span class="hljs-comment"># 解压完后，进入目录，查看是否安装完好，若有用法说明弹出则表示安装成功。</span><br>./interproscan.sh<br><br>Java version 11 is required to run InterProScan.<br>download jdk-11.0.15.1[https://www.oracle.com/java/technologies/downloads/<span class="hljs-comment">#java11]</span><br>jdk-11.0.15.1_linux-x64_bin.tar.gz<br><br><span class="hljs-comment">#export JAVA_HOME=/usr/local/jdk-11.0.8</span><br><span class="hljs-comment">#export CLASSPATH=.:$JAVA_HOME/jre/lib:$JAVA_HOME/lib:$JAVA_HOME/lib/tools.jar</span><br><span class="hljs-comment">#export PATH=$PATH:$JAVA_HOME/bin:$JAVA_HOME/jre/bin</span><br><br><span class="hljs-built_in">export</span> JAVA_HOME=/path/jdk-11.0.15.1<br><span class="hljs-built_in">export</span> PATH=<span class="hljs-variable">$PATH</span>:<span class="hljs-variable">$JAVA_HOME</span>/bin<br>failed，still /usr/bin/java<br>vi ./interproscan.sh<br><span class="hljs-comment"># old JAVA=$(type -p java)</span><br>JAVA=$(<span class="hljs-built_in">type</span> -p /path/jdk-11.0.15.1/bin/java)<br><span class="hljs-comment">#https://blog.csdn.net/weixin_43484846/article/details/121400487</span><br><span class="hljs-comment">#此blog 用ln -s /path/jdk-11.0.15.1/bin/java  /usr/bin/java 未尝试此法</span><br>./interproscan.sh <br>Checking any hmm models that need indexing ... this may take a few minutes<br>Working...    <span class="hljs-keyword">done</span>.<br>Pressed and indexed 263 HMMs (263 names and 263 accessions).<br>Models pressed into binary file:   data/antifam/7.0/AntiFam.hmm.h3m<br>SSI index <span class="hljs-keyword">for</span> binary model file:   data/antifam/7.0/AntiFam.hmm.h3i<br>Profiles (MSV part) pressed into:  data/antifam/7.0/AntiFam.hmm.h3f<br>Profiles (remainder) pressed into: data/antifam/7.0/AntiFam.hmm.h3p<br>Working...    <span class="hljs-keyword">done</span>.<br>Pressed and indexed 66720 HMMs (66720 names).<br>Models pressed into binary file:   data/gene3d/4.3.0/gene3d_main.hmm.h3m<br>SSI index <span class="hljs-keyword">for</span> binary model file:   data/gene3d/4.3.0/gene3d_main.hmm.h3i<br>Profiles (MSV part) pressed into:  data/gene3d/4.3.0/gene3d_main.hmm.h3f<br>Profiles (remainder) pressed into: data/gene3d/4.3.0/gene3d_main.hmm.h3p<br>Working...    <span class="hljs-keyword">done</span>.<br>Pressed and indexed 2383 HMMs (2383 names).<br>Models pressed into binary file:   data/hamap/2021_04/hamap.hmm.lib.h3m<br>SSI index <span class="hljs-keyword">for</span> binary model file:   data/hamap/2021_04/hamap.hmm.lib.h3i<br>Profiles (MSV part) pressed into:  data/hamap/2021_04/hamap.hmm.lib.h3f<br>Profiles (remainder) pressed into: data/hamap/2021_04/hamap.hmm.lib.h3p<br>Working...    <span class="hljs-keyword">done</span>.<br>Pressed and indexed 134000 HMMs (134000 names).<br>Models pressed into binary file:   data/panther/15.0/panther.hmm.h3m<br>SSI index <span class="hljs-keyword">for</span> binary model file:   data/panther/15.0/panther.hmm.h3i<br>Profiles (MSV part) pressed into:  data/panther/15.0/panther.hmm.h3f<br>Profiles (remainder) pressed into: data/panther/15.0/panther.hmm.h3p<br>Working...    <span class="hljs-keyword">done</span>.<br>Pressed and indexed 19179 HMMs (19179 names and 19179 accessions).<br>Models pressed into binary file:   data/pfam/34.0/pfam_a.hmm.h3m<br>SSI index <span class="hljs-keyword">for</span> binary model file:   data/pfam/34.0/pfam_a.hmm.h3i<br>Profiles (MSV part) pressed into:  data/pfam/34.0/pfam_a.hmm.h3f<br>Profiles (remainder) pressed into: data/pfam/34.0/pfam_a.hmm.h3p<br>Working...    <span class="hljs-keyword">done</span>.<br>Pressed and indexed 3283 HMMs (3283 names and 3283 accessions).<br>Models pressed into binary file:   data/pirsf/3.10/sf_hmm_all.h3m<br>SSI index <span class="hljs-keyword">for</span> binary model file:   data/pirsf/3.10/sf_hmm_all.h3i<br>Profiles (MSV part) pressed into:  data/pirsf/3.10/sf_hmm_all.h3f<br>Profiles (remainder) pressed into: data/pirsf/3.10/sf_hmm_all.h3p<br>Working...    <span class="hljs-keyword">done</span>.<br>Pressed and indexed 1510 HMMs (1510 names and 1510 accessions).<br>Models pressed into binary file:   data/pirsr/2021_05/sr_hmm_all.h3m<br>SSI index <span class="hljs-keyword">for</span> binary model file:   data/pirsr/2021_05/sr_hmm_all.h3i<br>Profiles (MSV part) pressed into:  data/pirsr/2021_05/sr_hmm_all.h3f<br>Profiles (remainder) pressed into: data/pirsr/2021_05/sr_hmm_all.h3p<br>Working...    <span class="hljs-keyword">done</span>.<br>Pressed and indexed 299 HMMs (299 names and 299 accessions).<br>Models pressed into binary file:   data/sfld/4/sfld.hmm.h3m<br>SSI index <span class="hljs-keyword">for</span> binary model file:   data/sfld/4/sfld.hmm.h3i<br>Profiles (MSV part) pressed into:  data/sfld/4/sfld.hmm.h3f<br>Profiles (remainder) pressed into: data/sfld/4/sfld.hmm.h3p<br>Working...    <span class="hljs-keyword">done</span>.<br>Pressed and indexed 15438 HMMs (15438 names and 15438 accessions).<br>Models pressed into binary file:   data/superfamily/1.75/hmmlib_1.75.h3m<br>SSI index <span class="hljs-keyword">for</span> binary model file:   data/superfamily/1.75/hmmlib_1.75.h3i<br>Profiles (MSV part) pressed into:  data/superfamily/1.75/hmmlib_1.75.h3f<br>Profiles (remainder) pressed into: data/superfamily/1.75/hmmlib_1.75.h3p<br>Working...    <span class="hljs-keyword">done</span>.<br>Pressed and indexed 4488 HMMs (4488 names and 4488 accessions).<br>Models pressed into binary file:   data/tigrfam/15.0/TIGRFAMs_HMM.LIB.h3m<br>SSI index <span class="hljs-keyword">for</span> binary model file:   data/tigrfam/15.0/TIGRFAMs_HMM.LIB.h3i<br>Profiles (MSV part) pressed into:  data/tigrfam/15.0/TIGRFAMs_HMM.LIB.h3f<br>Profiles (remainder) pressed into: data/tigrfam/15.0/TIGRFAMs_HMM.LIB.h3p<br>Completed indexing the hmm models.<br>30/05/2022 22:41:20:513 Welcome to InterProScan-5.56-89.0<br>30/05/2022 22:41:20:516 Running InterProScan v5 <span class="hljs-keyword">in</span> STANDALONE mode... on Linux<br>usage: java -XX:+UseParallelGC -XX:ParallelGCThreads=2 -XX:+AggressiveOpts -XX:+UseFastAccessorMethods -Xms128M<br>            -Xmx2048M -jar interproscan-5.jar<br><br><span class="hljs-comment"># 进行初始化，此命令大致是准备好HMM模型以供hmmscan使用</span><br>python3 initial_setup.py<br><span class="hljs-comment"># no fat local</span><br>/xtdisk/xueyb_group/wangchenA/miniconda3/bin/python3 initial_setup.py<br></code></pre></td></tr></table></figure>

<h2 id="2-测试与参数"><a href="#2-测试与参数" class="headerlink" title="2. 测试与参数"></a>2. 测试与参数</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><code class="hljs bash">./interproscan.sh<br><span class="hljs-comment"># 短暂等待后，即会输出帮助信息</span><br>30/05/2022 22:44:40:746 Welcome to InterProScan-5.56-89.0<br>30/05/2022 22:44:40:747 Running InterProScan v5 <span class="hljs-keyword">in</span> STANDALONE mode... on Linux<br>usage: java -XX:+UseParallelGC -XX:ParallelGCThreads=2 -XX:+AggressiveOpts -XX:+UseFastAccessorMethods -Xms128M<br>            -Xmx2048M -jar interproscan-5.jar<br><br><span class="hljs-comment">#完成上面的步骤后，即可进行测试（以下两条命令选一即可）</span><br><span class="hljs-comment"># 程序不联网，使用本地的Lookup Service</span><br>./interproscan.sh -i test_all_appl.fasta -f tsv -dp<br><span class="hljs-comment"># 使用程序自带的测试数据进行测试</span><br>./interproscan.sh -i test_all_appl.fasta -f tsv<br>    <span class="hljs-comment"># -dp参数的含义：含有-dp则是关闭了连接EBI联网搜索的权限而使用本地化搜索，默认情况下是支持EBI联网搜索，若只需本地则使用-dp。（可能不正确）</span><br>    -dp,--disable-precalc Optional.  Disables use of the precalculated match lookup service.  All match calculations will be run locally.<br></code></pre></td></tr></table></figure>

<ul>
<li>-appl &#x2F; -applications 搜索应用的名字(可选)<ul>
<li>默认情况下，运行所有的分析，也可以自己指定一个或多个。</li>
</ul>
</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-comment"># 指定pfam搜索</span><br>./interproscan.sh -appl Pfam -i /path/to/sequences.fasta<br><span class="hljs-comment">#同时指定多个进行搜索</span><br>./interproscan.sh -appl CDD,COILS,Gene3D,HAMAP,MobiDBLite,PANTHER,Pfam,PIRSF,PRINTS,PROSITEPATTERNS,PROSITEPROFILES,SFLD,SMART,SUPERFAMILY,TIGRFAM -i /path/to/sequences.fasta<br></code></pre></td></tr></table></figure>

<ul>
<li>-i &#x2F;-fasta 序列文件<ul>
<li>输入的是蛋白或者核酸的fasta文件，将会返回TSV,XML,Gff3三个输出结果</li>
</ul>
</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs bash">./interproscan.sh -i /path/to/sequences.fasta<br><span class="hljs-comment"># 输出结果为sequences.tsv, sequence.xml, sequences.gff3</span><br></code></pre></td></tr></table></figure>

<ul>
<li>-t &#x2F; 序列类型<ul>
<li>默认是蛋白序列，若为核苷酸序列，则需要加上此参数</li>
</ul>
</li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs sh">./interproscan.sh -t n -i /path/to/sequences.fasta<br></code></pre></td></tr></table></figure>

<p>其他参数</p>
<h2 id="3-包含分析"><a href="#3-包含分析" class="headerlink" title="3. 包含分析"></a>3. 包含分析</h2><blockquote>
<ul>
<li>CDD</li>
<li>COLIS</li>
<li>Gene3D</li>
<li>HAMAP</li>
<li>MOBIDB</li>
<li>PANTHER</li>
<li>Pfam</li>
<li>PIRSF</li>
<li>SMART</li>
<li>…</li>
</ul>
</blockquote>
<h2 id="4-线程"><a href="#4-线程" class="headerlink" title="4. 线程"></a>4. 线程</h2><p>在单机运行InterProScan时，可在配置文件<code>interproscan.properties</code>中设置线程，其默认设置如下</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment"># Set the number of embedded workers to the number of processors that you would like to employ</span><br><span class="hljs-comment"># on the machine you are using to run InterProScan.</span><br><span class="hljs-comment">#number of embedded workers  a master process can have</span><br>number.of.embedded.workers=6<br>maxnumber.of.embedded.workers=8<br></code></pre></td></tr></table></figure>

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment">#对于一个40核的机器，推荐的设置如下</span><br><span class="hljs-comment">#Number of embedded workers at start time</span><br>number.of.embedded.workers=1<br><span class="hljs-comment">#Maximum number of embedded workers</span><br>maxnumber.of.embedded.workers=35<br></code></pre></td></tr></table></figure>

<p>推荐单次输入蛋白序列10000条，核酸序列1000条。若输入数据过多，建议将其分组运行。</p>
<p>使用<code>-appl</code>参数设置比对的项目，减少程序耗时</p>
<p>进行比对的蛋白序列中不应含有星号，否则将有如下报错。将序列中的星号删除即可。</p>
<p>InterProScan自带的blast+中rpsblast的编译环境与本机环境不同。通过替换这支程序来解决。下载ncbi提供的已编译版本，使用其中的rpsblast替换InterProScan自带的rpsblast即可。</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment"># backup original version</span><br><span class="hljs-built_in">cp</span> bin/blast/ncbi-blast-2.6.0+/rpsblast bin/blast/ncbi-blast-2.6.0+/rpsblast.bak<br><br><span class="hljs-comment"># copy the new version to the binary folder</span><br><span class="hljs-built_in">cp</span> ncbi-blast-2.6.0+/bin/rpsblast bin/blast/ncbi-blast-2.6.0+/rpsblast<br></code></pre></td></tr></table></figure>

<h2 id="5-运行"><a href="#5-运行" class="headerlink" title="5. 运行"></a>5. 运行</h2><p>一次运行同时实现多个信息注释</p>
<ul>
<li>InterPro注释</li>
<li>Pfam数据库注释(可以通过hmmscan搜索pfam数据库完成)</li>
<li>GO注释(可以基于NR和Pfam等数据库，然后BLAST2GO完成,)</li>
<li>Reactome通路注释，不同于KEGG</li>
</ul>
<p>命令如下</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs stylus">./interproscan-<span class="hljs-number">5.29</span>-<span class="hljs-number">68.0</span>/interproscan<span class="hljs-selector-class">.sh</span> -appl Pfam  -f TSV -<span class="hljs-selector-tag">i</span> sample<span class="hljs-selector-class">.fa</span> -cpu <span class="hljs-number">50</span> -<span class="hljs-selector-tag">b</span> sample -goterms -iprlookup -pa<br></code></pre></td></tr></table></figure>

<p><code>-appl</code>告诉软件要执行哪些数据分析，勾选的越多，分析速度越慢，Pfam就行。</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br></pre></td><td class="code"><pre><code class="hljs sh">software/interproscan-5.56-89.0/interproscan.sh<br>31/05/2022 11:39:27:267 Welcome to InterProScan-5.56-89.0<br>31/05/2022 11:39:27:269 Running InterProScan v5 <span class="hljs-keyword">in</span> STANDALONE mode... on Linux<br>usage: java -XX:+UseParallelGC -XX:ParallelGCThreads=2 -XX:+AggressiveOpts -XX:+UseFastAccessorMethods -Xms128M<br>            -Xmx2048M -jar interproscan-5.jar<br><br><br>Please give us your feedback by sending an email to<br><br>interhelp@ebi.ac.uk<br><br>-appl,--applications &lt;ANALYSES&gt;     Optional, <span class="hljs-string">&#x27;comma separated list&#x27;</span> of analyses.  Default: all available analysis<br><br>-b,--output-file-base &lt;OUTPUT-FILE-BASE&gt;     Optional, <span class="hljs-string">&#x27;base output filename&#x27;</span> (relative or absolute path). Note that this option, the --output-dir (-d) option and the --outfile (-o) option are mutually exclusive.  The appropriate file extension <span class="hljs-keyword">for</span> the output format(s) will be appended automatically. By default the input file path/name will be used.<br><br>-d,--output-dir &lt;OUTPUT-DIR&gt;     Optional, output directory.  Note that this option, the --outfile (-o) option and the --output-file-base (-b) option are mutually exclusive. The output filename(s) are the same as the input filename, with the appropriate file extension(s) <span class="hljs-keyword">for</span> the output format(s) appended automatically .<br><br>-o,--outfile &lt;EXPLICIT_OUTPUT_FILENAME&gt; Optional explicit output file name (relative or absolute path).  Note that this option, the --output-dir (-d) option and the --output-file-base (-b) option are mutually exclusive. If this option is given, you MUST specify a single output format using the -f option.  The output file name will not be modified. Note that specifying an output file name using this option OVERWRITES ANY EXISTING FILE.<br><br>-cpu,--cpu &lt;CPU&gt;     Optional, number of cores <span class="hljs-keyword">for</span> inteproscan.<br><br>-dp,--disable-precalc     Optional.  Disables use of the precalculated match lookup service.  All match calculations will be <span class="hljs-string">&#x27;run locally&#x27;</span>.<br><br>-dra,--disable-residue-annot     Optional, excludes sites from the XML, JSON output<br><br>-etra,--enable-tsv-residue-annot     Optional, includes sites <span class="hljs-keyword">in</span> TSV output<br><br>-exclappl,--excl-applications &lt;EXC-ANALYSES&gt;     Optional, comma separated list of <span class="hljs-string">&#x27;analyses you want to exclude&#x27;</span>.<br><br>-f,--formats &lt;OUTPUT-FORMATS&gt;     Optional, case-insensitive, comma separated <span class="hljs-string">&#x27;list of output formats&#x27;</span>. Supported formats are <span class="hljs-string">&#x27;TSV, XML, JSON, GFF3, HTML and SVG&#x27;</span>. <span class="hljs-string">&#x27;Default for protein sequences are TSV, XML and GFF3&#x27;</span>, or <span class="hljs-string">&#x27;for nucleotide sequences GFF3 and XML&#x27;</span>.<br><br>-iprlookup,--iprlookup     Also include <span class="hljs-string">&#x27;lookup of corresponding InterPro annotation in the TSV and GFF3 output formats&#x27;</span>.<br><br>-goterms,--goterms     Optional, <span class="hljs-string">&#x27;switch on lookup of corresponding Gene Ontology annotation (IMPLIES -iprlookup option)&#x27;</span><br><br>-pa,--pathways Optional, switch on <span class="hljs-string">&#x27;lookup of corresponding Pathway annotation (IMPLIES -iprlookup option)&#x27;</span><br><br>-<span class="hljs-built_in">help</span>,--<span class="hljs-built_in">help</span>     Optional, display <span class="hljs-built_in">help</span> information<br><br>-i,--input &lt;INPUT-FILE-PATH&gt;     Optional, path to fasta file that should be loaded on Master startup. Alternatively, <span class="hljs-keyword">in</span> CONVERT mode, the InterProScan 5 XML file to convert.<br><br>-incldepappl,--incl-dep-applications &lt;INC-DEP-ANALYSES&gt;   Optional, comma separated list of deprecated analyses that you want included.  If this option is not <span class="hljs-built_in">set</span>, deprecated analyses will not run.<br><br>-ms,--minsize &lt;MINIMUM-SIZE&gt;     Optional, <span class="hljs-string">&#x27;minimum nucleotide size of ORF to report[100aa?]&#x27;</span>. Will only be considered <span class="hljs-keyword">if</span> n is specified as a sequence <span class="hljs-built_in">type</span>. Please be aware of the fact that <span class="hljs-keyword">if</span> you specify a too short value it might be that the analysis takes a very long time!<br><br>-t,--seqtype &lt;SEQUENCE-TYPE&gt;     Optional, the <span class="hljs-built_in">type</span> of the input sequences (dna/rna (n) or protein (p)).  The <span class="hljs-string">&#x27;default sequence type is protein&#x27;</span>. Optional, specify temporary file directory (relative or absolute path). The <span class="hljs-string">&#x27;default location is temp/&#x27;</span>.<br><br>-verbose,--verbose Optional, display more ‘verbose冗长’ <span class="hljs-built_in">log</span> output<br><br>-version,--version     Optional, display version number<br><br>-vl,--verbose-level &lt;VERBOSE-LEVEL&gt;     Optional, display <span class="hljs-string">&#x27;verbose log output at level specified&#x27;</span>.<br><br>-vtsv,--output-tsv-version     Optional, includes a TSV version file along with any TSV <br>output (when TSV output requested)<br><br>Copyright © EMBL European Bioinformatics Institute, Hinxton, Cambridge, UK. (http://www.ebi.ac.uk) The InterProScan software itself is provided under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0.html). Third party components (e.g. member database binaries and models) are subject to separate licensing - please see the individual member database websites <span class="hljs-keyword">for</span> details.<br><br><span class="hljs-string">&#x27;Available analyses&#x27;</span>:<br><span class="hljs-string">&#x27;TIGRFAM (15.0)&#x27;</span> : TIGRFAMs are <span class="hljs-string">&#x27;protein families&#x27;</span> based on <span class="hljs-string">&#x27;hidden Markov models (HMMs)&#x27;</span>.<br><br><span class="hljs-string">&#x27;SFLD (4)&#x27;</span> : SFLD is a database of <span class="hljs-string">&#x27;protein families&#x27;</span> based on <span class="hljs-string">&#x27;hidden Markov models (HMMs)&#x27;</span>.<br><br><span class="hljs-string">&#x27;√SUPERFAMILY (1.75)&#x27;</span> : SUPERFAMILY is a database of <span class="hljs-string">&#x27;structural and functional annotations&#x27;</span> <span class="hljs-keyword">for</span> <span class="hljs-string">&#x27;all proteins and genomes&#x27;</span>.<br><br><span class="hljs-string">&#x27;√PANTHER (15.0)&#x27;</span> : The PANTHER (<span class="hljs-string">&#x27;Protein ANalysis THrough Evolutionary Relationships&#x27;</span>) Classification System is a unique resource that <span class="hljs-string">&#x27;classifies genes by their functions, using published scientific experimental evidence and evolutionary relationships to predict function&#x27;</span> even <span class="hljs-keyword">in</span> the absence of direct experimental evidence.<br><br><span class="hljs-string">&#x27;√Gene3D (4.3.0)&#x27;</span> : <span class="hljs-string">&#x27;Structural assignment&#x27;</span> <span class="hljs-keyword">for</span> whole genes and genomes using the <span class="hljs-string">&#x27;CATH domain structure database&#x27;</span>.<br><br>Hamap (2021_04) : High-quality Automated and Manual Annotation of <span class="hljs-string">&#x27;Microbial Proteomes&#x27;</span>.<br><br>Coils (2.2.1) : Prediction of coiled coil regions? <span class="hljs-keyword">in</span> proteins.<br><br><span class="hljs-string">&#x27;ProSiteProfiles (2022_01)&#x27;</span>: PROSITE consists of documentation entries describing <span class="hljs-string">&#x27;protein domains, families and functional sites&#x27;</span> as well as <span class="hljs-string">&#x27;associated patterns and profiles&#x27;</span> to identify them.<br><br><span class="hljs-string">&#x27;√SMART (7.1)&#x27;</span> : SMART allows the identification and analysis of <span class="hljs-string">&#x27;domain architectures&#x27;</span> based on <span class="hljs-string">&#x27;hidden Markov models&#x27;</span> (HMMs). <br><br><span class="hljs-string">&#x27;CDD (3.18)&#x27;</span> : CDD predicts protein <span class="hljs-string">&#x27;domains and families&#x27;</span> based on <span class="hljs-string">&#x27;a collection of well-annotated multiple sequence alignment models&#x27;</span>.<br><br><span class="hljs-string">&#x27;√PRINTS (42.0)&#x27;</span> : A compendium of <span class="hljs-string">&#x27;protein fingerprints&#x27;</span> - a fingerprint is a group of <span class="hljs-string">&#x27;conserved motifs&#x27;</span> used to characterise a protein family.<br><br>PIRSR (2021_05) : PIRSR is a database of protein families based on hidden Markov models (HMMs) and Site Rules.<br><br>ProSitePatterns (2022_01) : PROSITE consists of documentation entries describing <span class="hljs-string">&#x27;protein domains, families and functional&#x27;</span> sites as well as associated patterns and profiles to identify them.<br><br>AntiFam (7.0) : AntiFam is a resource of profile-HMMs designed to identify spurious protein predictions.<br><br><span class="hljs-string">&#x27;√Pfam (34.0)&#x27;</span> : A <span class="hljs-string">&#x27;large collection of protein families&#x27;</span>, each represented by <span class="hljs-string">&#x27;multiple sequence alignments and hidden Markov models (HMMs)&#x27;</span>.<br><br>MobiDBLite (2.0) : Prediction of intrinsically disordered regions <span class="hljs-keyword">in</span> proteins.<br><br><span class="hljs-string">&#x27;PIRSF (3.10)&#x27;</span> : The PIRSF concept is used as a guiding principle to <span class="hljs-string">&#x27;provide comprehensive and non-overlapping clustering of UniProtKB sequences into a hierarchical order&#x27;</span> to reflect their evolutionary relationships.<br><br><span class="hljs-string">&#x27;Deactivated analyses&#x27;</span>:<br>SignalP_GRAM_NEGATIVE (4.1) : Analysis SignalP_GRAM_NEGATIVE is deactivated, because the resources expected at the following paths <span class="hljs-keyword">do</span> not exist: bin/signalp/4.1/signalp<br><br>TMHMM (2.0c) : Analysis TMHMM is deactivated, because the resources expected at the following paths <span class="hljs-keyword">do</span> not exist: bin/tmhmm/2.0c/decodeanhmm, data/tmhmm/2.0c/TMHMM2.0c.model<br><br>SignalP_EUK (4.1) : Analysis SignalP_EUK is deactivated, because the resources expected at the following paths <span class="hljs-keyword">do</span> not exist: bin/signalp/4.1/signalp<br><br>Phobius (1.01) : Analysis Phobius is deactivated, because the resources expected at the following paths <span class="hljs-keyword">do</span> not exist: bin/phobius/1.01/phobius.pl<br><br>SignalP_GRAM_POSITIVE (4.1) : Analysis SignalP_GRAM_POSITIVE is deactivated, because the resources expected at the following paths <span class="hljs-keyword">do</span> not exist: bin/signalp/4.1/signalp<br></code></pre></td></tr></table></figure>



<p>IF8-PlantBio 杜鹃属和海南冀鹏科植物：<strong>Pfam</strong>, <strong>PRINTS</strong>, <strong>SMART</strong>, ProDom and PROSITE</p>
<p>IF8-PlantBio 常绿杜鹃基因组：TIGRFAM, Phobius, SignalP, <strong>SUPERFAMILY</strong>, <strong>PANTHER</strong>, <strong>Gene3D</strong>, ProSite, Coils, <strong>PRINTS</strong>, <strong>SMART</strong>, <strong>Pfam</strong>, PIRSF, TMHMM and GO. </p>
<p>IF3.6-frontierInPlantSci 牛油果树：<strong>PFAM</strong>, <strong>Gene3D</strong>, <strong>PANTHER</strong>, <strong>CDD</strong>, <strong>SUPERFAMILY</strong>, ProSite, and GO</p>
<p>IF3.4-GBE 叶蝉基因组：<strong>Pfam</strong>,  <strong>Superfamily</strong> , <strong>Gene3D</strong>, <strong>SMART</strong>, <strong>CDD</strong></p>
<p>ProSiteProfiles, ProSitePatterns</p>
<p>-iprlookup,–iprlookup     Also include ‘lookup of corresponding <strong>InterPro</strong> annotation in the TSV and GFF3 output formats’.</p>
<p>-goterms,–goterms     Optional, ‘switch on lookup of corresponding <strong>Gene Ontology</strong> annotation (IMPLIES -iprlookup option)’</p>
<p>-pa,–pathways Optional, switch on ‘lookup of corresponding <strong>Pathway</strong> annotation (IMPLIES -iprlookup option)’</p>
<p>-vl</p>
<p>Pfam,PRINTS,SMART,SUPERFAMILY,PANTHER,Gene3D,CDD,ProSiteProfiles?ProSitePatterns?</p>

              
            </div>
            <hr/>
            <div>
              <div class="post-metas my-3">
  
    <div class="post-meta mr-3 d-flex align-items-center">
      <i class="iconfont icon-category"></i>
      

<span class="category-chains">
  
  
    
      <span class="category-chain">
        
  <a href="/GeekFocus/categories/Software/" class="category-chain-item">Software</a>
  
  

      </span>
    
  
</span>

    </div>
  
  
    <div class="post-meta">
      <i class="iconfont icon-tags"></i>
      
        <a href="/GeekFocus/tags/FuncAnno/">#FuncAnno</a>
      
    </div>
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>【FuncAnno】【InterProScan】</div>
      <div>https://wangc1106.github.io/GeekFocus/2022/06/05/2022-06-05-FuncAnno-InterPro/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>Author</div>
          <div>banner</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>Posted on</div>
          <div>June 5, 2022</div>
        </div>
      
      
      <div class="license-meta-item">
        <div>Licensed under</div>
        <div>
          
            
            
              <a target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
              <span class="hint--top hint--rounded" aria-label="BY - Attribution">
                <i class="iconfont icon-by"></i>
              </span>
              </a>
            
          
        </div>
      </div>
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/GeekFocus/2022/06/05/2022-06-05-RunningAugustus/" title="LTR-RT/LTR REs">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">LTR-RT/LTR REs</span>
                        <span class="visible-mobile">Previous</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/GeekFocus/2022/06/05/2022-06-05-FuncAnno-BLASTP/" title="【FuncAnno】【BLAST】">
                        <span class="hidden-mobile">【FuncAnno】【BLAST】</span>
                        <span class="visible-mobile">Next</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="margin-left: -1rem">
    <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;Table of Contents</p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>
  </div>
</div>





  



  



  



  



  






    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">Search</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">Keyword</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <span>banner</span> <i class="iconfont icon-love"></i> <span>2021-2022©</span> 
    </div>
  
  
    <div class="statistics">
  
  

  
    
      <span id="leancloud-site-pv-container" style="display: none">
        总访问量 
        <span id="leancloud-site-pv"></span>
         次
      </span>
    
    
      <span id="leancloud-site-uv-container" style="display: none">
        总访客数 
        <span id="leancloud-site-uv"></span>
         人
      </span>
    
    

  
</div>

  
  
  
</div>

    <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
    </div>
  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.0/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/GeekFocus/js/events.js" ></script>
<script  src="/GeekFocus/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>






  
    <script  src="/GeekFocus/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.18.2/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      headingSelector : CONFIG.toc.headingSelector || 'h1,h2,h3,h4,h5,h6',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      collapseDepth   : CONFIG.toc.collapseDepth || 0,
      scrollSmooth    : true,
      headingsOffset  : -boardTop
    });
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }
  });
</script>


  <script>
  (function() {
    var enableLang = CONFIG.code_language.enable && CONFIG.code_language.default;
    var enableCopy = CONFIG.copy_btn;
    if (!enableLang && !enableCopy) {
      return;
    }

    function getBgClass(ele) {
      return Fluid.utils.getBackgroundLightness(ele) >= 0 ? 'code-widget-light' : 'code-widget-dark';
    }

    var copyTmpl = '';
    copyTmpl += '<div class="code-widget">';
    copyTmpl += 'LANG';
    copyTmpl += '</div>';
    jQuery('.markdown-body pre').each(function() {
      var $pre = jQuery(this);
      if ($pre.find('code.mermaid').length > 0) {
        return;
      }
      if ($pre.find('span.line').length > 0) {
        return;
      }

      var lang = '';

      if (enableLang) {
        lang = CONFIG.code_language.default;
        if ($pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2 && $pre.children().hasClass('hljs')) {
          lang = $pre[0].children[0].classList[1];
        } else if ($pre[0].getAttribute('data-language')) {
          lang = $pre[0].getAttribute('data-language');
        } else if ($pre.parent().hasClass('sourceCode') && $pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2) {
          lang = $pre[0].children[0].classList[1];
          $pre.parent().addClass('code-wrapper');
        } else if ($pre.parent().hasClass('markdown-body') && $pre[0].classList.length === 0) {
          $pre.wrap('<div class="code-wrapper"></div>');
        }
        lang = lang.toUpperCase().replace('NONE', CONFIG.code_language.default);
      }
      $pre.append(copyTmpl.replace('LANG', lang).replace('code-widget">',
        getBgClass($pre[0]) + (enableCopy ? ' code-widget copy-btn" data-clipboard-snippet><i class="iconfont icon-copy"></i>' : ' code-widget">')));

      if (enableCopy) {
        Fluid.utils.createScript('https://lib.baomitu.com/clipboard.js/2.0.10/clipboard.min.js', function() {
          var clipboard = new window.ClipboardJS('.copy-btn', {
            target: function(trigger) {
              var nodes = trigger.parentNode.childNodes;
              for (var i = 0; i < nodes.length; i++) {
                if (nodes[i].tagName === 'CODE') {
                  return nodes[i];
                }
              }
            }
          });
          clipboard.on('success', function(e) {
            e.clearSelection();
            e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-copy', 'icon-success');
            setTimeout(function() {
              e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-success', 'icon-copy');
            }, 2000);
          });
        });
      }
    });
  })();
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/GeekFocus/js/local-search.js" ></script>

  <script defer src="/GeekFocus/js/leancloud.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/GeekFocus/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">Blog works best with JavaScript enabled</div>
  </noscript>
</body>
</html>
