

<!DOCTYPE html>
<html lang="en" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/GeekFocus/img/fluid.png">
  <link rel="icon" href="/GeekFocus/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="banner">
  <meta name="keywords" content="Bioinformatics">
  
    <meta name="description" content="Genome survey">
<meta property="og:type" content="article">
<meta property="og:title" content="Genome Survey">
<meta property="og:url" content="https://wangc1106.github.io/GeekFocus/2022/02/23/2022-02-21-Survey/index.html">
<meta property="og:site_name" content="DrBanner">
<meta property="og:description" content="Genome survey">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://upload-images.jianshu.io/upload_images/11004598-2ca590314cf3d635.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/374/format/webp">
<meta property="og:image" content="http://www.biomarker.com.cn/wp-content/uploads/2017/03/5-12.png">
<meta property="article:published_time" content="2022-02-23T15:03:53.000Z">
<meta property="article:modified_time" content="2022-06-05T12:06:21.244Z">
<meta property="article:author" content="banner">
<meta property="article:tag" content="assembly">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="https://upload-images.jianshu.io/upload_images/11004598-2ca590314cf3d635.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/374/format/webp">
  
  
  
  <title>Genome Survey - DrBanner</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/GeekFocus/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/GeekFocus/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/GeekFocus/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/GeekFocus/","version":"1.9.1","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"right","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":false,"follow_dnt":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":"https://wangc1106.github.io/GeekFocus","path":"window.location.pathname","ignore_local":false}},"search_path":"/GeekFocus/local-search.xml"};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/GeekFocus/js/utils.js" ></script>
  <script  src="/GeekFocus/js/color-schema.js" ></script>

  
<meta name="generator" content="Hexo 6.2.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 70vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/GeekFocus/">
      <strong>DrBanner</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/">
                <i class="iconfont icon-home-fill"></i>
                Home
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/archives/">
                <i class="iconfont icon-archive-fill"></i>
                Archives
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/categories/">
                <i class="iconfont icon-category-fill"></i>
                Categories
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/tags/">
                <i class="iconfont icon-tags-fill"></i>
                Tags
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/about/">
                <i class="iconfont icon-user-fill"></i>
                About
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              &nbsp;<i class="iconfont icon-search"></i>&nbsp;
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/GeekFocus/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="Genome Survey"></span>
          
        </div>

        
          
  <div class="mt-3">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-author" aria-hidden="true"></i>
        banner
      </span>
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2022-02-23 23:03" pubdate>
          February 23, 2022 pm
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          8.9k words
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          75 mins
        
      </span>
    

    
    
      
        <span id="leancloud-page-views-container" class="post-meta" style="display: none">
          <i class="iconfont icon-eye" aria-hidden="true"></i>
          <span id="leancloud-page-views"></span> views
        </span>
        
      
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">Genome Survey</h1>
            
              <p class="note note-info">
                
                  
                    Last updated on June 5, 2022 pm
                  
                
              </p>
            
            <div class="markdown-body">
              
              <h1 id="Genome-survey"><a href="#Genome-survey" class="headerlink" title="Genome survey"></a>Genome survey</h1><span id="more"></span>

<p>基因组杂合度和重复序列对后续基因组组装有很大影响。</p>
<p><strong>高杂合</strong>基因组往往无法合并姊妹染色体，导致组装结果<strong>偏大</strong>。</p>
<p>而<strong>重复序列</strong>在组装中被折叠，使组装中出现缺口、错误，导致组装结果<strong>偏小</strong>。</p>
<p>不同的生物基因组之间杂合率和重复序列含量差异巨大，因此在基因组测序前需要对基因组特征进行调研，以确定测序方案，周期等。</p>
<p>目前常用的调研手段有三种：</p>
<ol>
<li>用流式细胞仪测定细胞核内的<strong>DNA总量</strong></li>
<li>用核型分析方法，识别<strong>染色体数量、倍性</strong></li>
<li>用调研图，通过二代测序，估算<strong>基因组大小、杂合度、重复序列比例、GC含量</strong>等。</li>
</ol>
<p>不同技术手段不同侧重，其中<font color="red">调研图</font>以<font color="green">低成本，低难度和更多的评估内容</font>使用最多，同时<font color="red"><strong>Survey所测二代数据还可用于回比基因组，评估组装质量</strong></font>。调研图<font color="green">基于数学统计学手段</font>获取物种信息的方式，因此对于已经研究的较为清晰的物种——主要是<font color="green"><strong>普通二倍体和简单多倍体</strong></font>，其<font color="green">染色体条数、倍性、大概基因组大小是已知</font>，此时仅选择<strong>survey</strong>足以满足基因组特征需要，但对于<strong>多倍体复杂基因组</strong>更推荐<strong>补充核型分析和流式</strong>，以和调研图相互印证补充。</p>
<hr>
<p><strong>Survey 方案</strong></p>
<p>通过质控、NT比对，获得高质量的clean data，为后续分析奠定良好基础；</p>
<p>基因组Survey基于小片段文库的低深度测序数据（50X）左右；</p>
<p>通过K-mer分析，有效的评估基因组大小、GC含量、杂合度以及重复序列的含量等信息；</p>
<p>全面了解某一物种基因组特征的有效方法；</p>
<p>为后续的全基因denove测序的组装策略的制定提供理论依据。</p>
<p><strong>分析内容</strong>：污染程度评估，基因组大小评估，杂合度评估，重复率评估，GC含量评估。【初步了解三代组装难度和预期大小】</p>
<p><strong>基因组复杂度预估?</strong></p>
<blockquote>
<p>普通基因组的定义？<br>单倍体、纯合二倍体或者杂合度&lt;0.5%，且重复序列含量&lt;50%，GC含量为35%到65%之间的二倍体。</p>
<p>复杂基因组的定义？<br>杂合度&gt;0.5%，重复序列含量&gt;50%，多倍体，GC含量处于异常的范围（GC含量&lt;35%或者GC含量&gt;65%的二倍体）。</p>
<p><font color="red">二倍体复杂基因组进一步细分为</font><br><font color="red"> 微杂合基因组（0.5%&lt;杂合率&lt;&#x3D;0.8%)</font><br><font color="red"> 高杂合基因组（杂合率&gt;0.8%)</font><br> <font color="red">高重复基因组（重复序列比例&gt;50%）</font></p>
<p>基因组大小：<br> 基因组越大，测序花钱越多</p>
</blockquote>
<blockquote>
<ul>
<li>简单基因组: 杂合度低于0.5%, GC含量在35%~65%, 重复序列低于50%</li>
<li>二倍体普通基因组: 杂合度在0.5%~1.2%中间，重复序列低于50%。或杂合度低于0.5%，重复序列低于65%</li>
<li>高复杂基因组: 杂合度&gt;1.2% 或 重复率大于65%</li>
</ul>
</blockquote>
<hr>
<h1 id="质控"><a href="#质控" class="headerlink" title="质控"></a>质控</h1><p><img src="https://upload-images.jianshu.io/upload_images/11004598-2ca590314cf3d635.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/374/format/webp" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<p>各提出10,000对比对到<strong>NT库</strong>，如果都比对到同源物种，说明无污染，如果比对到细菌真菌，可能数据有污染。</p>
<h2 id="rawdata-qc"><a href="#rawdata-qc" class="headerlink" title="rawdata qc"></a>rawdata qc</h2><h2 id="trim"><a href="#trim" class="headerlink" title="trim"></a>trim</h2><h2 id="cleandata-qc"><a href="#cleandata-qc" class="headerlink" title="cleandata qc"></a>cleandata qc</h2><h2 id="NT数据库比对"><a href="#NT数据库比对" class="headerlink" title="NT数据库比对"></a>NT数据库比对</h2><p>Partially non-redundant nucleotide from all traditional divisions of GenBank, EMBL, and DDBJ excluding GSS,STS, PAT, EST, HTG, and WGS.</p>
<p><a target="_blank" rel="noopener" href="https://ftp.ncbi.nih.gov/blast/db/FASTA/">NT库</a></p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs sh">blastn \<br>-query reads_2.fa \<br>-db /local_data1/public_data/database/genome_DB/nt/20190417_all/nt \<br>-out reads_2.csv \<br>-outfmt <span class="hljs-string">&quot;10 evalue length qseqid qlen qstart qend sacc slen sstart send pident nident sstrand qcovs qseq sseq sgi stitle&quot;</span> \<br>-num_threads 4 -evalue 1e-5 -max_target_seqs 1<br></code></pre></td></tr></table></figure>



<hr>
<p><a target="_blank" rel="noopener" href="https://blog.csdn.net/u012110870/article/details/82500748">https://blog.csdn.net/u012110870/article/details/82500748</a></p>
<h1 id="k-mers"><a href="#k-mers" class="headerlink" title="k-mers"></a>k-mers</h1><p>最简单的策略就是基于k-mer对基因组做一个简单的了解, 使用jellyfish统计k-mers，然后作图</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><code class="hljs sh">jellyfish count  -m 21 -s 20G -t 20 -o 21mer_out  -C --min-qual-char=? &lt;(zcat test_1.fq.gz) &lt;(zcat test_2.fq.gz)<br><span class="hljs-comment"># -m k-mers的K</span><br><span class="hljs-comment"># -s Hash大小, 根据文件大小确定</span><br><span class="hljs-comment"># -t 线程</span><br><span class="hljs-comment"># -o 输出前缀</span><br><span class="hljs-comment"># -C 统计正负链</span><br><span class="hljs-comment"># --min-qual-char 过滤低质量碱基。 在Phred+33中,?=30</span><br>jellyfish histo -o 21mer_out.histo 21mer_out<br></code></pre></td></tr></table></figure>

<p>用R作图</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs stylus"><span class="hljs-function"><span class="hljs-title">pdf</span><span class="hljs-params">(<span class="hljs-string">&quot;21_mer.out.pdf&quot;</span>)</span></span><br>dataframe19 &lt;- read<span class="hljs-selector-class">.table</span>(<span class="hljs-string">&quot;21mer_out.histo&quot;</span>)<br><span class="hljs-function"><span class="hljs-title">plot</span><span class="hljs-params">(dataframe19[<span class="hljs-number">1</span>:<span class="hljs-number">200</span>,], type=<span class="hljs-string">&quot;l&quot;</span>)</span></span><br>dev<span class="hljs-selector-class">.off</span>()<br></code></pre></td></tr></table></figure>

<p>由于只有一个主峰，说明该物种的杂合度并不高，基本上也就是二倍体。如果图中出现多个峰，说明它可能是多倍体或者是基因组杂合度高。</p>
<p>推荐文献: Genomic DNA k-mer spectra: models and modalities</p>
<h1 id="基于组装"><a href="#基于组装" class="headerlink" title="基于组装"></a>基于组装</h1><p>基于K-mers可以较好的预测基因组大小，并定性的了解基因组的复杂情况，如果想更具体的了解基因组的复杂度，可以先将50X以上的段片段进行组装，然后进行分析。</p>
<p>组装的工具比较多，推荐用SOAPdenovo，因为速度快。abyss?</p>
<p>新建一个contig.config, 增加如下内容</p>
<figure class="highlight ini"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><code class="hljs ini"><span class="hljs-attr">max_rd_len</span>=<span class="hljs-number">150</span><br><span class="hljs-section">[LIB]</span><br><span class="hljs-attr">avg_ins</span>=<span class="hljs-number">200</span><br><span class="hljs-attr">reverse_seq</span>=<span class="hljs-number">0</span><br><span class="hljs-attr">asm_flags</span>=<span class="hljs-number">3</span><br><span class="hljs-attr">rd_len_cutoff</span>=<span class="hljs-number">100</span><br><span class="hljs-attr">rank</span>=<span class="hljs-number">1</span><br><span class="hljs-attr">pair_num_cutoff</span>=<span class="hljs-number">3</span><br><span class="hljs-attr">map_len</span>=<span class="hljs-number">32</span><br><span class="hljs-attr">q1</span>=read_1.fq<br><span class="hljs-attr">q2</span>=read_2.fq<br></code></pre></td></tr></table></figure>

<p>组装出参考序列</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs awk">~<span class="hljs-regexp">/opt/</span>biosoft<span class="hljs-regexp">/SOAPdenovo2/</span>SOAPdenovo-<span class="hljs-number">63</span>mer all -s contig.config -R -K <span class="hljs-number">63</span> -p <span class="hljs-number">30</span> -o assembly/graph<br></code></pre></td></tr></table></figure>

<p>最后graph.scafSeq是拼接后的序列, 提取出大于300bp的序列.</p>
<figure class="highlight stata"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs stata"># <span class="hljs-keyword">adjust</span> <span class="hljs-keyword">format</span><br>bioawk -c fastx -v name=1 &#x27;&#123;<span class="hljs-keyword">if</span>(<span class="hljs-built_in">length</span>(<span class="hljs-variable">$seq</span>)&gt;300) <span class="hljs-keyword">print</span> <span class="hljs-string">&quot;&gt;&quot;</span>name <span class="hljs-string">&quot;\n&quot;</span> <span class="hljs-variable">$seq</span>;name+=1&#125;&#x27; assembly/<span class="hljs-keyword">graph</span>.scafSeq &gt;contig.fa<br></code></pre></td></tr></table></figure>

<h1 id="杂合度估计"><a href="#杂合度估计" class="headerlink" title="杂合度估计"></a>杂合度估计</h1><p>将原来的序列回贴到contig上，并用samtools+bcftools进行snp calling.统计变异的碱基占总体的比例。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs perl"><span class="hljs-keyword">mkdir</span> -p <span class="hljs-keyword">index</span><br>bwa <span class="hljs-keyword">index</span> contig.fa -p <span class="hljs-keyword">index</span>/contig<br>bwa mem -v <span class="hljs-number">2</span> -t <span class="hljs-number">10</span> <span class="hljs-keyword">index</span>/contig read_1.fq read_2.f<span class="hljs-string">q | samtools sort -n &gt; align.bam</span><br><span class="hljs-string">samtools mpileup -f contig align.bam |</span> bcftools call -mv -Oz -o variants.gz<br></code></pre></td></tr></table></figure>

<p>一方面由于SOAPdenovo组装过程中会出错, 另一方面samtools在变异检测上也存在很高的假阳性, 所以总得先按照深度和质量过滤一批假阳性。</p>
<figure class="highlight 1c"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs 1c">bcftools view -i &#x27; DP &gt; 30 &amp;&amp; MQ &gt; 30&#x27; -H variants.vcf.gz <span class="hljs-string">| wc -l</span><br><span class="hljs-meta"># 325219, 无过滤是445113</span><br></code></pre></td></tr></table></figure>

<p>变异数目占基因组大小的比例就是杂合度。我的contig大概是200M，找到0.3M左右的变异，也就是0.0015，即0.15%.</p>
<h1 id="重复序列估计"><a href="#重复序列估计" class="headerlink" title="重复序列估计"></a>重复序列估计</h1><p>基于同源注释，用RepeatMasker寻找重复序列. 这里要注意分析的fasta的ID不能过长，也就是最好是<code>&gt;scaffold_1</code>这种形式，不然会报错。</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs awk">~<span class="hljs-regexp">/opt/</span>biosoft<span class="hljs-regexp">/RepeatMsker/</span>RepeatMasker -e ncbi -species arabidopsis -pa <span class="hljs-number">10</span> -gff -dir ./ contig.fa<br><span class="hljs-comment"># -e ncbi</span><br><span class="hljs-comment"># -species 选择物种 用~/opt/biosoft/RepeatMasker/util/queryRepeatDatabase.pl -tree 了解</span><br><span class="hljs-comment"># -pa 并行计算</span><br><span class="hljs-comment"># -gff 输出gff注释</span><br><span class="hljs-comment"># -dir 输出路径</span><br></code></pre></td></tr></table></figure>

<p>输出结果中主要关注如下三个</p>
<ul>
<li>output.fa.masked, 将重复序列用N代替</li>
<li>output.fa.out.gff, 以gff2形式存放重复序列出现的位置</li>
<li>output.fa.tbl, 该文件记录着分类信息</li>
</ul>
<figure class="highlight asciidoc"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs asciidoc">==================================================<br>file name: anno.fasta<br>sequences:         62027<br>total length:  273135210 bp  (273135210 bp excl N/X-runs)<br>GC level:         36.80 %<br>bases masked:   79642191 bp ( 29.16 %)<br>==================================================<br></code></pre></td></tr></table></figure>

<p>也就是说我们的物种有30%的重复序列，作为参考，拟南芥125Mb 14%重复序列, 水稻389M，36%重复</p>
<hr>
<h1 id="Grand-Omics"><a href="#Grand-Omics" class="headerlink" title="Grand Omics"></a><font color="green">Grand Omics</font></h1><p>快速：K-mer分析方法，可快速确定基因组特征</p>
<p>专业：简单基因组和复杂基因组（高杂合、多倍体等）采用不同算法软件</p>
<p>建库：Illumina PE150普通文库；或MGI PE150 普通文库。总量 ≥ 1μg；推荐数据量 ≥ 50X</p>
<p>分析内容：原始数据质控；数据污染检测；基因组大小预估；基因组杂合度预估</p>
<h2 id="数据质控及污染评估"><a href="#数据质控及污染评估" class="headerlink" title="数据质控及污染评估"></a>数据质控及污染评估</h2><p>对原始数据质控过滤，然后对原始数据（Raw data）及过滤后数据（Clean data）进行质量统计。随机取100,000条质控后的reads，统计reads在<font color="red">nt库</font>中的分布情况及比对上的物种分布，以评估数据污染情况。</p>
<h2 id="基因组大小和杂合度估计"><a href="#基因组大小和杂合度估计" class="headerlink" title="基因组大小和杂合度估计"></a>基因组大小和杂合度估计</h2><p>利用<font color="red">偏正态分布模型（skew normal distribution model）</font>，<font color="red">负二项式模型 (negative binomial model)</font> 对K-mer 数据进行拟合分析，进行基因组大小以及杂合度的评估，并生成最终基因组评估结果。</p>
<p>一般选择<strong>17-mer</strong>评估基因组大小，ATCG四种碱基组成长度为17的核苷酸片段有4(17次方)<del>17G，足以覆盖一般的正常基因组；15mer</del>1G。对于正常基因组可能覆盖度不够，导致估计不准确。&gt;15G，采用19-mer或21-mer。K-mer越大，包含错误位点的K-mer越多。为了<strong>避免回文序列，K为奇数</strong>。</p>
<h2 id="GC-depth-分析及污染评估"><a href="#GC-depth-分析及污染评估" class="headerlink" title="GC-depth 分析及污染评估"></a>GC-depth 分析及污染评估</h2><p>对<font color="red">组装的基因组序列</font>以<font color="red">5kb为windows</font>，无重复计算片段的<font color="red">平均GC含量和平均深度</font>并作图。基于每一个windows对应的平均GC和平均深度进行绘图。可以根据此图分析测序数据是否存在GC偏向性以及样本是否存在污染。</p>
<h1 id="Biomarker"><a href="#Biomarker" class="headerlink" title="Biomarker"></a><font color="green">Biomarker</font></h1><p>基因组survey以测序技术为基础，基于小片段文库的低深度测序，通过K-mer分析，快速获得基因组大小、杂合度、重复序列比例等基本信息，为制定该物种的全基因组de novo测序策略提供有效依据。</p>
<h2 id="调研图分析原理"><a href="#调研图分析原理" class="headerlink" title="调研图分析原理"></a>调研图分析原理</h2><p><img src="http://www.biomarker.com.cn/wp-content/uploads/2017/03/5-12.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="5"></p>
<p>k-mer，将核酸序列以滑窗分成包含k个碱基的短序列，“mer”这个单词的来源<font color="red">monomeric unit，单体单元</font>。K一般为奇数（避免正反链混淆?）。统计所有reads中所出现的k-mer类型及各类型k-mer的深度（或者频率），绘制特定k-mer下不同深度k-mer片段的频数统计图，通常选择K-mer分布最多的峰为主峰，从而得到<font color="red"><strong>基因组大小&#x3D;K-mer总数&#x2F;K-mer主峰深度值</strong></font>。</p>
<p>由于基因组存在<font color="red">杂合位点和重复序列</font>，k-mer曲线不是标准泊松分布，在主峰前后出现其他的峰，如果存在一定<font color="red">杂合度</font>，会导致在<font color="red">主峰横坐标的二分之一处</font>出现<font color="red">杂合峰</font>，而一定的<font color="red">重复度</font>则会在<font color="red">主峰对应的横坐标整数倍</font>处出现<font color="red">重复峰</font>。</p>
<h2 id="调研图分析内容"><a href="#调研图分析内容" class="headerlink" title="调研图分析内容"></a>调研图分析内容</h2><p>评估基因组<font color="red">大小</font>；评估基因组<font color="red">杂合</font>情况；评估<font color="red">重复序列</font>含量；评估基因组<font color="red">GC含量</font>；为后续精细图阶段的文库构建提供策略建议。</p>
<h2 id="基因组调研图survey的意义"><a href="#基因组调研图survey的意义" class="headerlink" title="基因组调研图survey的意义"></a>基因组调研图survey的意义</h2><p>启动全基因组测序的<font color="red">必要前提</font></p>
<p>了解与<font color="red"><strong>近缘物种间</strong>的<strong>基因组差异</strong>信息</font></p>
<p>获得某<font color="red">物种基因组的基本信息及难易程度</font></p>
<h1 id="Novogene"><a href="#Novogene" class="headerlink" title="Novogene"></a><font color="green">Novogene</font></h1><p>方案：DNA10 ug；文库350bp；PE150测序，测序深度为50X</p>
<h2 id="分析内容"><a href="#分析内容" class="headerlink" title="分析内容"></a>分析内容</h2><ul>
<li>K-mer分析</li>
<li>预估物种基因组大小</li>
<li>GC含量和外源污染分析</li>
<li>杂合度评估</li>
<li>重复序列评估</li>
<li>初步组装</li>
<li>基于初步组装的版本，可以进行SSR标记开发和同源注释【highlevel】simple sequence repeats</li>
</ul>
<h1 id="BerryGenomics"><a href="#BerryGenomics" class="headerlink" title="BerryGenomics"></a><font color="green">BerryGenomics</font></h1><p>基于小片段文库低深度测序数据，通过K-mer分析，有效评估基因组大小、GC含量、杂合度高低及重复序列含量等信息，为后续的组装策略的制定提供理论依据。</p>
<p><a target="_blank" rel="noopener" href="https://pdf.sciencedirectassets.com/272196/1-s2.0-S0092867419X0015X/1-s2.0-S0092867420306188/main.pdf?X-Amz-Security-Token=IQoJb3JpZ2luX2VjEOj//////////wEaCXVzLWVhc3QtMSJHMEUCIEP3b6JimIkwVtoSmPnQ7fbUGBl4eNBBDDnBn4Sgw6y2AiEA9ob5g/J4zQrFLsTKjhs51/F/1tP61OMYZBAruOkb0fgq+gMIQRAEGgwwNTkwMDM1NDY4NjUiDMQhUFOWJRI31LL0UyrXA4jlyI72DidSpx8zf2iw85fV3Vmv5GwQ6zAU0go3NjgbvnEQeGHSyjCD3bfuIdLxx9+JR3iu8nTJwgeiw9+Pub8vYWrjzALVOyDRYnN4C6P68gzwDtI75rViTKSOYGpm0LXGHIuNV3O6NG9GNYo9nKQUdsmQfjBcBVHGBhXgRhFbmvWhLVufPeyZIY/sw6wiX3diCmbAuE7tTj4O0tA4ht9m/9y/rMizkGh9ZPtbzZ7JDkZm0+cpNEzpq3cQrGGNFrTcYFMQLf9YYSiY5LNr+zDXEI/cud8Oj8u3J75X7AtKXVgDFx3WOLQId9J+kxO1tSZ5/glrj2Mrth4nSXKk2TcSj6sirRM7+QbGIcYbhUHe+97GuZSfy6tJSD8POUgAUJZwoh9mtVgEIgRBR3uHjlgquc2QL4w5HsPkDpKmCRjT5ZIlJKfBli2DLWB81D9uot9Aw5SNzZ+mx3uR2mpTibb5v54gw7HtVunol7a5sq4awm3L8KRSRvUwk+fwX38TdFUH7OxZaFWIlBk536NC4ceJTACy5V5Ae+f7lhfpLN5wHzi3JomqhWFjfqqDLY11VHUBE2hJ2MtKtSQwilpBH+qtouIVYoOLFsKgJGfh0HvzkymqMQdxizDUhM2QBjqlAa9D4CUBq0ExkdHUsNdoKZ3Mz+0m1f72NRgmttBXR/CcVH6PUnSta5nZU2WQy6dUTWTx6pl8YHzW/DDLcwdU82BkQOE1n6CSzHGjaKA75v3T4RWRU9wOiObTdjqpmmMKTp2xohm4N78+/ZejYhLhk1Tx/CnUl2ue4z1/Utn17JSca5oVGbePorLRmqyYs5F3oL08SP73vwHxeNaoI7f7ucpUWqEcHA==&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20220221T081720Z&X-Amz-SignedHeaders=host&X-Amz-Expires=300&X-Amz-Credential=ASIAQ3PHCVTYYN624BKH/20220221/us-east-1/s3/aws4_request&X-Amz-Signature=16e8a26f9b71e59d421ff693fe49b021ce1222b688728bb93ce93ea82fb5968d&hash=ccc160836e462b394593db49a0b599b8a106fca7b355f59e9e05635b2a62f335&host=68042c943591013ac2b2430a89b270f6af2c76d8dfd086a07176afe7c76c2c61&pii=S0092867420306188&tid=spdf-0c4aa40e-20ae-4f30-8d50-b9cee2090b93&sid=5071d2f0595c814bfc9b6e825f0c6315e333gxrqa&type=client&ua=570456075759005851&rr=6e0e89a71f5d7c4a">大豆高质量泛基因组图谱构建</a></p>
<p>我国大豆消费对外依赖严重，而提高大豆产量则必须对大豆种质的遗传多样性进行综合评价。中科院遗传与发育生物学研究所等单位与贝瑞基因合作，利用最新组装策略构建了高质量的基于图形的泛基因组。该研究利用Illumina平台对来自世界大豆主产国的2,898份大豆种质进行重测序（13×），利用PacBio（96×）、Bionano和Hi-C技术对26份代表性种质进行基因组组装和泛基因组构建。并以ZH13基因组为基准基因组，整合PAVs后构建基于基因组图谱。将重测序数据重新比对到该基因组进行SVs检测和GWAS分析。在开花的主效位点<em>E3</em>发现了5种不同的单倍型；种皮颜色相关基因<em>CHS</em>的<em>I</em>位点发现5种主要单倍型；14号染色体上的缺铁褪绿症QTL内存在一个调控Fe2+&#x2F;Zn2+转运蛋白基因<em>SoyZH13_14G179600</em>。</p>
<hr>
<h1 id="Pipeline"><a href="#Pipeline" class="headerlink" title="Pipeline"></a><font color="green">Pipeline</font></h1><ul>
<li><p><strong>K-mer analysis</strong> (genome size,  repetive seq, heterozygous)</p>
<ul>
<li><p><a target="_blank" rel="noopener" href="http://www.cbcb.umd.edu/software/jellyfish/">JELLYFISH</a>是<a target="_blank" rel="noopener" href="http://www.cbcb.umd.edu/">CBCB(Center for Bioinformatics and Computational Biology)</a>的Guillaume Marçais 和 <a target="_blank" rel="noopener" href="http://www.cs.cmu.edu/~ckingsf/">Carl Kingsford</a> 研发的一款计数 DNA 的 k-mers 的软件。该软件运用 Hash 表来存储数据，同时能多线程运行，速度快，内存消耗小。该软件只能运行在64位的Linux系统下。其<a target="_blank" rel="noopener" href="http://bioinformatics.oxfordjournals.org/content/27/6/764">文章</a>于2011年发表在杂志 <a target="_blank" rel="noopener" href="http://bioinformatics.oxfordjournals.org/">Bioinformatics</a> 上。<a target="_blank" rel="noopener" href="https://www.cnblogs.com/zhanmaomao/p/9285582.html">https://www.cnblogs.com/zhanmaomao/p/9285582.html</a>,  <a target="_blank" rel="noopener" href="https://github.com/gmarcais/Jellyfish#:~:text=Jellyfish%20Overview%20Jellyfish%20is%20a%20tool%20for%20fast%2C,central%20step%20in%20many%20analyses%20of%20DNA%20sequence">https://github.com/gmarcais/Jellyfish#:~:text=Jellyfish%20Overview%20Jellyfish%20is%20a%20tool%20for%20fast%2C,central%20step%20in%20many%20analyses%20of%20DNA%20sequence</a>.</p>
</li>
<li><pre><code class="sh">$ wget 最新targz
$ tar zxvf .tar.gz
$ mkdir jellyfish
$ cd jellyfish-2.3.0
$ ./configure --prefix=Your/Path/to/jellyfish
如果安装在当前目录中，会报错。
$ make -j 8
$ make install
</code></pre>
</li>
</ul>
<table>
<thead>
<tr>
<th>jellyfish</th>
<th>Kmer</th>
<th>Clean</th>
<th>Insert</th>
<th>Depth</th>
<th>Kpeak</th>
<th>HetP</th>
<th>Heter</th>
<th>repeatP</th>
<th>Repeat</th>
<th>Size</th>
</tr>
</thead>
<tbody><tr>
<td>[4]</td>
<td>17</td>
<td>30.29G</td>
<td>220 bp</td>
<td>62.99X</td>
<td>54.98X</td>
<td>26X</td>
<td>0.18%</td>
<td>106X</td>
<td>291.49Mb?(60.60%)</td>
<td>480.97 Mb</td>
</tr>
<tr>
<td>[5]</td>
<td>21</td>
<td>54.23G</td>
<td>270bp</td>
<td>83.69×</td>
<td>69 ×</td>
<td>34 ×</td>
<td>0.72%</td>
<td>139 ×</td>
<td>43.22%</td>
<td>648.07 Mb</td>
</tr>
<tr>
<td>[6]?</td>
<td>17</td>
<td>89G</td>
<td>350bp</td>
<td></td>
<td>48x</td>
<td>24x</td>
<td>0.69%</td>
<td>96x</td>
<td><strong>91.05%</strong></td>
<td>1.86G</td>
</tr>
<tr>
<td>[7]</td>
<td>19[21,23,25,27]</td>
<td>54.99G</td>
<td>270bp</td>
<td>45.37x</td>
<td>38x</td>
<td>19x</td>
<td>0.33%</td>
<td>76x</td>
<td><strong>70.62%</strong></td>
<td>1.2G</td>
</tr>
<tr>
<td>[8]自源四倍</td>
<td>17</td>
<td>27G</td>
<td>220</td>
<td></td>
<td>57</td>
<td>28</td>
<td><strong>1.37%</strong></td>
<td>114</td>
<td>42.81%</td>
<td>800M</td>
</tr>
<tr>
<td>[9]</td>
<td>17</td>
<td></td>
<td></td>
<td></td>
<td>52</td>
<td>26</td>
<td><strong>2.28%</strong></td>
<td>104</td>
<td><strong>89.2%</strong></td>
<td>4159 M</td>
</tr>
</tbody></table>
<ul>
<li>[6] : K-mer analysis:1.8G; flow cytometry: 8.50 ± 0.38 Gb. 巨大偏差？杂合度0.69%，重复ratio 91.05%</li>
</ul>
</li>
<li><p><strong>assembly</strong></p>
<ul>
<li><p>[4] : soap devovo, abyss. </p>
<ul>
<li>K-mer 31, 54, 63, 70, 77 ,83. 结果只有77，105 Kmer soapdenovo； AByss K43</li>
<li>optimal k-mer size was selected from the N50 length</li>
<li>usable reads &gt; 200 bases were selected to realign the contig </li>
<li>paired-end relationship between reads was coincident between contigs</li>
<li>scaffolds were constructed using insert size paired-ends</li>
</ul>
</li>
<li><p>[5] :</p>
</li>
<li><p>[8] :conduct de novo assembly. K-mer sizes of 20, 37, 55, 63, 71, 77, 83, and 95 were examined using default parameters. Assembly with k-mer 63 by SOAP de novo was selected because it has the optimal reading for N50</p>
<table>
<thead>
<tr>
<th>SOAPdenovo2</th>
<th>K-mer</th>
<th>ConN50</th>
<th>N90</th>
<th>TotalCon</th>
<th>ScaN50</th>
<th>N90</th>
<th>TotalSca</th>
<th>Genome</th>
</tr>
</thead>
<tbody><tr>
<td>[4]</td>
<td>77</td>
<td>1.48k</td>
<td>236</td>
<td>405M</td>
<td>3.55k</td>
<td>375</td>
<td>409M</td>
<td>480.97 M</td>
</tr>
<tr>
<td>[5]</td>
<td>65</td>
<td>1.59k</td>
<td>379</td>
<td>449M</td>
<td>3.5k</td>
<td>1262</td>
<td>543M</td>
<td>648.07 M</td>
</tr>
<tr>
<td>[6]?</td>
<td>41</td>
<td>287</td>
<td>127</td>
<td></td>
<td>1138</td>
<td>150</td>
<td></td>
<td></td>
</tr>
<tr>
<td>[7]</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>191bp</td>
<td></td>
<td>1.37G</td>
<td>1.2G</td>
</tr>
<tr>
<td>[8]</td>
<td>63</td>
<td>1.08k</td>
<td>219</td>
<td>170M</td>
<td>1.96k</td>
<td>1110</td>
<td>173M</td>
<td>800M</td>
</tr>
<tr>
<td>[9]</td>
<td></td>
<td>533</td>
<td></td>
<td></td>
<td>778</td>
<td>119</td>
<td>4.5G</td>
<td>4159M</td>
</tr>
</tbody></table>
</li>
</ul>
</li>
<li><p>GC</p>
<ul>
<li>[4] mid GC: 38%,  too high (&gt;65%) and too low (&lt;25%)</li>
<li>[4] GC depth two layer: only one of the two sets of homologous chromosomes in the diploid was assembled, which resulted in the emergence of the lower layer</li>
<li>[5] GC depth map was divided into 2 layers (Fig. 2), which was mainly due to the high heterozygosity.</li>
<li><font color="red"><strong>找近缘的GC做比较；看是否GC分层</strong></font></li>
<li>[8] the GC depth was slightly blocked into 4 layers (Fig. 2), which was in part caused by the polyploidy and the 1.37% high heterozygosity rate. 自源四倍体</li>
</ul>
</li>
<li><p><strong>Repetitive sequences</strong></p>
<ul>
<li>[4] :<strong>LTR_FINDER, MITE-Hunter, RepeatScout, and PILER-DF, PASTEClassifier, Repeat Masker, SciRoKo</strong> <ul>
<li>repetitive sequences ~147.89 Mb, 47.55% of the genome,</li>
<li>class I transposable element; class II transposable element</li>
<li>SSR(167,859SSR);  motif types</li>
</ul>
</li>
<li>[5] : total of 851,957 SSRs were identified,<strong>MISA</strong>. <ul>
<li>245 motif types</li>
</ul>
</li>
<li>[7] : <strong>MIcro-SAtellite(MISA)</strong><ul>
<li>Identification and characterisation of SSR motifs.</li>
<li>SSR marker verification</li>
</ul>
</li>
</ul>
</li>
<li><p><strong>Gene predition &amp; annotation</strong></p>
</li>
</ul>
<hr>
<p>ref</p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/092103a24ce4">https://www.jianshu.com/p/092103a24ce4</a></p>
<p><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/366933242">https://zhuanlan.zhihu.com/p/366933242</a></p>
<p>[4] Genome Survey Sequencing for the Characterization of the Genetic Background of Rosa roxburghii Tratt and Leaf Ascorbate Metabolism Genes [Plos one]</p>
<p>[5] Genome survey sequencing and genetic diversity of cultivated <em>Akebia trifoliata</em> assessed via phenotypes and SSR markers [Molecular Biology Reports <strong>2.3IF</strong>]</p>
<p>[6] : Genome survey sequencing of red swamp crayfish <em>Procambarus clarkii</em> [<strong>2.3IF</strong>] </p>
<p>[7] Genome survey sequencing and characterization of simple sequence repeat (SSR) markers in Platostoma palustre (Blume) A.J.Paton (Chinese mesona) [scientific reports <strong>4.3IF</strong>]</p>
<p>[8] Genome survey sequencing of *Dioscorea zingiberensis [genome <strong>2.1IF</strong>]</p>
<p>[9]Genome survey sequencing of Atractylodes lancea and identification of its SSR markers[Bioscience Reports, IF3.84]</p>

              
            </div>
            <hr/>
            <div>
              <div class="post-metas my-3">
  
    <div class="post-meta mr-3 d-flex align-items-center">
      <i class="iconfont icon-category"></i>
      

<span class="category-chains">
  
  
    
      <span class="category-chain">
        
  <a href="/GeekFocus/categories/Biology/" class="category-chain-item">Biology</a>
  
  

      </span>
    
  
</span>

    </div>
  
  
    <div class="post-meta">
      <i class="iconfont icon-tags"></i>
      
        <a href="/GeekFocus/tags/assembly/">#assembly</a>
      
    </div>
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>Genome Survey</div>
      <div>https://wangc1106.github.io/GeekFocus/2022/02/23/2022-02-21-Survey/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>Author</div>
          <div>banner</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>Posted on</div>
          <div>February 23, 2022</div>
        </div>
      
      
      <div class="license-meta-item">
        <div>Licensed under</div>
        <div>
          
            
            
              <a target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
              <span class="hint--top hint--rounded" aria-label="BY - Attribution">
                <i class="iconfont icon-by"></i>
              </span>
              </a>
            
          
        </div>
      </div>
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/GeekFocus/2022/02/24/2022-02-24-memory/" title="Linux 内存 存储">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">Linux 内存 存储</span>
                        <span class="visible-mobile">Previous</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/GeekFocus/2022/02/23/2022-02-23-jellyfish-genomescope/" title="jellyfish &amp; Genomescope">
                        <span class="hidden-mobile">jellyfish &amp; Genomescope</span>
                        <span class="visible-mobile">Next</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="margin-left: -1rem">
    <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;Table of Contents</p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>
  </div>
</div>





  



  



  



  



  






    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">Search</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">Keyword</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <span>banner</span> <i class="iconfont icon-love"></i> <span>2021-2022©</span> 
    </div>
  
  
    <div class="statistics">
  
  

  
    
      <span id="leancloud-site-pv-container" style="display: none">
        总访问量 
        <span id="leancloud-site-pv"></span>
         次
      </span>
    
    
      <span id="leancloud-site-uv-container" style="display: none">
        总访客数 
        <span id="leancloud-site-uv"></span>
         人
      </span>
    
    

  
</div>

  
  
  
</div>

    <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
    </div>
  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.0/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/GeekFocus/js/events.js" ></script>
<script  src="/GeekFocus/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>






  
    <script  src="/GeekFocus/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.18.2/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      headingSelector : CONFIG.toc.headingSelector || 'h1,h2,h3,h4,h5,h6',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      collapseDepth   : CONFIG.toc.collapseDepth || 0,
      scrollSmooth    : true,
      headingsOffset  : -boardTop
    });
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }
  });
</script>


  <script>
  (function() {
    var enableLang = CONFIG.code_language.enable && CONFIG.code_language.default;
    var enableCopy = CONFIG.copy_btn;
    if (!enableLang && !enableCopy) {
      return;
    }

    function getBgClass(ele) {
      return Fluid.utils.getBackgroundLightness(ele) >= 0 ? 'code-widget-light' : 'code-widget-dark';
    }

    var copyTmpl = '';
    copyTmpl += '<div class="code-widget">';
    copyTmpl += 'LANG';
    copyTmpl += '</div>';
    jQuery('.markdown-body pre').each(function() {
      var $pre = jQuery(this);
      if ($pre.find('code.mermaid').length > 0) {
        return;
      }
      if ($pre.find('span.line').length > 0) {
        return;
      }

      var lang = '';

      if (enableLang) {
        lang = CONFIG.code_language.default;
        if ($pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2 && $pre.children().hasClass('hljs')) {
          lang = $pre[0].children[0].classList[1];
        } else if ($pre[0].getAttribute('data-language')) {
          lang = $pre[0].getAttribute('data-language');
        } else if ($pre.parent().hasClass('sourceCode') && $pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2) {
          lang = $pre[0].children[0].classList[1];
          $pre.parent().addClass('code-wrapper');
        } else if ($pre.parent().hasClass('markdown-body') && $pre[0].classList.length === 0) {
          $pre.wrap('<div class="code-wrapper"></div>');
        }
        lang = lang.toUpperCase().replace('NONE', CONFIG.code_language.default);
      }
      $pre.append(copyTmpl.replace('LANG', lang).replace('code-widget">',
        getBgClass($pre[0]) + (enableCopy ? ' code-widget copy-btn" data-clipboard-snippet><i class="iconfont icon-copy"></i>' : ' code-widget">')));

      if (enableCopy) {
        Fluid.utils.createScript('https://lib.baomitu.com/clipboard.js/2.0.10/clipboard.min.js', function() {
          var clipboard = new window.ClipboardJS('.copy-btn', {
            target: function(trigger) {
              var nodes = trigger.parentNode.childNodes;
              for (var i = 0; i < nodes.length; i++) {
                if (nodes[i].tagName === 'CODE') {
                  return nodes[i];
                }
              }
            }
          });
          clipboard.on('success', function(e) {
            e.clearSelection();
            e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-copy', 'icon-success');
            setTimeout(function() {
              e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-success', 'icon-copy');
            }, 2000);
          });
        });
      }
    });
  })();
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/GeekFocus/js/local-search.js" ></script>

  <script defer src="/GeekFocus/js/leancloud.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/GeekFocus/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">Blog works best with JavaScript enabled</div>
  </noscript>
</body>
</html>
