

<!DOCTYPE html>
<html lang="en" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/GeekFocus/img/fluid.png">
  <link rel="icon" href="/GeekFocus/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="banner">
  <meta name="keywords" content="Bioinformatics">
  
    <meta name="description" content="主流三代测序平台Oxford 的 Nanopore，Pacific Biosciences（ PacBio）的 Single Molecule Real-Time（SMRT）Sequencing">
<meta property="og:type" content="article">
<meta property="og:title" content="HiFi-2-SMRT Sequencing - HiFi Reads [High fidelity reads]">
<meta property="og:url" content="https://wangc1106.github.io/GeekFocus/2022/02/20/2022-02-20-HiFi-2-SMRT/index.html">
<meta property="og:site_name" content="DrBanner">
<meta property="og:description" content="主流三代测序平台Oxford 的 Nanopore，Pacific Biosciences（ PacBio）的 Single Molecule Real-Time（SMRT）Sequencing">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/1.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/1.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/2.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/2.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/3.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/3.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/4.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/4.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/5.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/5.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/6.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/6.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/7.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/7.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/8.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/8.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/9.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/9.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/10.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/10.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/12.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/12.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/13.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/13.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/15.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/15.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/16.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/16.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/17.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/17.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/31.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/31.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/19.gif">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/19.gif">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/20.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/20.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/29.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/29.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/30.png">
<meta property="og:image" content="https://wangc1106.github.io/GeekFocus/img/2022-02-20-HiFi-2-SMRT/30.png">
<meta property="article:published_time" content="2022-02-20T14:11:03.547Z">
<meta property="article:modified_time" content="2022-06-08T07:11:02.421Z">
<meta property="article:author" content="banner">
<meta property="article:tag" content="assembly">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="https://wangc1106.github.io/GeekFocus/2022-02-20-HiFi-2-SMRT/1.png">
  
  
  
  <title>HiFi-2-SMRT Sequencing - HiFi Reads [High fidelity reads] - DrBanner</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/GeekFocus/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/GeekFocus/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/GeekFocus/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/GeekFocus/","version":"1.9.1","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"right","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":false,"follow_dnt":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":"https://wangc1106.github.io/GeekFocus","path":"window.location.pathname","ignore_local":false}},"search_path":"/GeekFocus/local-search.xml"};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/GeekFocus/js/utils.js" ></script>
  <script  src="/GeekFocus/js/color-schema.js" ></script>

  
<meta name="generator" content="Hexo 6.2.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 70vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/GeekFocus/">
      <strong>DrBanner</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/">
                <i class="iconfont icon-home-fill"></i>
                Home
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/archives/">
                <i class="iconfont icon-archive-fill"></i>
                Archives
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/categories/">
                <i class="iconfont icon-category-fill"></i>
                Categories
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/tags/">
                <i class="iconfont icon-tags-fill"></i>
                Tags
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/about/">
                <i class="iconfont icon-user-fill"></i>
                About
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              &nbsp;<i class="iconfont icon-search"></i>&nbsp;
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/GeekFocus/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="HiFi-2-SMRT Sequencing - HiFi Reads [High fidelity reads]"></span>
          
        </div>

        
          
  <div class="mt-3">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-author" aria-hidden="true"></i>
        banner
      </span>
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2022-02-20 22:11" pubdate>
          February 20, 2022 pm
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          6.4k words
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          54 mins
        
      </span>
    

    
    
      
        <span id="leancloud-page-views-container" class="post-meta" style="display: none">
          <i class="iconfont icon-eye" aria-hidden="true"></i>
          <span id="leancloud-page-views"></span> views
        </span>
        
      
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">HiFi-2-SMRT Sequencing - HiFi Reads [High fidelity reads]</h1>
            
              <p class="note note-info">
                
                  
                    Last updated on June 8, 2022 pm
                  
                
              </p>
            
            <div class="markdown-body">
              
              <h1 id="主流三代测序平台Oxford-的-Nanopore，Pacific-Biosciences（-PacBio）的-Single-Molecule-Real-Time（SMRT）Sequencing"><a href="#主流三代测序平台Oxford-的-Nanopore，Pacific-Biosciences（-PacBio）的-Single-Molecule-Real-Time（SMRT）Sequencing" class="headerlink" title="主流三代测序平台Oxford 的 Nanopore，Pacific Biosciences（ PacBio）的 Single Molecule Real-Time（SMRT）Sequencing"></a>主流三代测序平台Oxford 的 Nanopore，Pacific Biosciences（ PacBio）的 Single Molecule Real-Time（SMRT）Sequencing</h1><span id="more"></span>

<p>Ref:<a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/339875837">https://zhuanlan.zhihu.com/p/339875837</a></p>
<p><a target="_blank" rel="noopener" href="http://pacbiofileformats.readthedocs.io/en/5.0/">http://pacbiofileformats.readthedocs.io/en/5.0/</a></p>
<p>PacBio优势：</p>
<ul>
<li>在不影响吞吐量和准确性前提下，提供目前<font color="green">最长的读取长度</font></li>
</ul>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/1.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/1.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<ul>
<li>如果不含系统误差，准确度可达<font color="green"> 99.999％</font></li>
</ul>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/2.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/2.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<ul>
<li>可测取富<font color="green">含AT或GC区域</font>，<font color="green">高度重复序列</font>，<font color="green">回文序列</font>等，不会产生GC的较大偏差</li>
<li>可以直接<font color="green">测取化学修饰</font>，在<font color="green">表观遗传学</font>中有重要应用</li>
</ul>
<h1 id="文库构建"><a href="#文库构建" class="headerlink" title="文库构建"></a>文库构建</h1><p>将样本中的DNA或RNA分子提取后，构建<font color="green">哑铃状分子结构<strong>SMRTbell</strong></font>：</p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/3.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/3.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<p>黄色，紫色：双链DNA分子; 蓝色：接头（Adapter）</p>
<p>文库分子展开，一个完整的<font color="green">圆环</font>：</p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/4.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/4.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<p>圆环结构有利于周而复始的滚环复制，利于纠错。</p>
<p>将样本中所有的DNA片段都构建哑铃状分子结构，组成的集合叫<font color="green">文库</font>（SMRTbell Library），随后被放到测序芯片中。</p>
<p>以基因组HiFi文库为例（15-20K文库）。当得到gDNA后，先利用G-tube管或Megaruptor System将基因组<strong>片段化</strong>至合适大小，而后通过<strong>去除单链悬突</strong>、<strong>损伤修复</strong>和<strong>末端修复</strong>等步骤，得到完整的双链插入片段。接下来，通过将<strong>接头连接</strong>至双链DNA来创建SMRTbell文库，从而得到环状模板。完成接头连接后，需要对连接产物进行纯化，利用<strong>酶处理</strong>来消化线性或内部损伤环形DNA分子（游离的Hairpin Adapter、两端未连接Adapter的DNA模板、已成环但内部有损伤的DNA模板），酶处理完毕后，一般会利用<strong>Bulepippin</strong>或Sage ELF System切胶<strong>回收目标大小范围</strong>内的文库。文库质检</p>
<h1 id="测序芯片"><a href="#测序芯片" class="headerlink" title="测序芯片"></a>测序芯片</h1><p>以 RSII 测序平台为例，测序仪芯片（SMRT Cell）：</p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/5.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/5.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<p>放大：上面整齐排列着<font color="green">15万个</font>直径为<font color="green">70纳米的测序微孔</font>（<font color="red">Zero-Model Waveguides，ZMWs</font>）。</p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/6.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/6.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<h1 id="上机测序"><a href="#上机测序" class="headerlink" title="上机测序"></a>上机测序</h1><h2 id="1-构建测序复合物"><a href="#1-构建测序复合物" class="headerlink" title="1. 构建测序复合物"></a>1. <strong>构建测序复合物</strong></h2><p>测序复合物：聚合酶，测序模板，测序引物</p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/7.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/7.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<h2 id="2-复合物撒入测序小孔"><a href="#2-复合物撒入测序小孔" class="headerlink" title="2. 复合物撒入测序小孔"></a>2. <strong>复合物撒入测序小孔</strong></h2><p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/8.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/8.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<h2 id="3-固定测序复合物"><a href="#3-固定测序复合物" class="headerlink" title="3. 固定测序复合物"></a>3. <strong>固定测序复合物</strong></h2><p>由于<font color="green">聚合酶加了<strong>生物素</strong></font>，在<font color="green">芯片玻璃底板有<strong>链酶亲和素</strong></font>。利用生物素和链酶亲和素的亲和力，包含聚合酶的<strong>测序复合物会被<font color="green">固定</font>在玻璃底板。</strong></p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/9.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/9.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<h2 id="4-构建带有荧光基团的dNTP"><a href="#4-构建带有荧光基团的dNTP" class="headerlink" title="4. 构建带有荧光基团的dNTP"></a>4. <strong>构建带有荧光基团的dNTP</strong></h2><p>在芯片溶液中含有许多游离dNTP，所谓游离dNTP就是随机飘在溶液中的dNTP。</p>
<p>ATGC四种碱基的dNTP，在磷酸基团上分别带有<font color="green">四种颜色的荧光基团</font>。</p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/10.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/10.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<h2 id="5-边合成边测序"><a href="#5-边合成边测序" class="headerlink" title="5. 边合成边测序"></a>5. <strong>边合成边测序</strong></h2><p>在合成时，游离的dNTP被固定在底板上的酶捕获，<font color="green">激发光会从玻璃板底部发出</font>。</p>
<p><video src="/img/2022-02-20-SMRT-sequencing/11.mp4"></video><br><video src="2022-02-20-SMRT-sequencing/11.mp4"></video><br><strong>怎么保证每次测取一个碱基？</strong></p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/12.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/12.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<p>由于测序小孔直径很小，激发光的穿透能力会逐渐衰减，只能在小孔中传输很短的距离，所以<font color="green">只有当dNTP足够靠近底部，荧光基团才会被激发光照到，发出荧光</font>。其他游离dNTP虽然也有可能飘到小孔底部被激发光照到，但这种情况极少。<br>在一个碱基合成结束后，带有荧光基团的磷酸基团会从dNTP上掉落，发生猝灭，不影响其他碱基的信号检测。 在发生测序的小孔有各自的DNA片段和测序复合物，同一时间发出不同颜色的激发光，机器会检测到如下的光信号，实际同时会得到多达几万个光点。</p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/13.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/13.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<p>重复上述步骤，经过计算机分析光谱，最终我们拿到样本的测序文件。SMRT Sequencing 测序过程中，<font color="green">每秒读取三个碱基</font>，一个小时可检测大约一万多碱基。</p>
<h2 id="6-检测碱基甲基化"><a href="#6-检测碱基甲基化" class="headerlink" title="6. 检测碱基甲基化"></a>6. <strong>检测碱基甲基化</strong></h2><p>在SMRT Sequencing测序过程中，可以<font color="green">直接测到碱基被修饰状态</font>，聚合酶遇到碱基上带有甲基化的碱基，<font color="green">合成速度明显变慢</font>，而且<font color="green">光谱也会发生改变</font>。SMRT Sequencing 可以检测到碱基的甲基化修饰。</p>
<h1 id="测序模型"><a href="#测序模型" class="headerlink" title="测序模型"></a>测序模型</h1><h2 id="1-Circular-Consensus-Sequencing-CCS-–-HiFi"><a href="#1-Circular-Consensus-Sequencing-CCS-–-HiFi" class="headerlink" title="1. Circular Consensus Sequencing (CCS) – HiFi"></a>1. Circular Consensus Sequencing (CCS) – HiFi</h2><p>说这种测序模型前，就不得不提三代测序最大的缺点：<font color="green">碱基读取不准，错误率在12.5%</font>。每八个碱基读错一个。</p>
<p>碱基读取<font color="green">错误随机</font>，重读一遍同样位置的碱基，不一定发生相同错误。</p>
<p>对同一个序列，多测几遍，<font color="red">矫正</font>读错碱基。</p>
<p>滚环复制的优势: 利用测序复合物在<font color="red">环状文库分子循环测序</font><strong>同一个片段</strong>来<font color="green">消除错误率</font>。</p>
<p>HiFi reads（High Fidelity reads）是2019年由PacBio推出的基于<font color="red">环化共有序列（Circular Consensus Sequencing，CCS）</font>模式产生的既兼顾长读长（<font color="red">10-20kb</font>长度）又具有高精度（&gt;99%准确率）的测序结果。</p>
<p>超长酶读长中，去掉接头序列，即可获得多条subreads。多条subreads合并生成高度一致性的hifi reads。</p>
<p><video src="/img/2022-02-20-SMRT-sequencing/14.mp4"></video><br><video src="2022-02-20-SMRT-sequencing/14.mp4"></video><br>这种测序模型，复制出的 Reads叫 <font color="red">HiFi Reads [High fidelity reads]</font>，测序准确率 &gt; 99%。</p>
<p>HiFi reads are a <strong>type of data produced using the circular consensus sequencing (CCS) mode on one of the PacBio Sequel Systems</strong>.</p>
<p>HiFi reads（High fidelity reads）是 <font color="red">Sequel II</font>三代测序平台推出的兼顾长读长和高准确度的测序序列，一般采用CCS（Circular Consensus Sequencing）模式测序。</p>
<p>在单次测序中得到更多的HiFi reads往往需要<strong>平衡</strong>测序的 <font color="red">酶读长</font>和 <font color="red">插入片段的长度</font>，插入片段太长会导致酶无法进行滚环测序，插入片段太短又牺牲了三代长读长测序的优势。</p>
<p>根据前期的官方经验推荐，目前HiFi文库构建的<font color="red">插入片段一般为8-13 kb</font>左右。安诺优达构建约10 kb的HiFi文库在<font color="red">Sequel II</font>平台进行测序。原始下机数据单cell产出268 Gb数据，其中<font color="red">酶平均读长51 kb</font>，<font color="red">酶读长N50 124 kb</font>，<font color="red">subreads平均读长11 kb</font>，<font color="red">subreads N50 13 kb</font>。</p>
<ul>
<li>下机数据产出统计表</li>
</ul>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/15.png" srcset="/GeekFocus/img/loading.gif" lazyload><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/15.png" srcset="/GeekFocus/img/loading.gif" lazyload></p>
<p>进一步利用官方软件调取CCS，设置最小pass数为3，经过调取获得CCS总数据量为<strong>22.43 Gb</strong>，CCS 序列数目为<strong>172.5万</strong>条<strong>，平均长度</strong>13 kb<strong>。与下机总数据量相比，</strong>目前CCS reads的得率约为8%，并且能够兼顾reads的读长，达到平均13 kb左右，数据质量相当不错??</p>
<ul>
<li>CCS数据产出统计表</li>
</ul>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/16.png" srcset="/GeekFocus/img/loading.gif" lazyload><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/16.png" srcset="/GeekFocus/img/loading.gif" lazyload></p>
<p>对拿到的HiFi reads进行进一步的<font color="red">质量评估</font>，我们发现大部分HiFi reads的<font color="red">准确度?</font>都在0.95以上，其中约35%的reads（pass≥10）质量值达到QV30（99.9%），这样高质量的reads非常有助于研究者开展下游深入的研究。</p>
<ul>
<li>CCS质量分布图</li>
</ul>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/17.png" srcset="/GeekFocus/img/loading.gif" lazyload><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/17.png" srcset="/GeekFocus/img/loading.gif" lazyload></p>
<p>同时兼顾长读长和高准确度的HiFi reads究竟有何用处呢？2019年<em>BioRxiv</em>题为“<strong>Highly-accurate long-read sequencing improves variant detection and assembly of a human genome</strong>”的文章。在这篇文章里研究者利用约30X的CCS reads组装人基因组，通过<font color="green">FALCON、Canu3和 wtdbg2</font>等不同软件进行组装，<font color="red">contig N50达到15.43-28.95 Mb</font>。从组装连续性来看，<strong>CCS reads能够做到与<font color="red">传统的CLR reads</font>组装相当的结果，重要的是基因组碱基准确度得到了明显提升，基因组组装消耗的计算资源和时间大幅下降</strong>。进一步利用CCS reads进行<font color="red">SNP、InDel等变异检测</font>，发现CCS reads在小的变异检出率和准确度上都有显著提升，数据结果与30X的Illumina数据分析结果基本接近。</p>
<ul>
<li>文章中CCS reads进行SNV和InDel calling统计表</li>
</ul>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/31.png" srcset="/GeekFocus/img/loading.gif" lazyload><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/31.png" srcset="/GeekFocus/img/loading.gif" lazyload></p>
<p>综上，HiFi reads无论在<font color="red">基因组全变异检测（SNV、InDel、SV）</font>还是<font color="red">基因组<em>de novo</em></font>领域都适用。目前<font color="red">唯一的限制因素</font>是要获得足够的HiFi reads，测序成本投入比较<font color="red">昂贵</font>，从组装<strong>计算资源节省</strong>和<strong>项目时间缩短</strong>的角度看，HiFi reads未尝不是更好的选择。<strong>对于基因组<font color="red">重复序列较多的复杂基因组</font>，目前市场上<font color="red">传统长读长测序准确度不高</font>的特点给组装造成了一定的困难，高准确度的HiFi reads未来可能是一个更好的解决方案。</strong>而对于昆虫、中草药、藻类等重复序列较高、基因组较小的物种（＜700 Mb），目前利用一个8 M SMRT Cell 产出的数据量基本足以支持CCS组装，性价比更高。需要完善的HiFi文库建库流程和基于CCS reads组装的生信流程！</p>
<h2 id="2-Continuous-Long-Read-CLR-Sequencing"><a href="#2-Continuous-Long-Read-CLR-Sequencing" class="headerlink" title="2. Continuous Long Read (CLR) Sequencing"></a>2. Continuous Long Read (CLR) Sequencing</h2><p>这种测序的优势在于可以读取更长的 Reads。<font color="red">？为什么更长</font></p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/19.gif" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/19.gif" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<h1 id="其他影响因素"><a href="#其他影响因素" class="headerlink" title="其他影响因素"></a>其他影响因素</h1><h2 id="1-GC-bias-影响"><a href="#1-GC-bias-影响" class="headerlink" title="1. GC bias 影响"></a>1. <font color="red">GC bias 影响</font></h2><blockquote>
<p>什么是 GC bias？<br>PCR 时，如果模板里的<font color="red">G、C碱基含量高，PCR效率低</font>，A、T碱基含量高，PCR效率高。一般测序过程，如二代测序，都会有大量的PCR过程。这样就会有一个问题，<font color="red">G、C含量高的片段，读到的 Reads 数少</font>。</p>
</blockquote>
<p>SMRT 没有 PCR ，因此GC含量高低的 Reads 片段都会有相似的概率被测序， GC Bias 影响小。</p>
<h2 id="2-读长的限制因素"><a href="#2-读长的限制因素" class="headerlink" title="2. 读长的限制因素"></a>2. 读长的限制因素</h2><ul>
<li><strong>DNA模板断裂</strong>，用激发光长时间照射DNA链时，会发生断裂，DNA链会从酶上掉下来，测序终止。</li>
<li><strong>酶变性</strong>，酶被长时间照射时，酶会变性，失去聚合酶活性，测序终止。</li>
<li><strong>文库序列短</strong>，如果做文库序列片段大于 20~30 K ，且保证质量的文库是有技术难度的</li>
</ul>
<h2 id="3-测序通量"><a href="#3-测序通量" class="headerlink" title="3. 测序通量"></a>3. 测序通量</h2><p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/20.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/20.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<p>目前，主流的测序平台有三种，各有利弊，可以根据自己的课题来选择。</p>
<p>以RSII为例，将测序复合物，随机撒到15万个小孔中，正好有一个<font color="red">复合物进入到单个小孔的概率符合泊松分布</font>。理论情况是</p>
<ul>
<li>1&#x2F;3 的小孔中有<font color="red">一个</font>测序复合物，正常信号</li>
<li>1&#x2F;3 的小孔什么都<font color="red">没有</font>，无信号</li>
<li>1&#x2F;3 的小孔中有<font color="red">两个以上的测序复合物</font>，杂乱信号</li>
</ul>
<p>五万个小孔 * 10kb，所以一张芯片大约会产出500M的数据。</p>
<hr>
<h1 id="三代下机数据（以sequel-平台为主）"><a href="#三代下机数据（以sequel-平台为主）" class="headerlink" title="三代下机数据（以sequel 平台为主）"></a>三代下机数据（以sequel 平台为主）</h1><h2 id="测序过程"><a href="#测序过程" class="headerlink" title="测序过程"></a>测序过程</h2><ul>
<li><font color="green">A adapter</font>通用接头，两端的接头可以一样也可以不一样</li>
<li><font color="green">B barcode</font>(客户自己设计)</li>
<li><font color="green">I insert </font>插入片段，即我们测序的目的片段</li>
<li>由于SMRTbell是环状的，测序过程是边合成边测序，因此可以沿着新链合成的方向不停地读取序列，读取一圈又一圈</li>
</ul>
<h2 id="测序结果"><a href="#测序结果" class="headerlink" title="测序结果"></a>测序结果</h2><ul>
<li>根据SMRTbell的形状以及测序的过程，我们容易知道，测序出来的reads如上图所示，由接头序列, 条码序列, 插入序列间隔线性分布，即ABIB-ABIB—ABIB-ABIB—…（A: adapter, B: barcode, I: insert）</li>
<li><font color="green">ZMW read</font> 是测序出来的完整结果，也即是<font color="green">polymerase read</font>，聚合酶合成过的所有的序列。</li>
<li>PostPrimary 分析后输出<font color="green">HQ region</font>，由ZMW read <font color="green">去除两端低质量</font>区域得到。</li>
</ul>
<h2 id="测序文件"><a href="#测序文件" class="headerlink" title="测序文件"></a>测序文件</h2><ul>
<li><p>RS II：<a target="_blank" rel="noopener" href="https://www.itdaan.com/blog/2017/05/26/6e59c682169d236cb41a2b6ba146125e.html">https://www.itdaan.com/blog/2017/05/26/6e59c682169d236cb41a2b6ba146125e.html</a></p>
</li>
<li><p>Sequel</p>
<ul>
<li>在下机文件中，主要有三类文件，<font color="red">bam 文件</font>，<font color="red">bam.pbi 文件</font>，以及<font color="red">xml文件</font>。</li>
<li>无fastq文件，sequel平台中bam 文件成为了fq的替代者，更节约储存空间。这是文件格式的一个重大更新。</li>
<li>用于后续分析的文件一般是<font color="red">.subreads.bam</font>，这等同于<font color="red">RS II </font>中的<font color="red">.subreads.fastq</font></li>
<li>下面介绍三类主要文件的具体格式</li>
</ul>
</li>
</ul>
<h3 id="BAM"><a href="#BAM" class="headerlink" title="BAM"></a>BAM</h3><ul>
<li><p><font color="green">普通bam</font>文件大多是<font color="green">比对结果文件</font>，例如用重测序分析中BWA生成的bam文件就是reads与基因组的比对文件。但<font color="red">pacbio的下机文件是没有与基因组进行过比对过</font>的，其主要作用是<font color="red">储存序列</font>。</p>
</li>
<li><p>Bam文件主要分为两个部分，头一部分是<font color="red">Header</font>，储存测序的相关信息，另一部分也即是文件的主要部分是records，保存<font color="red">序列信息</font>。就以subreads.bam文件为例，分析下bam文件的具体格式。</p>
</li>
<li><p>可以用samtools view 命令查看bam文件</p>
<blockquote>
<p>第一列：reads信息</p>
<p>{movieName}&#x2F;{holeNumber}&#x2F;{qStart}_{qEnd}<br>[对于CCS：{movieName}&#x2F;{holeNumber}&#x2F;ccs]<br>MovieName 是cell的名字，holeNumer是ZMW孔的编号，qStart和qEnd是subreads相对于ZMW reads的位置。<br>第二列 (sum of flags)：比对信息 均为4 代表没有比对上，也表明了bam文件只储存了序列信息，而没有比对信息。<br>第三列 (RNAM)：参考序列 值为 ，代表无参考序列<br>第四列 (position) : 比对上的第一个碱基位置 0<br>第五列 (Mapping quality) : 比对质量分数 255<br>第六列 (CIGAR值) : 比对的具体情况<br>第七列 (MRNM, ) : mate 对应的染色体<br>第八列 (mate position) : mate对应的位置 0<br>第九列 (ISIZE, Inferred fragment size) : 推断的插入片段大小 0</p>
<p>第十列 (Sequence) : 序列信息 具体的ATCG<br>第十一列 (ASCII码) : 碱基质量分数 ASCII+33<br>第十二列: 可选区域 记录Reads 的总体属性包括信号长度，信号强度等信息。</p>
</blockquote>
</li>
</ul>
<ol>
<li>zmws.bam 以及ccs.bam似乎公司并不一定会提供</li>
<li>经过检查，一条zmw reads 可以产生多条 subreads，也就是说subreads.bam 中，序列只是被剪下来了。</li>
<li>scraps.bam 格式保存的是获取subreads时废弃的序列，包括adapter，以及一些低质量的序列</li>
<li><strong>CCS.bam保存的是矫正后的一致性序列</strong>。</li>
</ol>
<h3 id="BAM-pbi"><a href="#BAM-pbi" class="headerlink" title="BAM.pbi"></a>BAM.pbi</h3><ul>
<li><p>是bam文件的<font color="red">索引文件</font>(PacBio BAM index)，与上一个版本（<font color="red">RS II</font>）的<font color="red">*cmp.h5</font>文件<font color="red">兼容</font>，其格式类似于<font color="red">HDF5</font>， 通过<font color="red"><strong>BGZF格式压缩</strong></font>。</p>
</li>
<li><p>其存在主要有两个作用</p>
</li>
<li><h5 id="随机访问"><a href="#随机访问" class="headerlink" title="随机访问"></a><font color="red">随机</font>访问</h5><p>通过参考序列，基因组区域；通过read 组别；通过qurey name；通过ZMW；通过barcode；其他</p>
</li>
<li><h5 id="在无需完全访问BAM文件的情况下，获取信息"><a href="#在无需完全访问BAM文件的情况下，获取信息" class="headerlink" title="在无需完全访问BAM文件的情况下，获取信息"></a>在无需完全访问BAM文件的情况下，获取信息</h5></li>
</ul>
<p>   获取<font color="red">统计信息</font>；通过提供index访问记录信息</p>
<h3 id="xml"><a href="#xml" class="headerlink" title="xml"></a>xml</h3><ul>
<li>MetaData, 储存数据描述。可用于<strong>filter</strong> 或者<strong>subset</strong>等功能。</li>
<li>sts.xml 储存数据的统计信息。</li>
<li>SMRT Link CL tools in 5.0.0 dataset命令可以进行方便的操作。</li>
</ul>
<hr>
<h1 id="Pacbio-Sequel-下机数据实例"><a href="#Pacbio-Sequel-下机数据实例" class="headerlink" title="Pacbio Sequel 下机数据实例"></a>Pacbio Sequel 下机数据实例</h1><blockquote>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs stylus">&gt;m54134_180724_220656<span class="hljs-selector-class">.scraps</span><span class="hljs-selector-class">.bam</span><br>&gt;m54134_180724_220656<span class="hljs-selector-class">.subreads</span><span class="hljs-selector-class">.bam</span><br>&gt;m54134_180724_220656<span class="hljs-selector-class">.scraps</span><span class="hljs-selector-class">.bam</span><span class="hljs-selector-class">.pbi</span><br>&gt;m54134_180724_220656<span class="hljs-selector-class">.subreads</span><span class="hljs-selector-class">.bam</span><span class="hljs-selector-class">.pbi</span><br>&gt;m54134_180724_220656<span class="hljs-selector-class">.baz2bam_1</span><span class="hljs-selector-class">.log</span><br>&gt;m54134_180724_220656<span class="hljs-selector-class">.sts</span><span class="hljs-selector-class">.xml</span><br>&gt;m54134_180724_220656<span class="hljs-selector-class">.subreadset</span><span class="hljs-selector-class">.xml</span><br>&gt;m54134_180724_220656<span class="hljs-selector-class">.transferdone</span><br>&gt;m54134_180724_220656<span class="hljs-selector-class">.adapters</span>.fasta<br></code></pre></td></tr></table></figure>
</blockquote>
<p>m54134_180724_220656.subreads.bam</p>
<p><font color="red">m: 是movie 的缩写</font></p>
<p><font color="red">54134： 是设备的编号</font></p>
<p><font color="red">180724_220656： 是测序运行的开始时间，为2018年07月24日22时06分56秒</font></p>
<p><font color="red">subreads: 这部分是不同的数据类型，一般有scraps， subreads等</font></p>
<p><font color="red">bam: 这是数据的格式，一般有bam格式，xml，fasta格式等</font></p>
<p>BAM文件官网：<a target="_blank" rel="noopener" href="https://pacbiofileformats.readthedocs.io/en/3.0/BAM.html">https://pacbiofileformats.readthedocs.io/en/3.0/BAM.html</a></p>
<p><a target="_blank" rel="noopener" href="https://github.com/PacificBiosciences/PacBioFileFormats/blob/3.0/BAM.rst">https://github.com/PacificBiosciences/PacBioFileFormats/blob/3.0/BAM.rst</a></p>
<h1 id="CLR-HiFi-选择"><a href="#CLR-HiFi-选择" class="headerlink" title="CLR HiFi 选择"></a>CLR HiFi 选择</h1><p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/29.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/29.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>
<p><img src="/GeekFocus/2022-02-20-HiFi-2-SMRT/30.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"><br><img src="/GeekFocus/img/2022-02-20-HiFi-2-SMRT/30.png" srcset="/GeekFocus/img/loading.gif" lazyload alt="img"></p>

              
            </div>
            <hr/>
            <div>
              <div class="post-metas my-3">
  
    <div class="post-meta mr-3 d-flex align-items-center">
      <i class="iconfont icon-category"></i>
      

<span class="category-chains">
  
  
    
      <span class="category-chain">
        
  <a href="/GeekFocus/categories/Biology/" class="category-chain-item">Biology</a>
  
  

      </span>
    
  
</span>

    </div>
  
  
    <div class="post-meta">
      <i class="iconfont icon-tags"></i>
      
        <a href="/GeekFocus/tags/assembly/">#assembly</a>
      
    </div>
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>HiFi-2-SMRT Sequencing - HiFi Reads [High fidelity reads]</div>
      <div>https://wangc1106.github.io/GeekFocus/2022/02/20/2022-02-20-HiFi-2-SMRT/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>Author</div>
          <div>banner</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>Posted on</div>
          <div>February 20, 2022</div>
        </div>
      
      
      <div class="license-meta-item">
        <div>Licensed under</div>
        <div>
          
            
            
              <a target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
              <span class="hint--top hint--rounded" aria-label="BY - Attribution">
                <i class="iconfont icon-by"></i>
              </span>
              </a>
            
          
        </div>
      </div>
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/GeekFocus/2022/02/23/2022-02-23-jellyfish-genomescope/" title="jellyfish &amp; Genomescope">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">jellyfish &amp; Genomescope</span>
                        <span class="visible-mobile">Previous</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/GeekFocus/2022/02/20/2022-02-20-HiFi-1/" title="HiFi-1-three-generation-sequencing-techniqie">
                        <span class="hidden-mobile">HiFi-1-three-generation-sequencing-techniqie</span>
                        <span class="visible-mobile">Next</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="margin-left: -1rem">
    <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;Table of Contents</p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>
  </div>
</div>





  



  



  



  



  






    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">Search</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">Keyword</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <span>banner</span> <i class="iconfont icon-love"></i> <span>2021-2022©</span> 
    </div>
  
  
    <div class="statistics">
  
  

  
    
      <span id="leancloud-site-pv-container" style="display: none">
        总访问量 
        <span id="leancloud-site-pv"></span>
         次
      </span>
    
    
      <span id="leancloud-site-uv-container" style="display: none">
        总访客数 
        <span id="leancloud-site-uv"></span>
         人
      </span>
    
    

  
</div>

  
  
  
</div>

    <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
    </div>
  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.0/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/GeekFocus/js/events.js" ></script>
<script  src="/GeekFocus/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>






  
    <script  src="/GeekFocus/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.18.2/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      headingSelector : CONFIG.toc.headingSelector || 'h1,h2,h3,h4,h5,h6',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      collapseDepth   : CONFIG.toc.collapseDepth || 0,
      scrollSmooth    : true,
      headingsOffset  : -boardTop
    });
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }
  });
</script>


  <script>
  (function() {
    var enableLang = CONFIG.code_language.enable && CONFIG.code_language.default;
    var enableCopy = CONFIG.copy_btn;
    if (!enableLang && !enableCopy) {
      return;
    }

    function getBgClass(ele) {
      return Fluid.utils.getBackgroundLightness(ele) >= 0 ? 'code-widget-light' : 'code-widget-dark';
    }

    var copyTmpl = '';
    copyTmpl += '<div class="code-widget">';
    copyTmpl += 'LANG';
    copyTmpl += '</div>';
    jQuery('.markdown-body pre').each(function() {
      var $pre = jQuery(this);
      if ($pre.find('code.mermaid').length > 0) {
        return;
      }
      if ($pre.find('span.line').length > 0) {
        return;
      }

      var lang = '';

      if (enableLang) {
        lang = CONFIG.code_language.default;
        if ($pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2 && $pre.children().hasClass('hljs')) {
          lang = $pre[0].children[0].classList[1];
        } else if ($pre[0].getAttribute('data-language')) {
          lang = $pre[0].getAttribute('data-language');
        } else if ($pre.parent().hasClass('sourceCode') && $pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2) {
          lang = $pre[0].children[0].classList[1];
          $pre.parent().addClass('code-wrapper');
        } else if ($pre.parent().hasClass('markdown-body') && $pre[0].classList.length === 0) {
          $pre.wrap('<div class="code-wrapper"></div>');
        }
        lang = lang.toUpperCase().replace('NONE', CONFIG.code_language.default);
      }
      $pre.append(copyTmpl.replace('LANG', lang).replace('code-widget">',
        getBgClass($pre[0]) + (enableCopy ? ' code-widget copy-btn" data-clipboard-snippet><i class="iconfont icon-copy"></i>' : ' code-widget">')));

      if (enableCopy) {
        Fluid.utils.createScript('https://lib.baomitu.com/clipboard.js/2.0.10/clipboard.min.js', function() {
          var clipboard = new window.ClipboardJS('.copy-btn', {
            target: function(trigger) {
              var nodes = trigger.parentNode.childNodes;
              for (var i = 0; i < nodes.length; i++) {
                if (nodes[i].tagName === 'CODE') {
                  return nodes[i];
                }
              }
            }
          });
          clipboard.on('success', function(e) {
            e.clearSelection();
            e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-copy', 'icon-success');
            setTimeout(function() {
              e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-success', 'icon-copy');
            }, 2000);
          });
        });
      }
    });
  })();
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/GeekFocus/js/local-search.js" ></script>

  <script defer src="/GeekFocus/js/leancloud.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/GeekFocus/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">Blog works best with JavaScript enabled</div>
  </noscript>
</body>
</html>
