

<!DOCTYPE html>
<html lang="en" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/GeekFocus/img/fluid.png">
  <link rel="icon" href="/GeekFocus/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="banner">
  <meta name="keywords" content="Bioinformatics">
  
    <meta name="description" content="A fast, lock-free approach for efficient parallel counting of occurrences of k-mers">
<meta property="og:type" content="article">
<meta property="og:title" content="jellyfish &amp; Genomescope">
<meta property="og:url" content="https://wangc1106.github.io/GeekFocus/2022/02/23/2022-02-23-jellyfish-genomescope/index.html">
<meta property="og:site_name" content="DrBanner">
<meta property="og:description" content="A fast, lock-free approach for efficient parallel counting of occurrences of k-mers">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2022-02-23T14:49:06.478Z">
<meta property="article:modified_time" content="2022-06-05T12:06:47.174Z">
<meta property="article:author" content="banner">
<meta property="article:tag" content="assembly">
<meta name="twitter:card" content="summary_large_image">
  
  
  
  <title>jellyfish &amp; Genomescope - DrBanner</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/GeekFocus/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/GeekFocus/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/GeekFocus/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"wangc1106.github.io","root":"/GeekFocus/","version":"1.9.1","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"right","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":false,"follow_dnt":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":"UqIYufmNk2sE8MiBKKWc4Kja-gzGzoHsz","app_key":"iYpeWQcmNqpHwr2uVlnJYVji","server_url":"https://wangc1106.github.io/GeekFocus","path":"window.location.pathname","ignore_local":false}},"search_path":"/GeekFocus/local-search.xml"};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/GeekFocus/js/utils.js" ></script>
  <script  src="/GeekFocus/js/color-schema.js" ></script>

  
<meta name="generator" content="Hexo 6.2.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 70vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/GeekFocus/">
      <strong>DrBanner</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/">
                <i class="iconfont icon-home-fill"></i>
                Home
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/archives/">
                <i class="iconfont icon-archive-fill"></i>
                Archives
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/categories/">
                <i class="iconfont icon-category-fill"></i>
                Categories
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/tags/">
                <i class="iconfont icon-tags-fill"></i>
                Tags
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/GeekFocus/about/">
                <i class="iconfont icon-user-fill"></i>
                About
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              &nbsp;<i class="iconfont icon-search"></i>&nbsp;
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/GeekFocus/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="jellyfish &amp; Genomescope"></span>
          
        </div>

        
          
  <div class="mt-3">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-author" aria-hidden="true"></i>
        banner
      </span>
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2022-02-23 22:49" pubdate>
          February 23, 2022 pm
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          9.9k words
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          83 mins
        
      </span>
    

    
    
      
        <span id="leancloud-page-views-container" class="post-meta" style="display: none">
          <i class="iconfont icon-eye" aria-hidden="true"></i>
          <span id="leancloud-page-views"></span> views
        </span>
        
      
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">jellyfish &amp; Genomescope</h1>
            
              <p class="note note-info">
                
                  
                    Last updated on June 5, 2022 pm
                  
                
              </p>
            
            <div class="markdown-body">
              
              <h1 id="A-fast-lock-free-approach-for-efficient-parallel-counting-of-occurrences-of-k-mers"><a href="#A-fast-lock-free-approach-for-efficient-parallel-counting-of-occurrences-of-k-mers" class="headerlink" title="A fast, lock-free approach for efficient parallel counting of occurrences of k-mers"></a>A fast, lock-free approach for efficient parallel counting of occurrences of k-mers</h1><span id="more"></span>

<h1 id="1-Paper-2011"><a href="#1-Paper-2011" class="headerlink" title="1. Paper-2011 :"></a><a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/21217122/">1. Paper-2011</a> :</h1><ul>
<li><p><font color="red">Counting the number of occurrences of every k-mer</font> </p>
</li>
<li><p><font color="green">genome assembly</font></p>
</li>
<li><p><font color="purple">error correction</font> of sequencing reads</p>
</li>
<li><p><font color="daekred">fast multiple sequence alignment</font> </p>
</li>
<li><p><font color="blue">repeat detection</font></p>
</li>
<li><p><strong>current</strong> k-mer counting tools <strong>too slow</strong> , <strong>memory intensive</strong></p>
</li>
<li><p>multicore computers facilities <strong>parallel computational paradigm</strong></p>
</li>
<li><p>multithreaded, <font color="red">lock-free hash table</font> </p>
</li>
<li><p><strong>k-mers up to 31</strong> </p>
</li>
<li><p><strong>suffix arrays</strong></p>
</li>
</ul>
<hr>
<h1 id="2-github-doc"><a href="#2-github-doc" class="headerlink" title="2. github-doc :"></a><font color="red"><a target="_blank" rel="noopener" href="https://github.com/gmarcais/Jellyfish/tree/master/doc">2. github-doc</a></font> :</h1><h2 id="2-1-Counting-k-mers-in-sequencing-reads"><a href="#2-1-Counting-k-mers-in-sequencing-reads" class="headerlink" title="2.1 Counting k-mers in sequencing reads"></a>2.1 Counting k-mers in sequencing reads</h2><ul>
<li><p>The <code>-C</code> switch instructs to save in the hash only canonical规范 k-mers, while the count is the number of occurrences of both a k-mer and it reverse complement.</p>
</li>
<li><p><font color="green"><strong>-s</strong></font> 预估值设置：The size parameter (given with <font color="red">-s</font>) is an indication of <font color="red">the number k-mers [N*(L-K+1)]</font> that will be <font color="red"><strong>stored in the hash</strong></font>. For sequencing reads, this size should be the size of <font color="red">the genome plus the k-mers generated by sequencing errors</font>. For example, if the error rate is <font color="red">e</font> (e.g.Illumina reads, usually e~<font color="red">1%</font>), with an estimated genome size of G and a coverage of <font color="red">c</font>, the number of expected k-mers is <font color="red"><strong>G+Gcek</strong></font>.</p>
</li>
</ul>
<blockquote>
<p>NOTE: unlike in Jellyfish 1, this <code>-s</code> parameter is <font color="red">only an estimation</font>. <font color="red">If the size given is too small to fit all the k-mers, the hash size will be increased automatically or partial results will be written to disk and finally merged automatically</font>. Running <code>jellyfish merge</code> should never be necessary, as now jellyfish now takes care of this task on its own. </p>
</blockquote>
<ul>
<li><font color="green"><strong>-s 预估kmer以hash表存储所需内存大小，设置小了会自动扩大，并自动merge</strong></font>。</li>
<li>If the low frequency k-mers (k-mers occurring only once), which are mostly due to sequencing errors, are not of interest, one might consider counting only high-frequency k-mers (see section <a target="_blank" rel="noopener" href="https://github.com/gmarcais/Jellyfish/tree/master/doc#Counting-high-frequency-k-mers">Counting high frequency k-mers</a>), which uses less memory and is potentially faster. <font color="green"><strong>低频k-mers由测序错误导致，只考虑高频减少内存使用提高速度</strong></font>。</li>
</ul>
<h2 id="2-2-Counting-k-mers-in-a-genome"><a href="#2-2-Counting-k-mers-in-a-genome" class="headerlink" title="2.2 Counting k-mers in a genome"></a>2.2 Counting k-mers in a genome</h2><ul>
<li>In an 【actual genome or finished sequence？】, a k-mer and its reverse complement are not equivalent, hence using the <code>-C</code> switch does not make sense. In addition, the size for the hash can be set directly to the size of the genome.</li>
<li>-C没有意义？哈希表的size直接设置为基因组大小？不够自动调整？</li>
</ul>
<h2 id="2-3-Counting-high-frequency-k-mers"><a href="#2-3-Counting-high-frequency-k-mers" class="headerlink" title="2.3 Counting high-frequency k-mers"></a>2.3 Counting high-frequency k-mers</h2><p>Jellyfish offers <font color="red">two way</font> to count only high-frequency k-mers (<font color="red"><strong>count &gt;1</strong></font>), which reduces significantly the memory usage. Both methods are based on using <font color="red">Bloom filters</font>. The first method is a one pass method, which <font color="red">provides approximate count for some percentage of the k-mers</font>. The second method is a two pass method which <font color="red">provides exact count</font>. In both methods, most of the low-frequency k-mers are not reported.</p>
<h3 id="2-3-1-One-pass-method"><a href="#2-3-1-One-pass-method" class="headerlink" title="2.3.1 One pass method"></a>2.3.1 One pass method</h3><h3 id="2-3-2-Two-passes-method"><a href="#2-3-2-Two-passes-method" class="headerlink" title="2.3.2 Two passes method"></a>2.3.2 Two passes method</h3><h2 id="2-4-Counting-a-subset-of-k-mers"><a href="#2-4-Counting-a-subset-of-k-mers" class="headerlink" title="2.4 Counting a subset of k-mers"></a>2.4 Counting a subset of k-mers</h2><ul>
<li>It is possible to count the number of occurrences of <font color="red">only a subset of</font> predefined k-mers, using the <font color="red"><strong>–if</strong></font> switch. Only count the 20-mers of chromosome 20 in chromosome 1 of human</li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs sh">jellyfish count -m 20 -s 100M -C -t 10 -o 20and1.jf --<span class="hljs-keyword">if</span> chr20.fa chr1.fa<br></code></pre></td></tr></table></figure>

<ul>
<li><p>This <strong>reads all the canonical 20-mers from the file chr20.fa</strong> (<font color="green"><strong>but don’t count them</strong></font>) and <font color="green"><strong>loads them in memory</strong></font>. Then it <strong>reads chr1.fa and counts the number of occurrences of the 20-mers</strong> <font color="green"><strong>only of the k-mers already loaded in memory</strong></font>.</p>
</li>
<li><p>把chr20的kmer，存储。统计chr1的20-mer在上一步存储中已存在的部分的数目。？chr1和20的关联？</p>
</li>
<li><p>Now, the histogram will also report k-mers with a <font color="green"><strong>0 count</strong></font> (<font color="green"><strong>the 20-mers of chr20</strong></font> which <font color="green"><strong>do not exists in chr1</strong></font>):</p>
</li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs sh">$ jellyfish histo 20and1.jf | <span class="hljs-built_in">head</span> <br>0 49401674<br>1 1116213<br>2 425471<br></code></pre></td></tr></table></figure>

<p>Note that the file given to <code>--if</code> is a fasta file. If a list of k-mers is available instead (say one per line in a text file), it can be transformed into a fasta file with one k-mer per line using the following one liner Perl command:</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs sh">perl -ne <span class="hljs-string">&#x27;print(&quot;&gt;\n$_&quot;)&#x27;</span><br><span class="hljs-comment">#kmer line转fa，一行&gt;号，一行kmer</span><br></code></pre></td></tr></table></figure>

<h2 id="2-5-如何读取压缩文件"><a href="#2-5-如何读取压缩文件" class="headerlink" title="2.5 如何读取压缩文件"></a>2.5 如何读取压缩文件</h2><p> jellyfish只读入fq或fa</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs sh">zcat *.fastq.gz | jellyfish count /dev/fd/0 ...<br><span class="hljs-comment">#using the &lt;() redirection provided by the shell (e.g. bash, zsh):</span><br><span class="hljs-comment">#!/usr/bin/bash</span><br>jellyfish count &lt;(zcat file1.fastq.gz) &lt;(zcat file2.fasta.gz) ...<br></code></pre></td></tr></table></figure>

<hr>
<h1 id="3-Jellyfish介绍"><a href="#3-Jellyfish介绍" class="headerlink" title="3. Jellyfish介绍"></a>3. Jellyfish介绍</h1><p><a target="_blank" rel="noopener" href="http://www.cbcb.umd.edu/software/jellyfish/">JELLYFISH</a>是<a target="_blank" rel="noopener" href="http://www.cbcb.umd.edu/">CBCB(Center for Bioinformatics and Computational Biology)</a>的Guillaume Marçais 和 <a target="_blank" rel="noopener" href="http://www.cs.cmu.edu/~ckingsf/">Carl Kingsford</a> 研发的一款计数 DNA 的 k-mers 的软件。该软件运用 Hash 表来存储数据，同时能多线程运行，速度快，内存消耗小。该软件只能运行在64位的Linux系统下。其<a target="_blank" rel="noopener" href="http://bioinformatics.oxfordjournals.org/content/27/6/764">文章</a>于2011年发表在杂志 <a target="_blank" rel="noopener" href="http://bioinformatics.oxfordjournals.org/">Bioinformatics</a> 上。</p>
<h1 id="4-Jellyfish安装"><a href="#4-Jellyfish安装" class="headerlink" title="4. Jellyfish安装"></a>4. Jellyfish安装</h1><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><code class="hljs sh">$ wget 最新targz<br>$ tar zxvf .tar.gz<br>$ <span class="hljs-built_in">mkdir</span> jellyfish<br>$ <span class="hljs-built_in">cd</span> jellyfish-2.3.0<br>$ ./configure --prefix=Your/Path/to/jellyfish<br>如果安装在当前目录中，会报错。<br>$ make -j 8<br>$ make install<br></code></pre></td></tr></table></figure>

<h1 id="5-Jellyfish使用"><a href="#5-Jellyfish使用" class="headerlink" title="5. Jellyfish使用"></a>5. Jellyfish使用</h1><h2 id="jellyfish-count-【count-kmers-to-estimate-genome-size】"><a href="#jellyfish-count-【count-kmers-to-estimate-genome-size】" class="headerlink" title="jellyfish count 【count kmers to estimate genome size】"></a>jellyfish count 【count kmers to estimate genome size】</h2><p>Theory:</p>
<ul>
<li>If k is large enough so that each kmer found is unique in the genome,</li>
<li>and if the genome length (e.g. 1,000,000) is much larger than the kmer length (e.g. 21),</li>
<li>and if no PCR or sequencing errors,</li>
<li>then the number of kmers will be approximately equal to the length of the genome.</li>
</ul>
<p>！！！！<font color="red">Kmer counting tutorial</font>: <a target="_blank" rel="noopener" href="https://bioinformatics.uconn.edu/genome-size-estimation-tutorial/#">https://bioinformatics.uconn.edu/genome-size-estimation-tutorial/#</a></p>
<ul>
<li>Jellyfish: <a target="_blank" rel="noopener" href="https://github.com/gmarcais/Jellyfish">https://github.com/gmarcais/Jellyfish</a></li>
<li>Further jellyfish steps : <a target="_blank" rel="noopener" href="https://github.com/gmarcais/Jellyfish/tree/master/doc">https://github.com/gmarcais/Jellyfish/tree/master/doc</a></li>
</ul>
<p>Note: Aug 2019: conda install seems to work only with: conda install jellyfish&#x3D;2.2.3</p>
<p>Steps:</p>
<ul>
<li><font color="green">“&gt;Go through</font> the sequencing reads</li>
<li>For each <font color="green">new kmer seen</font>, <font color="green">add to table with count</font>.</li>
<li>If kmer <font color="green">seen before</font>, <font color="green">increment count</font>.</li>
<li>Find the <font color="green">average kmer frequency</font> (sequencing depth), e.g. 50</li>
<li><font color="green">Exclude kmers with a count of ~ 1</font>, as these are likely from <font color="green">errors</font></li>
<li>Add all the other kmers, and <font color="green"><strong>divide by average kmer frequency</strong></font> &#x3D;&gt; This is the approx genome length</li>
</ul>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">jellyfish</span> count -m <span class="hljs-number">21</span> -s <span class="hljs-number">100</span>M -t <span class="hljs-number">10</span> -C reads.fasta<br></code></pre></td></tr></table></figure>

<p>To use gzipped files: and <font color="green">paired-end reads</font>:</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">jellyfish</span> count -m <span class="hljs-number">21</span> -s <span class="hljs-number">100</span>M -t <span class="hljs-number">10</span> -C &lt;zcat R1.fq.gz) &lt;(zcat R2.fq.gz)<br></code></pre></td></tr></table></figure>

<ul>
<li>-m: kmer length, 21 is commonly used, <strong>17</strong></li>
<li>-s: size of hash table: should be genome size + extra kmers from seq errors. <font color="green">However, it does say that hash size <strong>will be increased automatically if needed</strong>.</font></li>
<li>-C: canonical. <font color="green">Reverse complement kmers are considered to be identical and are counted as the same thing</font>. <font color="green"><strong>This is recommended</strong></font>.</li>
<li>-t: number of threads</li>
<li>output: mer_counts.jf</li>
</ul>
<p>Plot the histogram:</p>
<figure class="highlight mipsasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs mipsasm"><span class="hljs-keyword">jellyfish </span>histo -t <span class="hljs-number">10</span> mer_counts.<span class="hljs-keyword">jf </span>&gt; reads.histo<br></code></pre></td></tr></table></figure>

<p><strong>???Plot the histogram with an x axis of one million instead of default 10,000</strong>:</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">jellyfish</span> histo -t <span class="hljs-number">10</span> --high=<span class="hljs-number">1000000</span> mer_counts.jf &gt; reads.histo<br></code></pre></td></tr></table></figure>

<ul>
<li>Discussion on setting to 1 million: <a target="_blank" rel="noopener" href="https://github.com/schatzlab/genomescope/issues/22">https://github.com/schatzlab/genomescope/issues/22</a></li>
<li>There may be very high counts of some kmers due to chloroplast sequences, spike-in sequences, etc.</li>
</ul>
<blockquote>
<p>使用fastq文件在默认参数上和fasta文件没有区别。生成的hash结果为二进制文件。</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><code class="hljs sh">-m | --mer-len=&lt;num&gt;<br> 使用的k-mer的长度。如果基因组大小为G，则k-mer长度选择为: k ~= <span class="hljs-built_in">log</span>(200G)<br>/log(4)。<br>-s | --size=&lt;num&gt;<br> Hash 的大小。最好设置的值大于总的独特的(distinct)k-mer数,这样生成的文件只<br>有一个。若该值不够大，则会生成多个<span class="hljs-built_in">hash</span>文件，以数字区分文件名。该值识别 <br>M 和 G。<br>-t | --threads=&lt;num&gt;  default: 1<br> 使用的CPU线程数<br>-o | --output=&lt;string&gt;  default: mer_counts.jf<br> 输出的结果文件前缀<br>-c | --counter-len=&lt;num&gt;  default:7<br> k-mer的计数结果所占的比特数,默认支持的最大数字是2^7=128。对于基因组测序覆盖<br>度为N，则要使设置的该值要大于N。该值越大，消耗内存越大。<br>-out-counter-len=&lt;num&gt;  default:4<br> 输出的二进制<span class="hljs-built_in">hash</span>文件中的计数结果所占的字节数,一个字节是8比特。则默认支持的最大<br>数字是2^32=4.3G<br><span class="hljs-comment">#The count of k-mers which cannot be represented with the given number of bytes will have a value equal to the maximum value that can be represented. Meaning, if the counter field uses 1 byte, any k-mers with count greater or equal to 255 will be reported of having a count of 255.</span><br>-C | --both-strand  default: <span class="hljs-literal">false</span><br> 【对正义链和反义链都进行计数】<br>-q | --quake  default: <span class="hljs-literal">false</span><br> quake兼容模式<br>--quality-start=&lt;num&gt;  default: 64<br> 起始碱基质量的ASCII值，默认为PHRED64<br>--min-quality=&lt;num&gt;  default: 0<br> 支持的最小的碱基质量值，低于此值的碱基将由N代替<br>-L | --lower-count=&lt;num&gt;<br> 【不输出数目低于此值的k-mer】<br>-U | --upper-count=&lt;num&gt;<br> 【不输出数目高于此值的k-mer】<br><span class="hljs-comment">#low frequency and high frequency k-mers can be skipped using the --L and --U` switches respectively. Although it might be more appropriate to filter out the low frequency $k$-mers using Bloom filters, as shown in section Counting-high-frequency-k-mers.</span><br></code></pre></td></tr></table></figure>
</blockquote>
<h2 id="jellyfish-merge-合并"><a href="#jellyfish-merge-合并" class="headerlink" title="jellyfish merge 合并"></a>jellyfish merge 合并</h2><p>Count的输出结果为二进制文件，可能输出多个hash文件，需要hash文件合并一个文件, merge 命令。</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment">#The merge subcommand is of little direct use with version version 2 of Jellyfish</span><br><span class="hljs-comment">#The count subcommand will merge intermediary files automatically as needed.</span><br>jellyfish merge -o mer_counts_merged.jf hash1 hash2 ...<br></code></pre></td></tr></table></figure>

<blockquote>
<p>-o | –output&#x3D;<string>  default: mer_counts_merged.jf<br>    输出的结果文件<br>–out-counter-len&#x3D;<num>  default: 4<br>输出的二进制hash文件中的计数结果所占的字节数,一个字节是8比特。则默认支持的最大数字是2^32&#x3D;4.3G</p>
</blockquote>
<h2 id="jellyfish-stats-统计"><a href="#jellyfish-stats-统计" class="headerlink" title="jellyfish stats 统计"></a>jellyfish stats 统计</h2><p>k-mer的结果以hash的二进制文件结果给出，需要统计出k-mer总数，特异的k-mer数目，只出现过一次的kmer数，出现了最多的k-mer的数目等信息。以stats命令来运行。使用方法：</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment">#Although these statistics could be computed from the histogram, it provides quick summary information</span><br>$ jellyfish stats <span class="hljs-built_in">hash</span><br>示例结果为：<br>Unique:    32355544    <span class="hljs-comment">#只出现过一次的k-mer的数目</span><br>Distinct:  88414020    <span class="hljs-comment">#特异性的k-mer数目，包含上一个的数据</span><br>Total:     432232807   <span class="hljs-comment">#总的k-mer数目</span><br>Max_count: 85348       <span class="hljs-comment">#同一个k-mer出现的最多的数目</span><br></code></pre></td></tr></table></figure>

<blockquote>
<p>-L | –lower-count&#x3D;<num><br>    不统计数目低于此值的k-mer<br>-U | –upper-count&#x3D;<num><br>    不统计数目高于此值的k-mer</p>
</blockquote>
<h2 id="jellyfish-histo-绘图"><a href="#jellyfish-histo-绘图" class="headerlink" title="jellyfish histo 绘图"></a>jellyfish histo 绘图</h2><p>对k-mer的计数结果有个直观的认识，则需要统计出现了x(x&#x3D;1,2,3…)次的kmer的数目y,以x，y为横纵坐标画出直方图。使用 histo 命令能给出 x 和 y 对应的值，将结果默认输出到标准输出。其使用方法为</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs sh">jellyfish histo -l 1 -h 1000 <span class="hljs-built_in">hash</span><br></code></pre></td></tr></table></figure>

<blockquote>
<p>-l | –low&#x3D;<num>  default: 1<br>    最低的 x 轴的值。同时结果会将低于此值的所有的k-mer的数目作为 (x-1) 的值。因<br>此该值为 2 和 1 的结果是一致的。<br>-h | –high&#x3D;<num>  【default: 10000？？】<br>    最高的 x 轴的值。同时结果会将高于此值的所有的k-mer的数目的和作为 (x+1) 的值。<br>-i | –increment&#x3D;<num>  default: 1<br>    x 轴取值是每隔该数值取值<br>-t | –threads&#x3D;<num>  default: 1<br>    使用的CPU线程数<br>-f | –full  default: false<br>    全部的直方图</p>
</blockquote>
<h2 id="jellyfish-dump-结果文件格式转换"><a href="#jellyfish-dump-结果文件格式转换" class="headerlink" title="jellyfish dump 结果文件格式转换"></a>jellyfish dump 结果文件格式转换</h2><p>由于count命令生成的结果为二进制的，如有需要，则可以转换成可读文本文件。使用 dump 命令，使用方法：</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs sh">jellyfish dump -c -t -U 1000 <span class="hljs-built_in">hash</span><br><span class="hljs-comment">#To output all the counts for all the k-mers in the file</span><br><span class="hljs-comment">#By default, the output is in FASTA format</span><br>jellyfish dump mer_counts.jf &gt; mer_counts_dumps.fa<br></code></pre></td></tr></table></figure>

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><code class="hljs sh">-c | --colum  default: <span class="hljs-literal">false</span><br> 生成结果为2列，第一列为k-mer序列，第二列为对应的数目。默认情况下是是fasta格<br>式，fasta的头为k-mer的数目，fasta的序列为k-mer的序列。<br>-t | --tab  default: <span class="hljs-literal">false</span><br> 当 -c 参数存在时，以tab来进行分隔两行。默认是以空格来分开的。<br><span class="hljs-comment"># Low frequency and high frequency k-mers can be skipped with the `-L` and `-U` switches respectively.</span><br>-L | --lower-count=&lt;num&gt;<br> 不输出小于该值的k-mer<br>-U | --upper-count=&lt;num&gt;<br> 不输出高于该值的k-mer<br>-o | --output=&lt;file&gt;<br> 输出文件的路径和名称<br></code></pre></td></tr></table></figure>



<h2 id="jellyfish-query-查询"><a href="#jellyfish-query-查询" class="headerlink" title="jellyfish query 查询"></a>jellyfish query 查询</h2><p>如果需要从Hash结果中查询指定的k-mer出现的次数，则要是用 query 命令。从标准输入读取k-mer的序列，从标准输出得到k-mer对应的数目。使用方法</p>
<figure class="highlight arcade"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs arcade">jellyfish query <span class="hljs-built_in">hash</span><br></code></pre></td></tr></table></figure>

<blockquote>
<p>-C | –both-strands  default: false<br>   同时查询k-mer序列的正负链<br>-i | –input&#x3D;<file><br>   输入的文件<br>-o | –output&#x3D;<file><br>   输出的文件</p>
</blockquote>
<figure class="highlight mipsasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs mipsasm"><span class="hljs-keyword">jellyfish </span>query mer_counts.<span class="hljs-keyword">jf </span>AACGTTG<br></code></pre></td></tr></table></figure>



<h2 id="jellyfish-info"><a href="#jellyfish-info" class="headerlink" title="jellyfish info"></a>jellyfish info</h2><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs sh"><span class="hljs-comment">#To get some information on 【how, when and where】 this jellyfish file was 【generated】, use the info subcommand</span><br><span class="hljs-comment">#outputs some information about the jellyfish file and the command used to generated it, in which directory and at what time the command was run</span><br>jellyfish info mer_counts.jf<br></code></pre></td></tr></table></figure>

<h2 id="jellyfish-mem"><a href="#jellyfish-mem" class="headerlink" title="jellyfish mem"></a>jellyfish mem</h2><p>The <code>mem</code> subcommand shows how much memory a count subcommand will need or conversely how large of a hash size will fit in a given amount of memory.</p>
<hr>
<p>ref：</p>
<p><a target="_blank" rel="noopener" href="https://www.annasyme.com/docs/jellyfish.html">https://www.annasyme.com/docs/jellyfish.html</a></p>
<hr>
<h1 id="Genomescope"><a href="#Genomescope" class="headerlink" title="Genomescope"></a>Genomescope</h1><p>安装github最新</p>
<p><font color="red">githubdoc</font>  : <a target="_blank" rel="noopener" href="https://github.com/schatzlab/genomescope">https://github.com/schatzlab/genomescope</a></p>
<p><font color="red">paperlink</font> : <a target="_blank" rel="noopener" href="https://doi.org/10.1093/bioinformatics/btx153">https://doi.org/10.1093/bioinformatics/btx153</a></p>
<h2 id="数据准备"><a href="#数据准备" class="headerlink" title="数据准备"></a>数据准备</h2><p>将paired-end数据整合？将read2数据的序列反向重复后与read1文件合并？使用Trinity附带的fastool程序完成转换。</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs sh">$ zcat read_1.clean.fq.gz | <span class="hljs-variable">$Trinity_Home</span>/trinity-plugins/fastool/fastool --illumina-trinity --to-fasta &gt; reads_1.fasta<br>$ zcat read_2.clean.fq.gz | <span class="hljs-variable">$Trinity_Home</span>/trinity-plugins/fastool/fastool --rev --illumina-trinity --to-fasta &gt; reads_2.fasta<br>$ <span class="hljs-built_in">cat</span> reads_1.fasta reads_2.fasta &gt; both.fasta<br></code></pre></td></tr></table></figure>

<h2 id="统计kmer"><a href="#统计kmer" class="headerlink" title="统计kmer"></a>统计kmer</h2><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs sh">jellyfish count -C -m [] -s [] -t [] s.fasta -o s.jf<br></code></pre></td></tr></table></figure>

<h2 id="绘直方图"><a href="#绘直方图" class="headerlink" title="绘直方图"></a>绘直方图</h2><figure class="highlight mipsasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs mipsasm"><span class="hljs-keyword">jellyfish </span>histo -t <span class="hljs-number">10</span> s.<span class="hljs-keyword">jf </span>&gt; s.histo<br></code></pre></td></tr></table></figure>

<h2 id="参数"><a href="#参数" class="headerlink" title="参数"></a>参数</h2><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs sh">-s 预估哈希表的大小，即G+Gce*k。G是Genome Size；c是coverage（genome survey测序通常低于100x）；e是测序错误率（illumina为1%）；k是kmer大小。<br>-C 表示考虑DNA正义与反义链，遇到反义kmer时，计入正义kmer频数中。<br></code></pre></td></tr></table></figure>

<h2 id="绘图"><a href="#绘图" class="headerlink" title="绘图"></a>绘图</h2><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs sh">Rscript genomescope.R histogram_file k-mer_length read_length output_dir [kmer_max] [verbose]<br></code></pre></td></tr></table></figure>


              
            </div>
            <hr/>
            <div>
              <div class="post-metas my-3">
  
    <div class="post-meta mr-3 d-flex align-items-center">
      <i class="iconfont icon-category"></i>
      

<span class="category-chains">
  
  
    
      <span class="category-chain">
        
  <a href="/GeekFocus/categories/Biology/" class="category-chain-item">Biology</a>
  
  

      </span>
    
  
</span>

    </div>
  
  
    <div class="post-meta">
      <i class="iconfont icon-tags"></i>
      
        <a href="/GeekFocus/tags/assembly/">#assembly</a>
      
    </div>
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>jellyfish &amp; Genomescope</div>
      <div>https://wangc1106.github.io/GeekFocus/2022/02/23/2022-02-23-jellyfish-genomescope/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>Author</div>
          <div>banner</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>Posted on</div>
          <div>February 23, 2022</div>
        </div>
      
      
      <div class="license-meta-item">
        <div>Licensed under</div>
        <div>
          
            
            
              <a target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
              <span class="hint--top hint--rounded" aria-label="BY - Attribution">
                <i class="iconfont icon-by"></i>
              </span>
              </a>
            
          
        </div>
      </div>
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/GeekFocus/2022/02/23/2022-02-21-Survey/" title="Genome Survey">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">Genome Survey</span>
                        <span class="visible-mobile">Previous</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/GeekFocus/2022/02/20/2022-02-20-HiFi-2-SMRT/" title="HiFi-2-SMRT Sequencing - HiFi Reads [High fidelity reads]">
                        <span class="hidden-mobile">HiFi-2-SMRT Sequencing - HiFi Reads [High fidelity reads]</span>
                        <span class="visible-mobile">Next</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="margin-left: -1rem">
    <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;Table of Contents</p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>
  </div>
</div>





  



  



  



  



  






    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">Search</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">Keyword</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <span>banner</span> <i class="iconfont icon-love"></i> <span>2021-2022©</span> 
    </div>
  
  
    <div class="statistics">
  
  

  
    
      <span id="leancloud-site-pv-container" style="display: none">
        总访问量 
        <span id="leancloud-site-pv"></span>
         次
      </span>
    
    
      <span id="leancloud-site-uv-container" style="display: none">
        总访客数 
        <span id="leancloud-site-uv"></span>
         人
      </span>
    
    

  
</div>

  
  
  
</div>

    <div style="height: 200px">
      <script type="text/javascript" src="//rf.revolvermaps.com/0/0/8.js?i=5knf34gs07t&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=33" async="async"></script>
    </div>
  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.0/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/GeekFocus/js/events.js" ></script>
<script  src="/GeekFocus/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>






  
    <script  src="/GeekFocus/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.18.2/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      headingSelector : CONFIG.toc.headingSelector || 'h1,h2,h3,h4,h5,h6',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      collapseDepth   : CONFIG.toc.collapseDepth || 0,
      scrollSmooth    : true,
      headingsOffset  : -boardTop
    });
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }
  });
</script>


  <script>
  (function() {
    var enableLang = CONFIG.code_language.enable && CONFIG.code_language.default;
    var enableCopy = CONFIG.copy_btn;
    if (!enableLang && !enableCopy) {
      return;
    }

    function getBgClass(ele) {
      return Fluid.utils.getBackgroundLightness(ele) >= 0 ? 'code-widget-light' : 'code-widget-dark';
    }

    var copyTmpl = '';
    copyTmpl += '<div class="code-widget">';
    copyTmpl += 'LANG';
    copyTmpl += '</div>';
    jQuery('.markdown-body pre').each(function() {
      var $pre = jQuery(this);
      if ($pre.find('code.mermaid').length > 0) {
        return;
      }
      if ($pre.find('span.line').length > 0) {
        return;
      }

      var lang = '';

      if (enableLang) {
        lang = CONFIG.code_language.default;
        if ($pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2 && $pre.children().hasClass('hljs')) {
          lang = $pre[0].children[0].classList[1];
        } else if ($pre[0].getAttribute('data-language')) {
          lang = $pre[0].getAttribute('data-language');
        } else if ($pre.parent().hasClass('sourceCode') && $pre[0].children.length > 0 && $pre[0].children[0].classList.length >= 2) {
          lang = $pre[0].children[0].classList[1];
          $pre.parent().addClass('code-wrapper');
        } else if ($pre.parent().hasClass('markdown-body') && $pre[0].classList.length === 0) {
          $pre.wrap('<div class="code-wrapper"></div>');
        }
        lang = lang.toUpperCase().replace('NONE', CONFIG.code_language.default);
      }
      $pre.append(copyTmpl.replace('LANG', lang).replace('code-widget">',
        getBgClass($pre[0]) + (enableCopy ? ' code-widget copy-btn" data-clipboard-snippet><i class="iconfont icon-copy"></i>' : ' code-widget">')));

      if (enableCopy) {
        Fluid.utils.createScript('https://lib.baomitu.com/clipboard.js/2.0.10/clipboard.min.js', function() {
          var clipboard = new window.ClipboardJS('.copy-btn', {
            target: function(trigger) {
              var nodes = trigger.parentNode.childNodes;
              for (var i = 0; i < nodes.length; i++) {
                if (nodes[i].tagName === 'CODE') {
                  return nodes[i];
                }
              }
            }
          });
          clipboard.on('success', function(e) {
            e.clearSelection();
            e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-copy', 'icon-success');
            setTimeout(function() {
              e.trigger.innerHTML = e.trigger.innerHTML.replace('icon-success', 'icon-copy');
            }, 2000);
          });
        });
      }
    });
  })();
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/GeekFocus/js/local-search.js" ></script>

  <script defer src="/GeekFocus/js/leancloud.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/GeekFocus/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">Blog works best with JavaScript enabled</div>
  </noscript>
</body>
</html>
